// @pulumi/aws-native/types/enums/connect/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserPhoneType = exports.QuickConnectType = exports.HoursOfOperationConfigDay = exports.ContactFlowType = exports.ContactFlowState = exports.ContactFlowModuleStatus = exports.ContactFlowModuleState = void 0;
exports.ContactFlowModuleState = {
    Active: "ACTIVE",
    Archived: "ARCHIVED",
};
exports.ContactFlowModuleStatus = {
    Published: "PUBLISHED",
    Saved: "SAVED",
};
exports.ContactFlowState = {
    Active: "ACTIVE",
    Archived: "ARCHIVED",
};
exports.ContactFlowType = {
    ContactFlow: "CONTACT_FLOW",
    CustomerQueue: "CUSTOMER_QUEUE",
    CustomerHold: "CUSTOMER_HOLD",
    CustomerWhisper: "CUSTOMER_WHISPER",
    AgentHold: "AGENT_HOLD",
    AgentWhisper: "AGENT_WHISPER",
    OutboundWhisper: "OUTBOUND_WHISPER",
    AgentTransfer: "AGENT_TRANSFER",
    QueueTransfer: "QUEUE_TRANSFER",
};
exports.HoursOfOperationConfigDay = {
    Sunday: "SUNDAY",
    Monday: "MONDAY",
    Tuesday: "TUESDAY",
    Wednesday: "WEDNESDAY",
    Thursday: "THURSDAY",
    Friday: "FRIDAY",
    Saturday: "SATURDAY",
};
exports.QuickConnectType = {
    PhoneNumber: "PHONE_NUMBER",
    Queue: "QUEUE",
    User: "USER",
};
exports.UserPhoneType = {
    SoftPhone: "SOFT_PHONE",
    DeskPhone: "DESK_PHONE",
};
