// @pulumi/aws-native/connect/userHierarchyGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserHierarchyGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Connect::UserHierarchyGroup
 */
class UserHierarchyGroup extends pulumi.CustomResource {
    /**
     * Create a UserHierarchyGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.instanceArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceArn'");
            }
            inputs["instanceArn"] = args ? args.instanceArn : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["parentGroupArn"] = args ? args.parentGroupArn : undefined;
            inputs["userHierarchyGroupArn"] = undefined /*out*/;
        }
        else {
            inputs["instanceArn"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["parentGroupArn"] = undefined /*out*/;
            inputs["userHierarchyGroupArn"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(UserHierarchyGroup.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing UserHierarchyGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new UserHierarchyGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of UserHierarchyGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === UserHierarchyGroup.__pulumiType;
    }
}
exports.UserHierarchyGroup = UserHierarchyGroup;
/** @internal */
UserHierarchyGroup.__pulumiType = 'aws-native:connect:UserHierarchyGroup';
