// @pulumi/aws-native/connect/user.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Connect::User
 */
class User extends pulumi.CustomResource {
    /**
     * Create a User resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.instanceArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceArn'");
            }
            if ((!args || args.phoneConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'phoneConfig'");
            }
            if ((!args || args.routingProfileArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'routingProfileArn'");
            }
            if ((!args || args.securityProfileArns === undefined) && !opts.urn) {
                throw new Error("Missing required property 'securityProfileArns'");
            }
            if ((!args || args.username === undefined) && !opts.urn) {
                throw new Error("Missing required property 'username'");
            }
            inputs["directoryUserId"] = args ? args.directoryUserId : undefined;
            inputs["hierarchyGroupArn"] = args ? args.hierarchyGroupArn : undefined;
            inputs["identityInfo"] = args ? args.identityInfo : undefined;
            inputs["instanceArn"] = args ? args.instanceArn : undefined;
            inputs["password"] = args ? args.password : undefined;
            inputs["phoneConfig"] = args ? args.phoneConfig : undefined;
            inputs["routingProfileArn"] = args ? args.routingProfileArn : undefined;
            inputs["securityProfileArns"] = args ? args.securityProfileArns : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["username"] = args ? args.username : undefined;
            inputs["userArn"] = undefined /*out*/;
        }
        else {
            inputs["directoryUserId"] = undefined /*out*/;
            inputs["hierarchyGroupArn"] = undefined /*out*/;
            inputs["identityInfo"] = undefined /*out*/;
            inputs["instanceArn"] = undefined /*out*/;
            inputs["password"] = undefined /*out*/;
            inputs["phoneConfig"] = undefined /*out*/;
            inputs["routingProfileArn"] = undefined /*out*/;
            inputs["securityProfileArns"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["userArn"] = undefined /*out*/;
            inputs["username"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(User.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing User resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new User(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of User.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === User.__pulumiType;
    }
}
exports.User = User;
/** @internal */
User.__pulumiType = 'aws-native:connect:User';
