// @pulumi/aws-native/elasticache/cacheCluster.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CacheCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::ElastiCache::CacheCluster
 *
 * @deprecated CacheCluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class CacheCluster extends pulumi.CustomResource {
    /**
     * Create a CacheCluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated CacheCluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("CacheCluster is deprecated: CacheCluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.cacheNodeType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cacheNodeType'");
            }
            if ((!args || args.engine === undefined) && !opts.urn) {
                throw new Error("Missing required property 'engine'");
            }
            if ((!args || args.numCacheNodes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'numCacheNodes'");
            }
            inputs["aZMode"] = args ? args.aZMode : undefined;
            inputs["autoMinorVersionUpgrade"] = args ? args.autoMinorVersionUpgrade : undefined;
            inputs["cacheNodeType"] = args ? args.cacheNodeType : undefined;
            inputs["cacheParameterGroupName"] = args ? args.cacheParameterGroupName : undefined;
            inputs["cacheSecurityGroupNames"] = args ? args.cacheSecurityGroupNames : undefined;
            inputs["cacheSubnetGroupName"] = args ? args.cacheSubnetGroupName : undefined;
            inputs["clusterName"] = args ? args.clusterName : undefined;
            inputs["configurationEndpointAddress"] = args ? args.configurationEndpointAddress : undefined;
            inputs["configurationEndpointPort"] = args ? args.configurationEndpointPort : undefined;
            inputs["engine"] = args ? args.engine : undefined;
            inputs["engineVersion"] = args ? args.engineVersion : undefined;
            inputs["logDeliveryConfigurations"] = args ? args.logDeliveryConfigurations : undefined;
            inputs["notificationTopicArn"] = args ? args.notificationTopicArn : undefined;
            inputs["numCacheNodes"] = args ? args.numCacheNodes : undefined;
            inputs["port"] = args ? args.port : undefined;
            inputs["preferredAvailabilityZone"] = args ? args.preferredAvailabilityZone : undefined;
            inputs["preferredAvailabilityZones"] = args ? args.preferredAvailabilityZones : undefined;
            inputs["preferredMaintenanceWindow"] = args ? args.preferredMaintenanceWindow : undefined;
            inputs["redisEndpointAddress"] = args ? args.redisEndpointAddress : undefined;
            inputs["redisEndpointPort"] = args ? args.redisEndpointPort : undefined;
            inputs["snapshotArns"] = args ? args.snapshotArns : undefined;
            inputs["snapshotName"] = args ? args.snapshotName : undefined;
            inputs["snapshotRetentionLimit"] = args ? args.snapshotRetentionLimit : undefined;
            inputs["snapshotWindow"] = args ? args.snapshotWindow : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["vpcSecurityGroupIds"] = args ? args.vpcSecurityGroupIds : undefined;
        }
        else {
            inputs["aZMode"] = undefined /*out*/;
            inputs["autoMinorVersionUpgrade"] = undefined /*out*/;
            inputs["cacheNodeType"] = undefined /*out*/;
            inputs["cacheParameterGroupName"] = undefined /*out*/;
            inputs["cacheSecurityGroupNames"] = undefined /*out*/;
            inputs["cacheSubnetGroupName"] = undefined /*out*/;
            inputs["clusterName"] = undefined /*out*/;
            inputs["configurationEndpointAddress"] = undefined /*out*/;
            inputs["configurationEndpointPort"] = undefined /*out*/;
            inputs["engine"] = undefined /*out*/;
            inputs["engineVersion"] = undefined /*out*/;
            inputs["logDeliveryConfigurations"] = undefined /*out*/;
            inputs["notificationTopicArn"] = undefined /*out*/;
            inputs["numCacheNodes"] = undefined /*out*/;
            inputs["port"] = undefined /*out*/;
            inputs["preferredAvailabilityZone"] = undefined /*out*/;
            inputs["preferredAvailabilityZones"] = undefined /*out*/;
            inputs["preferredMaintenanceWindow"] = undefined /*out*/;
            inputs["redisEndpointAddress"] = undefined /*out*/;
            inputs["redisEndpointPort"] = undefined /*out*/;
            inputs["snapshotArns"] = undefined /*out*/;
            inputs["snapshotName"] = undefined /*out*/;
            inputs["snapshotRetentionLimit"] = undefined /*out*/;
            inputs["snapshotWindow"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["vpcSecurityGroupIds"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(CacheCluster.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing CacheCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("CacheCluster is deprecated: CacheCluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new CacheCluster(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CacheCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CacheCluster.__pulumiType;
    }
}
exports.CacheCluster = CacheCluster;
/** @internal */
CacheCluster.__pulumiType = 'aws-native:elasticache:CacheCluster';
