// @pulumi/aws-native/apigatewayv2/apiGatewayManagedOverrides.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiGatewayManagedOverrides = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::ApiGatewayV2::ApiGatewayManagedOverrides
 *
 * @deprecated ApiGatewayManagedOverrides is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class ApiGatewayManagedOverrides extends pulumi.CustomResource {
    /**
     * Create a ApiGatewayManagedOverrides resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated ApiGatewayManagedOverrides is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("ApiGatewayManagedOverrides is deprecated: ApiGatewayManagedOverrides is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.apiId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'apiId'");
            }
            inputs["apiId"] = args ? args.apiId : undefined;
            inputs["integration"] = args ? args.integration : undefined;
            inputs["route"] = args ? args.route : undefined;
            inputs["stage"] = args ? args.stage : undefined;
        }
        else {
            inputs["apiId"] = undefined /*out*/;
            inputs["integration"] = undefined /*out*/;
            inputs["route"] = undefined /*out*/;
            inputs["stage"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ApiGatewayManagedOverrides.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ApiGatewayManagedOverrides resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ApiGatewayManagedOverrides is deprecated: ApiGatewayManagedOverrides is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new ApiGatewayManagedOverrides(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ApiGatewayManagedOverrides.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ApiGatewayManagedOverrides.__pulumiType;
    }
}
exports.ApiGatewayManagedOverrides = ApiGatewayManagedOverrides;
/** @internal */
ApiGatewayManagedOverrides.__pulumiType = 'aws-native:apigatewayv2:ApiGatewayManagedOverrides';
