// @pulumi/aws-native/apigatewayv2/domainName.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainName = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::ApiGatewayV2::DomainName
 *
 * @deprecated DomainName is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class DomainName extends pulumi.CustomResource {
    /**
     * Create a DomainName resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated DomainName is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("DomainName is deprecated: DomainName is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.domainName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'domainName'");
            }
            inputs["domainName"] = args ? args.domainName : undefined;
            inputs["domainNameConfigurations"] = args ? args.domainNameConfigurations : undefined;
            inputs["mutualTlsAuthentication"] = args ? args.mutualTlsAuthentication : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["regionalDomainName"] = undefined /*out*/;
            inputs["regionalHostedZoneId"] = undefined /*out*/;
        }
        else {
            inputs["domainName"] = undefined /*out*/;
            inputs["domainNameConfigurations"] = undefined /*out*/;
            inputs["mutualTlsAuthentication"] = undefined /*out*/;
            inputs["regionalDomainName"] = undefined /*out*/;
            inputs["regionalHostedZoneId"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(DomainName.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DomainName resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("DomainName is deprecated: DomainName is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new DomainName(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DomainName.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DomainName.__pulumiType;
    }
}
exports.DomainName = DomainName;
/** @internal */
DomainName.__pulumiType = 'aws-native:apigatewayv2:DomainName';
