// @pulumi/aws-native/glue/devEndpoint.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DevEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Glue::DevEndpoint
 *
 * @deprecated DevEndpoint is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class DevEndpoint extends pulumi.CustomResource {
    /**
     * Create a DevEndpoint resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated DevEndpoint is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("DevEndpoint is deprecated: DevEndpoint is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            inputs["arguments"] = args ? args.arguments : undefined;
            inputs["endpointName"] = args ? args.endpointName : undefined;
            inputs["extraJarsS3Path"] = args ? args.extraJarsS3Path : undefined;
            inputs["extraPythonLibsS3Path"] = args ? args.extraPythonLibsS3Path : undefined;
            inputs["glueVersion"] = args ? args.glueVersion : undefined;
            inputs["numberOfNodes"] = args ? args.numberOfNodes : undefined;
            inputs["numberOfWorkers"] = args ? args.numberOfWorkers : undefined;
            inputs["publicKey"] = args ? args.publicKey : undefined;
            inputs["publicKeys"] = args ? args.publicKeys : undefined;
            inputs["roleArn"] = args ? args.roleArn : undefined;
            inputs["securityConfiguration"] = args ? args.securityConfiguration : undefined;
            inputs["securityGroupIds"] = args ? args.securityGroupIds : undefined;
            inputs["subnetId"] = args ? args.subnetId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["workerType"] = args ? args.workerType : undefined;
        }
        else {
            inputs["arguments"] = undefined /*out*/;
            inputs["endpointName"] = undefined /*out*/;
            inputs["extraJarsS3Path"] = undefined /*out*/;
            inputs["extraPythonLibsS3Path"] = undefined /*out*/;
            inputs["glueVersion"] = undefined /*out*/;
            inputs["numberOfNodes"] = undefined /*out*/;
            inputs["numberOfWorkers"] = undefined /*out*/;
            inputs["publicKey"] = undefined /*out*/;
            inputs["publicKeys"] = undefined /*out*/;
            inputs["roleArn"] = undefined /*out*/;
            inputs["securityConfiguration"] = undefined /*out*/;
            inputs["securityGroupIds"] = undefined /*out*/;
            inputs["subnetId"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["workerType"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(DevEndpoint.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DevEndpoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("DevEndpoint is deprecated: DevEndpoint is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new DevEndpoint(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DevEndpoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DevEndpoint.__pulumiType;
    }
}
exports.DevEndpoint = DevEndpoint;
/** @internal */
DevEndpoint.__pulumiType = 'aws-native:glue:DevEndpoint';
