// @pulumi/aws-native/glue/mltransform.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MLTransform = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Glue::MLTransform
 *
 * @deprecated MLTransform is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class MLTransform extends pulumi.CustomResource {
    /**
     * Create a MLTransform resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated MLTransform is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("MLTransform is deprecated: MLTransform is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.inputRecordTables === undefined) && !opts.urn) {
                throw new Error("Missing required property 'inputRecordTables'");
            }
            if ((!args || args.role === undefined) && !opts.urn) {
                throw new Error("Missing required property 'role'");
            }
            if ((!args || args.transformParameters === undefined) && !opts.urn) {
                throw new Error("Missing required property 'transformParameters'");
            }
            inputs["description"] = args ? args.description : undefined;
            inputs["glueVersion"] = args ? args.glueVersion : undefined;
            inputs["inputRecordTables"] = args ? args.inputRecordTables : undefined;
            inputs["maxCapacity"] = args ? args.maxCapacity : undefined;
            inputs["maxRetries"] = args ? args.maxRetries : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["numberOfWorkers"] = args ? args.numberOfWorkers : undefined;
            inputs["role"] = args ? args.role : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["timeout"] = args ? args.timeout : undefined;
            inputs["transformEncryption"] = args ? args.transformEncryption : undefined;
            inputs["transformParameters"] = args ? args.transformParameters : undefined;
            inputs["workerType"] = args ? args.workerType : undefined;
        }
        else {
            inputs["description"] = undefined /*out*/;
            inputs["glueVersion"] = undefined /*out*/;
            inputs["inputRecordTables"] = undefined /*out*/;
            inputs["maxCapacity"] = undefined /*out*/;
            inputs["maxRetries"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["numberOfWorkers"] = undefined /*out*/;
            inputs["role"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["timeout"] = undefined /*out*/;
            inputs["transformEncryption"] = undefined /*out*/;
            inputs["transformParameters"] = undefined /*out*/;
            inputs["workerType"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(MLTransform.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing MLTransform resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("MLTransform is deprecated: MLTransform is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new MLTransform(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MLTransform.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MLTransform.__pulumiType;
    }
}
exports.MLTransform = MLTransform;
/** @internal */
MLTransform.__pulumiType = 'aws-native:glue:MLTransform';
