// @pulumi/aws-native/glue/schema.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schema = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * This resource represents a schema of Glue Schema Registry.
 */
class Schema extends pulumi.CustomResource {
    /**
     * Create a Schema resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.compatibility === undefined) && !opts.urn) {
                throw new Error("Missing required property 'compatibility'");
            }
            if ((!args || args.dataFormat === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataFormat'");
            }
            if ((!args || args.schemaDefinition === undefined) && !opts.urn) {
                throw new Error("Missing required property 'schemaDefinition'");
            }
            inputs["checkpointVersion"] = args ? args.checkpointVersion : undefined;
            inputs["compatibility"] = args ? args.compatibility : undefined;
            inputs["dataFormat"] = args ? args.dataFormat : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["registry"] = args ? args.registry : undefined;
            inputs["schemaDefinition"] = args ? args.schemaDefinition : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["initialSchemaVersionId"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["checkpointVersion"] = undefined /*out*/;
            inputs["compatibility"] = undefined /*out*/;
            inputs["dataFormat"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["initialSchemaVersionId"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["registry"] = undefined /*out*/;
            inputs["schemaDefinition"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Schema.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Schema resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Schema(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Schema.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Schema.__pulumiType;
    }
}
exports.Schema = Schema;
/** @internal */
Schema.__pulumiType = 'aws-native:glue:Schema';
