// @pulumi/aws-native/apigateway/usagePlan.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsagePlan = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::ApiGateway::UsagePlan
 */
class UsagePlan extends pulumi.CustomResource {
    /**
     * Create a UsagePlan resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["apiStages"] = args ? args.apiStages : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["quota"] = args ? args.quota : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["throttle"] = args ? args.throttle : undefined;
            inputs["usagePlanName"] = args ? args.usagePlanName : undefined;
        }
        else {
            inputs["apiStages"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["quota"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["throttle"] = undefined /*out*/;
            inputs["usagePlanName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(UsagePlan.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing UsagePlan resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new UsagePlan(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of UsagePlan.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === UsagePlan.__pulumiType;
    }
}
exports.UsagePlan = UsagePlan;
/** @internal */
UsagePlan.__pulumiType = 'aws-native:apigateway:UsagePlan';
