// @pulumi/aws-native/fsx/fileSystem.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileSystem = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::FSx::FileSystem
 *
 * @deprecated FileSystem is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class FileSystem extends pulumi.CustomResource {
    /**
     * Create a FileSystem resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated FileSystem is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("FileSystem is deprecated: FileSystem is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fileSystemType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fileSystemType'");
            }
            if ((!args || args.subnetIds === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subnetIds'");
            }
            inputs["backupId"] = args ? args.backupId : undefined;
            inputs["fileSystemType"] = args ? args.fileSystemType : undefined;
            inputs["fileSystemTypeVersion"] = args ? args.fileSystemTypeVersion : undefined;
            inputs["kmsKeyId"] = args ? args.kmsKeyId : undefined;
            inputs["lustreConfiguration"] = args ? args.lustreConfiguration : undefined;
            inputs["ontapConfiguration"] = args ? args.ontapConfiguration : undefined;
            inputs["securityGroupIds"] = args ? args.securityGroupIds : undefined;
            inputs["storageCapacity"] = args ? args.storageCapacity : undefined;
            inputs["storageType"] = args ? args.storageType : undefined;
            inputs["subnetIds"] = args ? args.subnetIds : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["windowsConfiguration"] = args ? args.windowsConfiguration : undefined;
            inputs["dNSName"] = undefined /*out*/;
            inputs["lustreMountName"] = undefined /*out*/;
        }
        else {
            inputs["backupId"] = undefined /*out*/;
            inputs["dNSName"] = undefined /*out*/;
            inputs["fileSystemType"] = undefined /*out*/;
            inputs["fileSystemTypeVersion"] = undefined /*out*/;
            inputs["kmsKeyId"] = undefined /*out*/;
            inputs["lustreConfiguration"] = undefined /*out*/;
            inputs["lustreMountName"] = undefined /*out*/;
            inputs["ontapConfiguration"] = undefined /*out*/;
            inputs["securityGroupIds"] = undefined /*out*/;
            inputs["storageCapacity"] = undefined /*out*/;
            inputs["storageType"] = undefined /*out*/;
            inputs["subnetIds"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["windowsConfiguration"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(FileSystem.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing FileSystem resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("FileSystem is deprecated: FileSystem is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new FileSystem(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FileSystem.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FileSystem.__pulumiType;
    }
}
exports.FileSystem = FileSystem;
/** @internal */
FileSystem.__pulumiType = 'aws-native:fsx:FileSystem';
