// @pulumi/aws-native/databrew/job.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Job = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::DataBrew::Job.
 */
class Job extends pulumi.CustomResource {
    /**
     * Create a Job resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            inputs["dataCatalogOutputs"] = args ? args.dataCatalogOutputs : undefined;
            inputs["databaseOutputs"] = args ? args.databaseOutputs : undefined;
            inputs["datasetName"] = args ? args.datasetName : undefined;
            inputs["encryptionKeyArn"] = args ? args.encryptionKeyArn : undefined;
            inputs["encryptionMode"] = args ? args.encryptionMode : undefined;
            inputs["jobSample"] = args ? args.jobSample : undefined;
            inputs["logSubscription"] = args ? args.logSubscription : undefined;
            inputs["maxCapacity"] = args ? args.maxCapacity : undefined;
            inputs["maxRetries"] = args ? args.maxRetries : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["outputLocation"] = args ? args.outputLocation : undefined;
            inputs["outputs"] = args ? args.outputs : undefined;
            inputs["profileConfiguration"] = args ? args.profileConfiguration : undefined;
            inputs["projectName"] = args ? args.projectName : undefined;
            inputs["recipe"] = args ? args.recipe : undefined;
            inputs["roleArn"] = args ? args.roleArn : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["timeout"] = args ? args.timeout : undefined;
            inputs["type"] = args ? args.type : undefined;
            inputs["validationConfigurations"] = args ? args.validationConfigurations : undefined;
        }
        else {
            inputs["dataCatalogOutputs"] = undefined /*out*/;
            inputs["databaseOutputs"] = undefined /*out*/;
            inputs["datasetName"] = undefined /*out*/;
            inputs["encryptionKeyArn"] = undefined /*out*/;
            inputs["encryptionMode"] = undefined /*out*/;
            inputs["jobSample"] = undefined /*out*/;
            inputs["logSubscription"] = undefined /*out*/;
            inputs["maxCapacity"] = undefined /*out*/;
            inputs["maxRetries"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["outputLocation"] = undefined /*out*/;
            inputs["outputs"] = undefined /*out*/;
            inputs["profileConfiguration"] = undefined /*out*/;
            inputs["projectName"] = undefined /*out*/;
            inputs["recipe"] = undefined /*out*/;
            inputs["roleArn"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["timeout"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["validationConfigurations"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Job.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Job resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Job(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Job.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Job.__pulumiType;
    }
}
exports.Job = Job;
/** @internal */
Job.__pulumiType = 'aws-native:databrew:Job';
