// @pulumi/aws-native/opsworkscm/server.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Server = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::OpsWorksCM::Server
 */
class Server extends pulumi.CustomResource {
    /**
     * Create a Server resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.instanceProfileArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceProfileArn'");
            }
            if ((!args || args.instanceType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceType'");
            }
            if ((!args || args.serviceRoleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceRoleArn'");
            }
            inputs["associatePublicIpAddress"] = args ? args.associatePublicIpAddress : undefined;
            inputs["backupId"] = args ? args.backupId : undefined;
            inputs["backupRetentionCount"] = args ? args.backupRetentionCount : undefined;
            inputs["customCertificate"] = args ? args.customCertificate : undefined;
            inputs["customDomain"] = args ? args.customDomain : undefined;
            inputs["customPrivateKey"] = args ? args.customPrivateKey : undefined;
            inputs["disableAutomatedBackup"] = args ? args.disableAutomatedBackup : undefined;
            inputs["engine"] = args ? args.engine : undefined;
            inputs["engineAttributes"] = args ? args.engineAttributes : undefined;
            inputs["engineModel"] = args ? args.engineModel : undefined;
            inputs["engineVersion"] = args ? args.engineVersion : undefined;
            inputs["instanceProfileArn"] = args ? args.instanceProfileArn : undefined;
            inputs["instanceType"] = args ? args.instanceType : undefined;
            inputs["keyPair"] = args ? args.keyPair : undefined;
            inputs["preferredBackupWindow"] = args ? args.preferredBackupWindow : undefined;
            inputs["preferredMaintenanceWindow"] = args ? args.preferredMaintenanceWindow : undefined;
            inputs["securityGroupIds"] = args ? args.securityGroupIds : undefined;
            inputs["serverName"] = args ? args.serverName : undefined;
            inputs["serviceRoleArn"] = args ? args.serviceRoleArn : undefined;
            inputs["subnetIds"] = args ? args.subnetIds : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["endpoint"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["associatePublicIpAddress"] = undefined /*out*/;
            inputs["backupId"] = undefined /*out*/;
            inputs["backupRetentionCount"] = undefined /*out*/;
            inputs["customCertificate"] = undefined /*out*/;
            inputs["customDomain"] = undefined /*out*/;
            inputs["customPrivateKey"] = undefined /*out*/;
            inputs["disableAutomatedBackup"] = undefined /*out*/;
            inputs["endpoint"] = undefined /*out*/;
            inputs["engine"] = undefined /*out*/;
            inputs["engineAttributes"] = undefined /*out*/;
            inputs["engineModel"] = undefined /*out*/;
            inputs["engineVersion"] = undefined /*out*/;
            inputs["instanceProfileArn"] = undefined /*out*/;
            inputs["instanceType"] = undefined /*out*/;
            inputs["keyPair"] = undefined /*out*/;
            inputs["preferredBackupWindow"] = undefined /*out*/;
            inputs["preferredMaintenanceWindow"] = undefined /*out*/;
            inputs["securityGroupIds"] = undefined /*out*/;
            inputs["serverName"] = undefined /*out*/;
            inputs["serviceRoleArn"] = undefined /*out*/;
            inputs["subnetIds"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Server.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Server resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Server(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Server.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Server.__pulumiType;
    }
}
exports.Server = Server;
/** @internal */
Server.__pulumiType = 'aws-native:opsworkscm:Server';
