// @pulumi/aws-native/nimblestudio/studio.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Studio = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a studio that contains other Nimble Studio resources
 */
class Studio extends pulumi.CustomResource {
    /**
     * Create a Studio resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.adminRoleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'adminRoleArn'");
            }
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.userRoleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'userRoleArn'");
            }
            inputs["adminRoleArn"] = args ? args.adminRoleArn : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["studioEncryptionConfiguration"] = args ? args.studioEncryptionConfiguration : undefined;
            inputs["studioName"] = args ? args.studioName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["userRoleArn"] = args ? args.userRoleArn : undefined;
            inputs["homeRegion"] = undefined /*out*/;
            inputs["ssoClientId"] = undefined /*out*/;
            inputs["studioId"] = undefined /*out*/;
            inputs["studioUrl"] = undefined /*out*/;
        }
        else {
            inputs["adminRoleArn"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["homeRegion"] = undefined /*out*/;
            inputs["ssoClientId"] = undefined /*out*/;
            inputs["studioEncryptionConfiguration"] = undefined /*out*/;
            inputs["studioId"] = undefined /*out*/;
            inputs["studioName"] = undefined /*out*/;
            inputs["studioUrl"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["userRoleArn"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Studio.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Studio resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Studio(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Studio.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Studio.__pulumiType;
    }
}
exports.Studio = Studio;
/** @internal */
Studio.__pulumiType = 'aws-native:nimblestudio:Studio';
