// @pulumi/aws-native/nimblestudio/studioComponent.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StudioComponent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a studio component which connects a non-Nimble Studio resource in your account to your studio
 */
class StudioComponent extends pulumi.CustomResource {
    /**
     * Create a StudioComponent resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.studioId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'studioId'");
            }
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            inputs["configuration"] = args ? args.configuration : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["ec2SecurityGroupIds"] = args ? args.ec2SecurityGroupIds : undefined;
            inputs["initializationScripts"] = args ? args.initializationScripts : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["scriptParameters"] = args ? args.scriptParameters : undefined;
            inputs["studioId"] = args ? args.studioId : undefined;
            inputs["subtype"] = args ? args.subtype : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["type"] = args ? args.type : undefined;
            inputs["studioComponentId"] = undefined /*out*/;
        }
        else {
            inputs["configuration"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["ec2SecurityGroupIds"] = undefined /*out*/;
            inputs["initializationScripts"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["scriptParameters"] = undefined /*out*/;
            inputs["studioComponentId"] = undefined /*out*/;
            inputs["studioId"] = undefined /*out*/;
            inputs["subtype"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(StudioComponent.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing StudioComponent resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StudioComponent(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StudioComponent.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StudioComponent.__pulumiType;
    }
}
exports.StudioComponent = StudioComponent;
/** @internal */
StudioComponent.__pulumiType = 'aws-native:nimblestudio:StudioComponent';
