// @pulumi/aws-native/neptune/dbinstance.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DBInstance = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Neptune::DBInstance
 *
 * @deprecated DBInstance is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class DBInstance extends pulumi.CustomResource {
    /**
     * Create a DBInstance resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated DBInstance is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("DBInstance is deprecated: DBInstance is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dBInstanceClass === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dBInstanceClass'");
            }
            inputs["allowMajorVersionUpgrade"] = args ? args.allowMajorVersionUpgrade : undefined;
            inputs["autoMinorVersionUpgrade"] = args ? args.autoMinorVersionUpgrade : undefined;
            inputs["availabilityZone"] = args ? args.availabilityZone : undefined;
            inputs["dBClusterIdentifier"] = args ? args.dBClusterIdentifier : undefined;
            inputs["dBInstanceClass"] = args ? args.dBInstanceClass : undefined;
            inputs["dBInstanceIdentifier"] = args ? args.dBInstanceIdentifier : undefined;
            inputs["dBParameterGroupName"] = args ? args.dBParameterGroupName : undefined;
            inputs["dBSnapshotIdentifier"] = args ? args.dBSnapshotIdentifier : undefined;
            inputs["dBSubnetGroupName"] = args ? args.dBSubnetGroupName : undefined;
            inputs["preferredMaintenanceWindow"] = args ? args.preferredMaintenanceWindow : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["endpoint"] = undefined /*out*/;
            inputs["port"] = undefined /*out*/;
        }
        else {
            inputs["allowMajorVersionUpgrade"] = undefined /*out*/;
            inputs["autoMinorVersionUpgrade"] = undefined /*out*/;
            inputs["availabilityZone"] = undefined /*out*/;
            inputs["dBClusterIdentifier"] = undefined /*out*/;
            inputs["dBInstanceClass"] = undefined /*out*/;
            inputs["dBInstanceIdentifier"] = undefined /*out*/;
            inputs["dBParameterGroupName"] = undefined /*out*/;
            inputs["dBSnapshotIdentifier"] = undefined /*out*/;
            inputs["dBSubnetGroupName"] = undefined /*out*/;
            inputs["endpoint"] = undefined /*out*/;
            inputs["port"] = undefined /*out*/;
            inputs["preferredMaintenanceWindow"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(DBInstance.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DBInstance resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("DBInstance is deprecated: DBInstance is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new DBInstance(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DBInstance.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DBInstance.__pulumiType;
    }
}
exports.DBInstance = DBInstance;
/** @internal */
DBInstance.__pulumiType = 'aws-native:neptune:DBInstance';
