// @pulumi/aws-native/codecommit/repository.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Repository = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::CodeCommit::Repository
 *
 * @deprecated Repository is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Repository extends pulumi.CustomResource {
    /**
     * Create a Repository resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Repository is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Repository is deprecated: Repository is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["code"] = args ? args.code : undefined;
            inputs["repositoryDescription"] = args ? args.repositoryDescription : undefined;
            inputs["repositoryName"] = args ? args.repositoryName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["triggers"] = args ? args.triggers : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["cloneUrlHttp"] = undefined /*out*/;
            inputs["cloneUrlSsh"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["cloneUrlHttp"] = undefined /*out*/;
            inputs["cloneUrlSsh"] = undefined /*out*/;
            inputs["code"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["repositoryDescription"] = undefined /*out*/;
            inputs["repositoryName"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["triggers"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Repository.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Repository resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Repository is deprecated: Repository is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Repository(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Repository.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Repository.__pulumiType;
    }
}
exports.Repository = Repository;
/** @internal */
Repository.__pulumiType = 'aws-native:codecommit:Repository';
