// @pulumi/aws-native/customerprofiles/domain.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Domain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A domain defined for 3rd party data source in Profile Service
 */
class Domain extends pulumi.CustomResource {
    /**
     * Create a Domain resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["deadLetterQueueUrl"] = args ? args.deadLetterQueueUrl : undefined;
            inputs["defaultEncryptionKey"] = args ? args.defaultEncryptionKey : undefined;
            inputs["defaultExpirationDays"] = args ? args.defaultExpirationDays : undefined;
            inputs["domainName"] = args ? args.domainName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["createdAt"] = undefined /*out*/;
            inputs["lastUpdatedAt"] = undefined /*out*/;
        }
        else {
            inputs["createdAt"] = undefined /*out*/;
            inputs["deadLetterQueueUrl"] = undefined /*out*/;
            inputs["defaultEncryptionKey"] = undefined /*out*/;
            inputs["defaultExpirationDays"] = undefined /*out*/;
            inputs["domainName"] = undefined /*out*/;
            inputs["lastUpdatedAt"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Domain.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Domain resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Domain(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Domain.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Domain.__pulumiType;
    }
}
exports.Domain = Domain;
/** @internal */
Domain.__pulumiType = 'aws-native:customerprofiles:Domain';
