// @pulumi/aws-native/stepfunctions/stateMachine.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StateMachine = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for StateMachine
 */
class StateMachine extends pulumi.CustomResource {
    /**
     * Create a StateMachine resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            inputs["definition"] = args ? args.definition : undefined;
            inputs["definitionS3Location"] = args ? args.definitionS3Location : undefined;
            inputs["definitionString"] = args ? args.definitionString : undefined;
            inputs["definitionSubstitutions"] = args ? args.definitionSubstitutions : undefined;
            inputs["loggingConfiguration"] = args ? args.loggingConfiguration : undefined;
            inputs["roleArn"] = args ? args.roleArn : undefined;
            inputs["stateMachineName"] = args ? args.stateMachineName : undefined;
            inputs["stateMachineType"] = args ? args.stateMachineType : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["tracingConfiguration"] = args ? args.tracingConfiguration : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["definition"] = undefined /*out*/;
            inputs["definitionS3Location"] = undefined /*out*/;
            inputs["definitionString"] = undefined /*out*/;
            inputs["definitionSubstitutions"] = undefined /*out*/;
            inputs["loggingConfiguration"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["roleArn"] = undefined /*out*/;
            inputs["stateMachineName"] = undefined /*out*/;
            inputs["stateMachineType"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["tracingConfiguration"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(StateMachine.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing StateMachine resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StateMachine(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StateMachine.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StateMachine.__pulumiType;
    }
}
exports.StateMachine = StateMachine;
/** @internal */
StateMachine.__pulumiType = 'aws-native:stepfunctions:StateMachine';
