// @pulumi/aws-native/evidently/feature.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Feature = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Evidently::Feature.
 */
class Feature extends pulumi.CustomResource {
    /**
     * Create a Feature resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            if ((!args || args.variations === undefined) && !opts.urn) {
                throw new Error("Missing required property 'variations'");
            }
            inputs["defaultVariation"] = args ? args.defaultVariation : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["entityOverrides"] = args ? args.entityOverrides : undefined;
            inputs["evaluationStrategy"] = args ? args.evaluationStrategy : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["project"] = args ? args.project : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["variations"] = args ? args.variations : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["defaultVariation"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["entityOverrides"] = undefined /*out*/;
            inputs["evaluationStrategy"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["project"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["variations"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Feature.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Feature resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Feature(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Feature.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Feature.__pulumiType;
    }
}
exports.Feature = Feature;
/** @internal */
Feature.__pulumiType = 'aws-native:evidently:Feature';
