// @pulumi/aws-native/cloudtrail/trail.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Trail = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Creates a trail that specifies the settings for delivery of log data to an Amazon S3 bucket. A maximum of five trails can exist in a region, irrespective of the region in which they were created.
 */
class Trail extends pulumi.CustomResource {
    /**
     * Create a Trail resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.isLogging === undefined) && !opts.urn) {
                throw new Error("Missing required property 'isLogging'");
            }
            if ((!args || args.s3BucketName === undefined) && !opts.urn) {
                throw new Error("Missing required property 's3BucketName'");
            }
            inputs["cloudWatchLogsLogGroupArn"] = args ? args.cloudWatchLogsLogGroupArn : undefined;
            inputs["cloudWatchLogsRoleArn"] = args ? args.cloudWatchLogsRoleArn : undefined;
            inputs["enableLogFileValidation"] = args ? args.enableLogFileValidation : undefined;
            inputs["eventSelectors"] = args ? args.eventSelectors : undefined;
            inputs["includeGlobalServiceEvents"] = args ? args.includeGlobalServiceEvents : undefined;
            inputs["insightSelectors"] = args ? args.insightSelectors : undefined;
            inputs["isLogging"] = args ? args.isLogging : undefined;
            inputs["isMultiRegionTrail"] = args ? args.isMultiRegionTrail : undefined;
            inputs["isOrganizationTrail"] = args ? args.isOrganizationTrail : undefined;
            inputs["kMSKeyId"] = args ? args.kMSKeyId : undefined;
            inputs["s3BucketName"] = args ? args.s3BucketName : undefined;
            inputs["s3KeyPrefix"] = args ? args.s3KeyPrefix : undefined;
            inputs["snsTopicName"] = args ? args.snsTopicName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["trailName"] = args ? args.trailName : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["snsTopicArn"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["cloudWatchLogsLogGroupArn"] = undefined /*out*/;
            inputs["cloudWatchLogsRoleArn"] = undefined /*out*/;
            inputs["enableLogFileValidation"] = undefined /*out*/;
            inputs["eventSelectors"] = undefined /*out*/;
            inputs["includeGlobalServiceEvents"] = undefined /*out*/;
            inputs["insightSelectors"] = undefined /*out*/;
            inputs["isLogging"] = undefined /*out*/;
            inputs["isMultiRegionTrail"] = undefined /*out*/;
            inputs["isOrganizationTrail"] = undefined /*out*/;
            inputs["kMSKeyId"] = undefined /*out*/;
            inputs["s3BucketName"] = undefined /*out*/;
            inputs["s3KeyPrefix"] = undefined /*out*/;
            inputs["snsTopicArn"] = undefined /*out*/;
            inputs["snsTopicName"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["trailName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Trail.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Trail resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Trail(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Trail.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Trail.__pulumiType;
    }
}
exports.Trail = Trail;
/** @internal */
Trail.__pulumiType = 'aws-native:cloudtrail:Trail';
