// @pulumi/aws-native/robomaker/simulationApplication.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimulationApplication = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An example resource schema demonstrating some basic constructs and validation rules.
 */
class SimulationApplication extends pulumi.CustomResource {
    /**
     * Create a SimulationApplication resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.robotSoftwareSuite === undefined) && !opts.urn) {
                throw new Error("Missing required property 'robotSoftwareSuite'");
            }
            if ((!args || args.simulationSoftwareSuite === undefined) && !opts.urn) {
                throw new Error("Missing required property 'simulationSoftwareSuite'");
            }
            inputs["currentRevisionId"] = args ? args.currentRevisionId : undefined;
            inputs["environment"] = args ? args.environment : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["renderingEngine"] = args ? args.renderingEngine : undefined;
            inputs["robotSoftwareSuite"] = args ? args.robotSoftwareSuite : undefined;
            inputs["simulationSoftwareSuite"] = args ? args.simulationSoftwareSuite : undefined;
            inputs["sources"] = args ? args.sources : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["currentRevisionId"] = undefined /*out*/;
            inputs["environment"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["renderingEngine"] = undefined /*out*/;
            inputs["robotSoftwareSuite"] = undefined /*out*/;
            inputs["simulationSoftwareSuite"] = undefined /*out*/;
            inputs["sources"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(SimulationApplication.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SimulationApplication resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SimulationApplication(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SimulationApplication.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SimulationApplication.__pulumiType;
    }
}
exports.SimulationApplication = SimulationApplication;
/** @internal */
SimulationApplication.__pulumiType = 'aws-native:robomaker:SimulationApplication';
