// @pulumi/aws-native/ecs/service.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Service = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::ECS::Service
 */
class Service extends pulumi.CustomResource {
    /**
     * Create a Service resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["capacityProviderStrategy"] = args ? args.capacityProviderStrategy : undefined;
            inputs["cluster"] = args ? args.cluster : undefined;
            inputs["deploymentConfiguration"] = args ? args.deploymentConfiguration : undefined;
            inputs["deploymentController"] = args ? args.deploymentController : undefined;
            inputs["desiredCount"] = args ? args.desiredCount : undefined;
            inputs["enableECSManagedTags"] = args ? args.enableECSManagedTags : undefined;
            inputs["enableExecuteCommand"] = args ? args.enableExecuteCommand : undefined;
            inputs["healthCheckGracePeriodSeconds"] = args ? args.healthCheckGracePeriodSeconds : undefined;
            inputs["launchType"] = args ? args.launchType : undefined;
            inputs["loadBalancers"] = args ? args.loadBalancers : undefined;
            inputs["networkConfiguration"] = args ? args.networkConfiguration : undefined;
            inputs["placementConstraints"] = args ? args.placementConstraints : undefined;
            inputs["placementStrategies"] = args ? args.placementStrategies : undefined;
            inputs["platformVersion"] = args ? args.platformVersion : undefined;
            inputs["propagateTags"] = args ? args.propagateTags : undefined;
            inputs["role"] = args ? args.role : undefined;
            inputs["schedulingStrategy"] = args ? args.schedulingStrategy : undefined;
            inputs["serviceName"] = args ? args.serviceName : undefined;
            inputs["serviceRegistries"] = args ? args.serviceRegistries : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["taskDefinition"] = args ? args.taskDefinition : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["serviceArn"] = undefined /*out*/;
        }
        else {
            inputs["capacityProviderStrategy"] = undefined /*out*/;
            inputs["cluster"] = undefined /*out*/;
            inputs["deploymentConfiguration"] = undefined /*out*/;
            inputs["deploymentController"] = undefined /*out*/;
            inputs["desiredCount"] = undefined /*out*/;
            inputs["enableECSManagedTags"] = undefined /*out*/;
            inputs["enableExecuteCommand"] = undefined /*out*/;
            inputs["healthCheckGracePeriodSeconds"] = undefined /*out*/;
            inputs["launchType"] = undefined /*out*/;
            inputs["loadBalancers"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["networkConfiguration"] = undefined /*out*/;
            inputs["placementConstraints"] = undefined /*out*/;
            inputs["placementStrategies"] = undefined /*out*/;
            inputs["platformVersion"] = undefined /*out*/;
            inputs["propagateTags"] = undefined /*out*/;
            inputs["role"] = undefined /*out*/;
            inputs["schedulingStrategy"] = undefined /*out*/;
            inputs["serviceArn"] = undefined /*out*/;
            inputs["serviceName"] = undefined /*out*/;
            inputs["serviceRegistries"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["taskDefinition"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Service.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Service resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Service(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Service.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Service.__pulumiType;
    }
}
exports.Service = Service;
/** @internal */
Service.__pulumiType = 'aws-native:ecs:Service';
