// @pulumi/aws-native/emr/cluster.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::EMR::Cluster
 *
 * @deprecated Cluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Cluster extends pulumi.CustomResource {
    /**
     * Create a Cluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Cluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Cluster is deprecated: Cluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.instances === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instances'");
            }
            if ((!args || args.jobFlowRole === undefined) && !opts.urn) {
                throw new Error("Missing required property 'jobFlowRole'");
            }
            if ((!args || args.serviceRole === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceRole'");
            }
            inputs["additionalInfo"] = args ? args.additionalInfo : undefined;
            inputs["applications"] = args ? args.applications : undefined;
            inputs["autoScalingRole"] = args ? args.autoScalingRole : undefined;
            inputs["bootstrapActions"] = args ? args.bootstrapActions : undefined;
            inputs["configurations"] = args ? args.configurations : undefined;
            inputs["customAmiId"] = args ? args.customAmiId : undefined;
            inputs["ebsRootVolumeSize"] = args ? args.ebsRootVolumeSize : undefined;
            inputs["instances"] = args ? args.instances : undefined;
            inputs["jobFlowRole"] = args ? args.jobFlowRole : undefined;
            inputs["kerberosAttributes"] = args ? args.kerberosAttributes : undefined;
            inputs["logEncryptionKmsKeyId"] = args ? args.logEncryptionKmsKeyId : undefined;
            inputs["logUri"] = args ? args.logUri : undefined;
            inputs["managedScalingPolicy"] = args ? args.managedScalingPolicy : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["releaseLabel"] = args ? args.releaseLabel : undefined;
            inputs["scaleDownBehavior"] = args ? args.scaleDownBehavior : undefined;
            inputs["securityConfiguration"] = args ? args.securityConfiguration : undefined;
            inputs["serviceRole"] = args ? args.serviceRole : undefined;
            inputs["stepConcurrencyLevel"] = args ? args.stepConcurrencyLevel : undefined;
            inputs["steps"] = args ? args.steps : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["visibleToAllUsers"] = args ? args.visibleToAllUsers : undefined;
            inputs["masterPublicDNS"] = undefined /*out*/;
        }
        else {
            inputs["additionalInfo"] = undefined /*out*/;
            inputs["applications"] = undefined /*out*/;
            inputs["autoScalingRole"] = undefined /*out*/;
            inputs["bootstrapActions"] = undefined /*out*/;
            inputs["configurations"] = undefined /*out*/;
            inputs["customAmiId"] = undefined /*out*/;
            inputs["ebsRootVolumeSize"] = undefined /*out*/;
            inputs["instances"] = undefined /*out*/;
            inputs["jobFlowRole"] = undefined /*out*/;
            inputs["kerberosAttributes"] = undefined /*out*/;
            inputs["logEncryptionKmsKeyId"] = undefined /*out*/;
            inputs["logUri"] = undefined /*out*/;
            inputs["managedScalingPolicy"] = undefined /*out*/;
            inputs["masterPublicDNS"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["releaseLabel"] = undefined /*out*/;
            inputs["scaleDownBehavior"] = undefined /*out*/;
            inputs["securityConfiguration"] = undefined /*out*/;
            inputs["serviceRole"] = undefined /*out*/;
            inputs["stepConcurrencyLevel"] = undefined /*out*/;
            inputs["steps"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["visibleToAllUsers"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Cluster.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Cluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Cluster is deprecated: Cluster is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Cluster(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Cluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Cluster.__pulumiType;
    }
}
exports.Cluster = Cluster;
/** @internal */
Cluster.__pulumiType = 'aws-native:emr:Cluster';
