// @pulumi/aws-native/emr/instanceFleetConfig.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstanceFleetConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::EMR::InstanceFleetConfig
 *
 * @deprecated InstanceFleetConfig is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class InstanceFleetConfig extends pulumi.CustomResource {
    /**
     * Create a InstanceFleetConfig resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated InstanceFleetConfig is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("InstanceFleetConfig is deprecated: InstanceFleetConfig is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterId'");
            }
            if ((!args || args.instanceFleetType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceFleetType'");
            }
            inputs["clusterId"] = args ? args.clusterId : undefined;
            inputs["instanceFleetType"] = args ? args.instanceFleetType : undefined;
            inputs["instanceTypeConfigs"] = args ? args.instanceTypeConfigs : undefined;
            inputs["launchSpecifications"] = args ? args.launchSpecifications : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["targetOnDemandCapacity"] = args ? args.targetOnDemandCapacity : undefined;
            inputs["targetSpotCapacity"] = args ? args.targetSpotCapacity : undefined;
        }
        else {
            inputs["clusterId"] = undefined /*out*/;
            inputs["instanceFleetType"] = undefined /*out*/;
            inputs["instanceTypeConfigs"] = undefined /*out*/;
            inputs["launchSpecifications"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["targetOnDemandCapacity"] = undefined /*out*/;
            inputs["targetSpotCapacity"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(InstanceFleetConfig.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing InstanceFleetConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("InstanceFleetConfig is deprecated: InstanceFleetConfig is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new InstanceFleetConfig(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of InstanceFleetConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === InstanceFleetConfig.__pulumiType;
    }
}
exports.InstanceFleetConfig = InstanceFleetConfig;
/** @internal */
InstanceFleetConfig.__pulumiType = 'aws-native:emr:InstanceFleetConfig';
