// @pulumi/aws-native/emr/studio.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Studio = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::EMR::Studio
 */
class Studio extends pulumi.CustomResource {
    /**
     * Create a Studio resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.authMode === undefined) && !opts.urn) {
                throw new Error("Missing required property 'authMode'");
            }
            if ((!args || args.defaultS3Location === undefined) && !opts.urn) {
                throw new Error("Missing required property 'defaultS3Location'");
            }
            if ((!args || args.engineSecurityGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'engineSecurityGroupId'");
            }
            if ((!args || args.serviceRole === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceRole'");
            }
            if ((!args || args.subnetIds === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subnetIds'");
            }
            if ((!args || args.vpcId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vpcId'");
            }
            if ((!args || args.workspaceSecurityGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceSecurityGroupId'");
            }
            inputs["authMode"] = args ? args.authMode : undefined;
            inputs["defaultS3Location"] = args ? args.defaultS3Location : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["engineSecurityGroupId"] = args ? args.engineSecurityGroupId : undefined;
            inputs["idpAuthUrl"] = args ? args.idpAuthUrl : undefined;
            inputs["idpRelayStateParameterName"] = args ? args.idpRelayStateParameterName : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["serviceRole"] = args ? args.serviceRole : undefined;
            inputs["subnetIds"] = args ? args.subnetIds : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["userRole"] = args ? args.userRole : undefined;
            inputs["vpcId"] = args ? args.vpcId : undefined;
            inputs["workspaceSecurityGroupId"] = args ? args.workspaceSecurityGroupId : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["studioId"] = undefined /*out*/;
            inputs["url"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["authMode"] = undefined /*out*/;
            inputs["defaultS3Location"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["engineSecurityGroupId"] = undefined /*out*/;
            inputs["idpAuthUrl"] = undefined /*out*/;
            inputs["idpRelayStateParameterName"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["serviceRole"] = undefined /*out*/;
            inputs["studioId"] = undefined /*out*/;
            inputs["subnetIds"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["url"] = undefined /*out*/;
            inputs["userRole"] = undefined /*out*/;
            inputs["vpcId"] = undefined /*out*/;
            inputs["workspaceSecurityGroupId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Studio.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Studio resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Studio(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Studio.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Studio.__pulumiType;
    }
}
exports.Studio = Studio;
/** @internal */
Studio.__pulumiType = 'aws-native:emr:Studio';
