// @pulumi/aws-native/quicksight/dataSource.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataSource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of the AWS::QuickSight::DataSource Resource Type.
 */
class DataSource extends pulumi.CustomResource {
    /**
     * Create a DataSource resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["alternateDataSourceParameters"] = args ? args.alternateDataSourceParameters : undefined;
            inputs["awsAccountId"] = args ? args.awsAccountId : undefined;
            inputs["credentials"] = args ? args.credentials : undefined;
            inputs["dataSourceId"] = args ? args.dataSourceId : undefined;
            inputs["dataSourceParameters"] = args ? args.dataSourceParameters : undefined;
            inputs["errorInfo"] = args ? args.errorInfo : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["permissions"] = args ? args.permissions : undefined;
            inputs["sslProperties"] = args ? args.sslProperties : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["type"] = args ? args.type : undefined;
            inputs["vpcConnectionProperties"] = args ? args.vpcConnectionProperties : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["createdTime"] = undefined /*out*/;
            inputs["lastUpdatedTime"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
        }
        else {
            inputs["alternateDataSourceParameters"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
            inputs["awsAccountId"] = undefined /*out*/;
            inputs["createdTime"] = undefined /*out*/;
            inputs["credentials"] = undefined /*out*/;
            inputs["dataSourceId"] = undefined /*out*/;
            inputs["dataSourceParameters"] = undefined /*out*/;
            inputs["errorInfo"] = undefined /*out*/;
            inputs["lastUpdatedTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["permissions"] = undefined /*out*/;
            inputs["sslProperties"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["vpcConnectionProperties"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(DataSource.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DataSource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DataSource(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DataSource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataSource.__pulumiType;
    }
}
exports.DataSource = DataSource;
/** @internal */
DataSource.__pulumiType = 'aws-native:quicksight:DataSource';
