// @pulumi/aws-native/ssmcontacts/contactChannel.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContactChannel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::SSMContacts::ContactChannel
 */
class ContactChannel extends pulumi.CustomResource {
    /**
     * Create a ContactChannel resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["channelAddress"] = args ? args.channelAddress : undefined;
            inputs["channelName"] = args ? args.channelName : undefined;
            inputs["channelType"] = args ? args.channelType : undefined;
            inputs["contactId"] = args ? args.contactId : undefined;
            inputs["deferActivation"] = args ? args.deferActivation : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["channelAddress"] = undefined /*out*/;
            inputs["channelName"] = undefined /*out*/;
            inputs["channelType"] = undefined /*out*/;
            inputs["contactId"] = undefined /*out*/;
            inputs["deferActivation"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ContactChannel.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ContactChannel resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ContactChannel(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ContactChannel.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ContactChannel.__pulumiType;
    }
}
exports.ContactChannel = ContactChannel;
/** @internal */
ContactChannel.__pulumiType = 'aws-native:ssmcontacts:ContactChannel';
