// @pulumi/aws-native/panorama/applicationInstance.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationInstance = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Schema for ApplicationInstance CloudFormation Resource
 */
class ApplicationInstance extends pulumi.CustomResource {
    /**
     * Create a ApplicationInstance resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.defaultRuntimeContextDevice === undefined) && !opts.urn) {
                throw new Error("Missing required property 'defaultRuntimeContextDevice'");
            }
            if ((!args || args.manifestPayload === undefined) && !opts.urn) {
                throw new Error("Missing required property 'manifestPayload'");
            }
            inputs["applicationInstanceIdToReplace"] = args ? args.applicationInstanceIdToReplace : undefined;
            inputs["defaultRuntimeContextDevice"] = args ? args.defaultRuntimeContextDevice : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["deviceId"] = args ? args.deviceId : undefined;
            inputs["manifestOverridesPayload"] = args ? args.manifestOverridesPayload : undefined;
            inputs["manifestPayload"] = args ? args.manifestPayload : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["runtimeRoleArn"] = args ? args.runtimeRoleArn : undefined;
            inputs["statusFilter"] = args ? args.statusFilter : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["applicationInstanceId"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
            inputs["createdTime"] = undefined /*out*/;
            inputs["defaultRuntimeContextDeviceName"] = undefined /*out*/;
            inputs["healthStatus"] = undefined /*out*/;
            inputs["lastUpdatedTime"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["statusDescription"] = undefined /*out*/;
        }
        else {
            inputs["applicationInstanceId"] = undefined /*out*/;
            inputs["applicationInstanceIdToReplace"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
            inputs["createdTime"] = undefined /*out*/;
            inputs["defaultRuntimeContextDevice"] = undefined /*out*/;
            inputs["defaultRuntimeContextDeviceName"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["deviceId"] = undefined /*out*/;
            inputs["healthStatus"] = undefined /*out*/;
            inputs["lastUpdatedTime"] = undefined /*out*/;
            inputs["manifestOverridesPayload"] = undefined /*out*/;
            inputs["manifestPayload"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["runtimeRoleArn"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["statusDescription"] = undefined /*out*/;
            inputs["statusFilter"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ApplicationInstance.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ApplicationInstance resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ApplicationInstance(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ApplicationInstance.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ApplicationInstance.__pulumiType;
    }
}
exports.ApplicationInstance = ApplicationInstance;
/** @internal */
ApplicationInstance.__pulumiType = 'aws-native:panorama:ApplicationInstance';
