// @pulumi/aws-native/panorama/packageVersion.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PackageVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Schema for PackageVersion Resource Type
 */
class PackageVersion extends pulumi.CustomResource {
    /**
     * Create a PackageVersion resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.packageId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'packageId'");
            }
            if ((!args || args.packageVersion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'packageVersion'");
            }
            if ((!args || args.patchVersion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'patchVersion'");
            }
            inputs["markLatest"] = args ? args.markLatest : undefined;
            inputs["ownerAccount"] = args ? args.ownerAccount : undefined;
            inputs["packageId"] = args ? args.packageId : undefined;
            inputs["packageVersion"] = args ? args.packageVersion : undefined;
            inputs["patchVersion"] = args ? args.patchVersion : undefined;
            inputs["updatedLatestPatchVersion"] = args ? args.updatedLatestPatchVersion : undefined;
            inputs["isLatestPatch"] = undefined /*out*/;
            inputs["packageArn"] = undefined /*out*/;
            inputs["packageName"] = undefined /*out*/;
            inputs["registeredTime"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["statusDescription"] = undefined /*out*/;
        }
        else {
            inputs["isLatestPatch"] = undefined /*out*/;
            inputs["markLatest"] = undefined /*out*/;
            inputs["ownerAccount"] = undefined /*out*/;
            inputs["packageArn"] = undefined /*out*/;
            inputs["packageId"] = undefined /*out*/;
            inputs["packageName"] = undefined /*out*/;
            inputs["packageVersion"] = undefined /*out*/;
            inputs["patchVersion"] = undefined /*out*/;
            inputs["registeredTime"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["statusDescription"] = undefined /*out*/;
            inputs["updatedLatestPatchVersion"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(PackageVersion.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing PackageVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PackageVersion(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PackageVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PackageVersion.__pulumiType;
    }
}
exports.PackageVersion = PackageVersion;
/** @internal */
PackageVersion.__pulumiType = 'aws-native:panorama:PackageVersion';
