// @pulumi/aws-native/mediaconnect/flowOutput.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowOutput = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::MediaConnect::FlowOutput
 */
class FlowOutput extends pulumi.CustomResource {
    /**
     * Create a FlowOutput resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.flowArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'flowArn'");
            }
            if ((!args || args.protocol === undefined) && !opts.urn) {
                throw new Error("Missing required property 'protocol'");
            }
            inputs["cidrAllowList"] = args ? args.cidrAllowList : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["destination"] = args ? args.destination : undefined;
            inputs["encryption"] = args ? args.encryption : undefined;
            inputs["flowArn"] = args ? args.flowArn : undefined;
            inputs["maxLatency"] = args ? args.maxLatency : undefined;
            inputs["minLatency"] = args ? args.minLatency : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["port"] = args ? args.port : undefined;
            inputs["protocol"] = args ? args.protocol : undefined;
            inputs["remoteId"] = args ? args.remoteId : undefined;
            inputs["smoothingLatency"] = args ? args.smoothingLatency : undefined;
            inputs["streamId"] = args ? args.streamId : undefined;
            inputs["vpcInterfaceAttachment"] = args ? args.vpcInterfaceAttachment : undefined;
            inputs["outputArn"] = undefined /*out*/;
        }
        else {
            inputs["cidrAllowList"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["destination"] = undefined /*out*/;
            inputs["encryption"] = undefined /*out*/;
            inputs["flowArn"] = undefined /*out*/;
            inputs["maxLatency"] = undefined /*out*/;
            inputs["minLatency"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["outputArn"] = undefined /*out*/;
            inputs["port"] = undefined /*out*/;
            inputs["protocol"] = undefined /*out*/;
            inputs["remoteId"] = undefined /*out*/;
            inputs["smoothingLatency"] = undefined /*out*/;
            inputs["streamId"] = undefined /*out*/;
            inputs["vpcInterfaceAttachment"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(FlowOutput.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing FlowOutput resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FlowOutput(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FlowOutput.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FlowOutput.__pulumiType;
    }
}
exports.FlowOutput = FlowOutput;
/** @internal */
FlowOutput.__pulumiType = 'aws-native:mediaconnect:FlowOutput';
