// @pulumi/aws-native/transfer/server.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Server = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Transfer::Server
 *
 * @deprecated Server is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Server extends pulumi.CustomResource {
    /**
     * Create a Server resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Server is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Server is deprecated: Server is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["certificate"] = args ? args.certificate : undefined;
            inputs["domain"] = args ? args.domain : undefined;
            inputs["endpointDetails"] = args ? args.endpointDetails : undefined;
            inputs["endpointType"] = args ? args.endpointType : undefined;
            inputs["identityProviderDetails"] = args ? args.identityProviderDetails : undefined;
            inputs["identityProviderType"] = args ? args.identityProviderType : undefined;
            inputs["loggingRole"] = args ? args.loggingRole : undefined;
            inputs["protocolDetails"] = args ? args.protocolDetails : undefined;
            inputs["protocols"] = args ? args.protocols : undefined;
            inputs["securityPolicyName"] = args ? args.securityPolicyName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["workflowDetails"] = args ? args.workflowDetails : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["serverId"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["certificate"] = undefined /*out*/;
            inputs["domain"] = undefined /*out*/;
            inputs["endpointDetails"] = undefined /*out*/;
            inputs["endpointType"] = undefined /*out*/;
            inputs["identityProviderDetails"] = undefined /*out*/;
            inputs["identityProviderType"] = undefined /*out*/;
            inputs["loggingRole"] = undefined /*out*/;
            inputs["protocolDetails"] = undefined /*out*/;
            inputs["protocols"] = undefined /*out*/;
            inputs["securityPolicyName"] = undefined /*out*/;
            inputs["serverId"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["workflowDetails"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Server.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Server resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Server is deprecated: Server is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Server(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Server.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Server.__pulumiType;
    }
}
exports.Server = Server;
/** @internal */
Server.__pulumiType = 'aws-native:transfer:Server';
