// @pulumi/aws-native/gamelift/matchmakingConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MatchmakingConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::GameLift::MatchmakingConfiguration
 *
 * @deprecated MatchmakingConfiguration is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class MatchmakingConfiguration extends pulumi.CustomResource {
    /**
     * Create a MatchmakingConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated MatchmakingConfiguration is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("MatchmakingConfiguration is deprecated: MatchmakingConfiguration is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.acceptanceRequired === undefined) && !opts.urn) {
                throw new Error("Missing required property 'acceptanceRequired'");
            }
            if ((!args || args.requestTimeoutSeconds === undefined) && !opts.urn) {
                throw new Error("Missing required property 'requestTimeoutSeconds'");
            }
            if ((!args || args.ruleSetName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ruleSetName'");
            }
            inputs["acceptanceRequired"] = args ? args.acceptanceRequired : undefined;
            inputs["acceptanceTimeoutSeconds"] = args ? args.acceptanceTimeoutSeconds : undefined;
            inputs["additionalPlayerCount"] = args ? args.additionalPlayerCount : undefined;
            inputs["backfillMode"] = args ? args.backfillMode : undefined;
            inputs["customEventData"] = args ? args.customEventData : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["flexMatchMode"] = args ? args.flexMatchMode : undefined;
            inputs["gameProperties"] = args ? args.gameProperties : undefined;
            inputs["gameSessionData"] = args ? args.gameSessionData : undefined;
            inputs["gameSessionQueueArns"] = args ? args.gameSessionQueueArns : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["notificationTarget"] = args ? args.notificationTarget : undefined;
            inputs["requestTimeoutSeconds"] = args ? args.requestTimeoutSeconds : undefined;
            inputs["ruleSetName"] = args ? args.ruleSetName : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["acceptanceRequired"] = undefined /*out*/;
            inputs["acceptanceTimeoutSeconds"] = undefined /*out*/;
            inputs["additionalPlayerCount"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
            inputs["backfillMode"] = undefined /*out*/;
            inputs["customEventData"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["flexMatchMode"] = undefined /*out*/;
            inputs["gameProperties"] = undefined /*out*/;
            inputs["gameSessionData"] = undefined /*out*/;
            inputs["gameSessionQueueArns"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["notificationTarget"] = undefined /*out*/;
            inputs["requestTimeoutSeconds"] = undefined /*out*/;
            inputs["ruleSetName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(MatchmakingConfiguration.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing MatchmakingConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("MatchmakingConfiguration is deprecated: MatchmakingConfiguration is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new MatchmakingConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MatchmakingConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MatchmakingConfiguration.__pulumiType;
    }
}
exports.MatchmakingConfiguration = MatchmakingConfiguration;
/** @internal */
MatchmakingConfiguration.__pulumiType = 'aws-native:gamelift:MatchmakingConfiguration';
