// @pulumi/aws-native/pinpoint/campaign.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Campaign = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Pinpoint::Campaign
 *
 * @deprecated Campaign is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Campaign extends pulumi.CustomResource {
    /**
     * Create a Campaign resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Campaign is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Campaign is deprecated: Campaign is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.applicationId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'applicationId'");
            }
            if ((!args || args.messageConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'messageConfiguration'");
            }
            if ((!args || args.schedule === undefined) && !opts.urn) {
                throw new Error("Missing required property 'schedule'");
            }
            if ((!args || args.segmentId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'segmentId'");
            }
            inputs["additionalTreatments"] = args ? args.additionalTreatments : undefined;
            inputs["applicationId"] = args ? args.applicationId : undefined;
            inputs["campaignHook"] = args ? args.campaignHook : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["holdoutPercent"] = args ? args.holdoutPercent : undefined;
            inputs["isPaused"] = args ? args.isPaused : undefined;
            inputs["limits"] = args ? args.limits : undefined;
            inputs["messageConfiguration"] = args ? args.messageConfiguration : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["priority"] = args ? args.priority : undefined;
            inputs["schedule"] = args ? args.schedule : undefined;
            inputs["segmentId"] = args ? args.segmentId : undefined;
            inputs["segmentVersion"] = args ? args.segmentVersion : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["treatmentDescription"] = args ? args.treatmentDescription : undefined;
            inputs["treatmentName"] = args ? args.treatmentName : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["campaignId"] = undefined /*out*/;
        }
        else {
            inputs["additionalTreatments"] = undefined /*out*/;
            inputs["applicationId"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
            inputs["campaignHook"] = undefined /*out*/;
            inputs["campaignId"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["holdoutPercent"] = undefined /*out*/;
            inputs["isPaused"] = undefined /*out*/;
            inputs["limits"] = undefined /*out*/;
            inputs["messageConfiguration"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["priority"] = undefined /*out*/;
            inputs["schedule"] = undefined /*out*/;
            inputs["segmentId"] = undefined /*out*/;
            inputs["segmentVersion"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["treatmentDescription"] = undefined /*out*/;
            inputs["treatmentName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Campaign.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Campaign resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Campaign is deprecated: Campaign is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Campaign(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Campaign.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Campaign.__pulumiType;
    }
}
exports.Campaign = Campaign;
/** @internal */
Campaign.__pulumiType = 'aws-native:pinpoint:Campaign';
