// @pulumi/aws-native/lex/botAlias.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BotAlias = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A Bot Alias enables you to change the version of a bot without updating applications that use the bot
 */
class BotAlias extends pulumi.CustomResource {
    /**
     * Create a BotAlias resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.botId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'botId'");
            }
            inputs["botAliasLocaleSettings"] = args ? args.botAliasLocaleSettings : undefined;
            inputs["botAliasName"] = args ? args.botAliasName : undefined;
            inputs["botAliasTags"] = args ? args.botAliasTags : undefined;
            inputs["botId"] = args ? args.botId : undefined;
            inputs["botVersion"] = args ? args.botVersion : undefined;
            inputs["conversationLogSettings"] = args ? args.conversationLogSettings : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["sentimentAnalysisSettings"] = args ? args.sentimentAnalysisSettings : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["botAliasId"] = undefined /*out*/;
            inputs["botAliasStatus"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["botAliasId"] = undefined /*out*/;
            inputs["botAliasLocaleSettings"] = undefined /*out*/;
            inputs["botAliasName"] = undefined /*out*/;
            inputs["botAliasStatus"] = undefined /*out*/;
            inputs["botAliasTags"] = undefined /*out*/;
            inputs["botId"] = undefined /*out*/;
            inputs["botVersion"] = undefined /*out*/;
            inputs["conversationLogSettings"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["sentimentAnalysisSettings"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(BotAlias.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing BotAlias resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BotAlias(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BotAlias.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BotAlias.__pulumiType;
    }
}
exports.BotAlias = BotAlias;
/** @internal */
BotAlias.__pulumiType = 'aws-native:lex:BotAlias';
