// @pulumi/aws-native/ssmincidents/responsePlan.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponsePlan = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource type definition for AWS::SSMIncidents::ResponsePlan
 */
class ResponsePlan extends pulumi.CustomResource {
    /**
     * Create a ResponsePlan resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.incidentTemplate === undefined) && !opts.urn) {
                throw new Error("Missing required property 'incidentTemplate'");
            }
            inputs["actions"] = args ? args.actions : undefined;
            inputs["chatChannel"] = args ? args.chatChannel : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["engagements"] = args ? args.engagements : undefined;
            inputs["incidentTemplate"] = args ? args.incidentTemplate : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["actions"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
            inputs["chatChannel"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["engagements"] = undefined /*out*/;
            inputs["incidentTemplate"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ResponsePlan.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ResponsePlan resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ResponsePlan(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ResponsePlan.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResponsePlan.__pulumiType;
    }
}
exports.ResponsePlan = ResponsePlan;
/** @internal */
ResponsePlan.__pulumiType = 'aws-native:ssmincidents:ResponsePlan';
