// @pulumi/aws-native/codepipeline/customActionType.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomActionType = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::CodePipeline::CustomActionType
 *
 * @deprecated CustomActionType is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class CustomActionType extends pulumi.CustomResource {
    /**
     * Create a CustomActionType resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated CustomActionType is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("CustomActionType is deprecated: CustomActionType is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.category === undefined) && !opts.urn) {
                throw new Error("Missing required property 'category'");
            }
            if ((!args || args.inputArtifactDetails === undefined) && !opts.urn) {
                throw new Error("Missing required property 'inputArtifactDetails'");
            }
            if ((!args || args.outputArtifactDetails === undefined) && !opts.urn) {
                throw new Error("Missing required property 'outputArtifactDetails'");
            }
            if ((!args || args.provider === undefined) && !opts.urn) {
                throw new Error("Missing required property 'provider'");
            }
            if ((!args || args.version === undefined) && !opts.urn) {
                throw new Error("Missing required property 'version'");
            }
            inputs["category"] = args ? args.category : undefined;
            inputs["configurationProperties"] = args ? args.configurationProperties : undefined;
            inputs["inputArtifactDetails"] = args ? args.inputArtifactDetails : undefined;
            inputs["outputArtifactDetails"] = args ? args.outputArtifactDetails : undefined;
            inputs["provider"] = args ? args.provider : undefined;
            inputs["settings"] = args ? args.settings : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["version"] = args ? args.version : undefined;
        }
        else {
            inputs["category"] = undefined /*out*/;
            inputs["configurationProperties"] = undefined /*out*/;
            inputs["inputArtifactDetails"] = undefined /*out*/;
            inputs["outputArtifactDetails"] = undefined /*out*/;
            inputs["provider"] = undefined /*out*/;
            inputs["settings"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(CustomActionType.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing CustomActionType resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("CustomActionType is deprecated: CustomActionType is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new CustomActionType(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CustomActionType.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomActionType.__pulumiType;
    }
}
exports.CustomActionType = CustomActionType;
/** @internal */
CustomActionType.__pulumiType = 'aws-native:codepipeline:CustomActionType';
