// @pulumi/aws-native/medialive/channel.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Channel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::MediaLive::Channel
 *
 * @deprecated Channel is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Channel extends pulumi.CustomResource {
    /**
     * Create a Channel resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Channel is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Channel is deprecated: Channel is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["cdiInputSpecification"] = args ? args.cdiInputSpecification : undefined;
            inputs["channelClass"] = args ? args.channelClass : undefined;
            inputs["destinations"] = args ? args.destinations : undefined;
            inputs["encoderSettings"] = args ? args.encoderSettings : undefined;
            inputs["inputAttachments"] = args ? args.inputAttachments : undefined;
            inputs["inputSpecification"] = args ? args.inputSpecification : undefined;
            inputs["logLevel"] = args ? args.logLevel : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["roleArn"] = args ? args.roleArn : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["vpc"] = args ? args.vpc : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["inputs"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["cdiInputSpecification"] = undefined /*out*/;
            inputs["channelClass"] = undefined /*out*/;
            inputs["destinations"] = undefined /*out*/;
            inputs["encoderSettings"] = undefined /*out*/;
            inputs["inputAttachments"] = undefined /*out*/;
            inputs["inputSpecification"] = undefined /*out*/;
            inputs["inputs"] = undefined /*out*/;
            inputs["logLevel"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["roleArn"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["vpc"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Channel.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Channel resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Channel is deprecated: Channel is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Channel(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Channel.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Channel.__pulumiType;
    }
}
exports.Channel = Channel;
/** @internal */
Channel.__pulumiType = 'aws-native:medialive:Channel';
