// @pulumi/aws-native/sso/permissionSet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PermissionSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for SSO PermissionSet
 */
class PermissionSet extends pulumi.CustomResource {
    /**
     * Create a PermissionSet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.instanceArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceArn'");
            }
            inputs["description"] = args ? args.description : undefined;
            inputs["inlinePolicy"] = args ? args.inlinePolicy : undefined;
            inputs["instanceArn"] = args ? args.instanceArn : undefined;
            inputs["managedPolicies"] = args ? args.managedPolicies : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["relayStateType"] = args ? args.relayStateType : undefined;
            inputs["sessionDuration"] = args ? args.sessionDuration : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["permissionSetArn"] = undefined /*out*/;
        }
        else {
            inputs["description"] = undefined /*out*/;
            inputs["inlinePolicy"] = undefined /*out*/;
            inputs["instanceArn"] = undefined /*out*/;
            inputs["managedPolicies"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["permissionSetArn"] = undefined /*out*/;
            inputs["relayStateType"] = undefined /*out*/;
            inputs["sessionDuration"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(PermissionSet.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing PermissionSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PermissionSet(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PermissionSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PermissionSet.__pulumiType;
    }
}
exports.PermissionSet = PermissionSet;
/** @internal */
PermissionSet.__pulumiType = 'aws-native:sso:PermissionSet';
