// @pulumi/aws-native/healthlake/fhirdatastore.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FHIRDatastore = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * HealthLake FHIR Datastore
 */
class FHIRDatastore extends pulumi.CustomResource {
    /**
     * Create a FHIRDatastore resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.datastoreTypeVersion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'datastoreTypeVersion'");
            }
            inputs["datastoreName"] = args ? args.datastoreName : undefined;
            inputs["datastoreTypeVersion"] = args ? args.datastoreTypeVersion : undefined;
            inputs["preloadDataConfig"] = args ? args.preloadDataConfig : undefined;
            inputs["sseConfiguration"] = args ? args.sseConfiguration : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["createdAt"] = undefined /*out*/;
            inputs["datastoreArn"] = undefined /*out*/;
            inputs["datastoreEndpoint"] = undefined /*out*/;
            inputs["datastoreId"] = undefined /*out*/;
            inputs["datastoreStatus"] = undefined /*out*/;
        }
        else {
            inputs["createdAt"] = undefined /*out*/;
            inputs["datastoreArn"] = undefined /*out*/;
            inputs["datastoreEndpoint"] = undefined /*out*/;
            inputs["datastoreId"] = undefined /*out*/;
            inputs["datastoreName"] = undefined /*out*/;
            inputs["datastoreStatus"] = undefined /*out*/;
            inputs["datastoreTypeVersion"] = undefined /*out*/;
            inputs["preloadDataConfig"] = undefined /*out*/;
            inputs["sseConfiguration"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(FHIRDatastore.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing FHIRDatastore resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FHIRDatastore(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FHIRDatastore.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FHIRDatastore.__pulumiType;
    }
}
exports.FHIRDatastore = FHIRDatastore;
/** @internal */
FHIRDatastore.__pulumiType = 'aws-native:healthlake:FHIRDatastore';
