// @pulumi/aws-native/route53recoverycontrol/safetyRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SafetyRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS Route53 Recovery Control basic constructs and validation rules.
 */
class SafetyRule extends pulumi.CustomResource {
    /**
     * Create a SafetyRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["assertionRule"] = args ? args.assertionRule : undefined;
            inputs["controlPanelArn"] = args ? args.controlPanelArn : undefined;
            inputs["gatingRule"] = args ? args.gatingRule : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["ruleConfig"] = args ? args.ruleConfig : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["safetyRuleArn"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
        }
        else {
            inputs["assertionRule"] = undefined /*out*/;
            inputs["controlPanelArn"] = undefined /*out*/;
            inputs["gatingRule"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["ruleConfig"] = undefined /*out*/;
            inputs["safetyRuleArn"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(SafetyRule.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SafetyRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SafetyRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SafetyRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SafetyRule.__pulumiType;
    }
}
exports.SafetyRule = SafetyRule;
/** @internal */
SafetyRule.__pulumiType = 'aws-native:route53recoverycontrol:SafetyRule';
