// @pulumi/aws-native/athena/preparedStatement.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PreparedStatement = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::Athena::PreparedStatement
 */
class PreparedStatement extends pulumi.CustomResource {
    /**
     * Create a PreparedStatement resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.queryStatement === undefined) && !opts.urn) {
                throw new Error("Missing required property 'queryStatement'");
            }
            if ((!args || args.statementName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'statementName'");
            }
            if ((!args || args.workGroup === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workGroup'");
            }
            inputs["description"] = args ? args.description : undefined;
            inputs["queryStatement"] = args ? args.queryStatement : undefined;
            inputs["statementName"] = args ? args.statementName : undefined;
            inputs["workGroup"] = args ? args.workGroup : undefined;
        }
        else {
            inputs["description"] = undefined /*out*/;
            inputs["queryStatement"] = undefined /*out*/;
            inputs["statementName"] = undefined /*out*/;
            inputs["workGroup"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(PreparedStatement.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing PreparedStatement resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PreparedStatement(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PreparedStatement.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PreparedStatement.__pulumiType;
    }
}
exports.PreparedStatement = PreparedStatement;
/** @internal */
PreparedStatement.__pulumiType = 'aws-native:athena:PreparedStatement';
