// @pulumi/aws-native/lookoutmetrics/alert.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Alert = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::LookoutMetrics::Alert
 */
class Alert extends pulumi.CustomResource {
    /**
     * Create a Alert resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.action === undefined) && !opts.urn) {
                throw new Error("Missing required property 'action'");
            }
            if ((!args || args.alertSensitivityThreshold === undefined) && !opts.urn) {
                throw new Error("Missing required property 'alertSensitivityThreshold'");
            }
            if ((!args || args.anomalyDetectorArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'anomalyDetectorArn'");
            }
            inputs["action"] = args ? args.action : undefined;
            inputs["alertDescription"] = args ? args.alertDescription : undefined;
            inputs["alertName"] = args ? args.alertName : undefined;
            inputs["alertSensitivityThreshold"] = args ? args.alertSensitivityThreshold : undefined;
            inputs["anomalyDetectorArn"] = args ? args.anomalyDetectorArn : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["action"] = undefined /*out*/;
            inputs["alertDescription"] = undefined /*out*/;
            inputs["alertName"] = undefined /*out*/;
            inputs["alertSensitivityThreshold"] = undefined /*out*/;
            inputs["anomalyDetectorArn"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Alert.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Alert resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Alert(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Alert.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Alert.__pulumiType;
    }
}
exports.Alert = Alert;
/** @internal */
Alert.__pulumiType = 'aws-native:lookoutmetrics:Alert';
