// @pulumi/aws-native/lookoutmetrics/anomalyDetector.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnomalyDetector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An Amazon Lookout for Metrics Detector
 */
class AnomalyDetector extends pulumi.CustomResource {
    /**
     * Create a AnomalyDetector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.anomalyDetectorConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'anomalyDetectorConfig'");
            }
            if ((!args || args.metricSetList === undefined) && !opts.urn) {
                throw new Error("Missing required property 'metricSetList'");
            }
            inputs["anomalyDetectorConfig"] = args ? args.anomalyDetectorConfig : undefined;
            inputs["anomalyDetectorDescription"] = args ? args.anomalyDetectorDescription : undefined;
            inputs["anomalyDetectorName"] = args ? args.anomalyDetectorName : undefined;
            inputs["kmsKeyArn"] = args ? args.kmsKeyArn : undefined;
            inputs["metricSetList"] = args ? args.metricSetList : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["anomalyDetectorConfig"] = undefined /*out*/;
            inputs["anomalyDetectorDescription"] = undefined /*out*/;
            inputs["anomalyDetectorName"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
            inputs["kmsKeyArn"] = undefined /*out*/;
            inputs["metricSetList"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(AnomalyDetector.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing AnomalyDetector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AnomalyDetector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AnomalyDetector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AnomalyDetector.__pulumiType;
    }
}
exports.AnomalyDetector = AnomalyDetector;
/** @internal */
AnomalyDetector.__pulumiType = 'aws-native:lookoutmetrics:AnomalyDetector';
