// @pulumi/aws-native/cloudwatch/alarm.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Alarm = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::CloudWatch::Alarm
 *
 * @deprecated Alarm is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Alarm extends pulumi.CustomResource {
    /**
     * Create a Alarm resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Alarm is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Alarm is deprecated: Alarm is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.comparisonOperator === undefined) && !opts.urn) {
                throw new Error("Missing required property 'comparisonOperator'");
            }
            if ((!args || args.evaluationPeriods === undefined) && !opts.urn) {
                throw new Error("Missing required property 'evaluationPeriods'");
            }
            inputs["actionsEnabled"] = args ? args.actionsEnabled : undefined;
            inputs["alarmActions"] = args ? args.alarmActions : undefined;
            inputs["alarmDescription"] = args ? args.alarmDescription : undefined;
            inputs["alarmName"] = args ? args.alarmName : undefined;
            inputs["comparisonOperator"] = args ? args.comparisonOperator : undefined;
            inputs["datapointsToAlarm"] = args ? args.datapointsToAlarm : undefined;
            inputs["dimensions"] = args ? args.dimensions : undefined;
            inputs["evaluateLowSampleCountPercentile"] = args ? args.evaluateLowSampleCountPercentile : undefined;
            inputs["evaluationPeriods"] = args ? args.evaluationPeriods : undefined;
            inputs["extendedStatistic"] = args ? args.extendedStatistic : undefined;
            inputs["insufficientDataActions"] = args ? args.insufficientDataActions : undefined;
            inputs["metricName"] = args ? args.metricName : undefined;
            inputs["metrics"] = args ? args.metrics : undefined;
            inputs["namespace"] = args ? args.namespace : undefined;
            inputs["oKActions"] = args ? args.oKActions : undefined;
            inputs["period"] = args ? args.period : undefined;
            inputs["statistic"] = args ? args.statistic : undefined;
            inputs["threshold"] = args ? args.threshold : undefined;
            inputs["thresholdMetricId"] = args ? args.thresholdMetricId : undefined;
            inputs["treatMissingData"] = args ? args.treatMissingData : undefined;
            inputs["unit"] = args ? args.unit : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["actionsEnabled"] = undefined /*out*/;
            inputs["alarmActions"] = undefined /*out*/;
            inputs["alarmDescription"] = undefined /*out*/;
            inputs["alarmName"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
            inputs["comparisonOperator"] = undefined /*out*/;
            inputs["datapointsToAlarm"] = undefined /*out*/;
            inputs["dimensions"] = undefined /*out*/;
            inputs["evaluateLowSampleCountPercentile"] = undefined /*out*/;
            inputs["evaluationPeriods"] = undefined /*out*/;
            inputs["extendedStatistic"] = undefined /*out*/;
            inputs["insufficientDataActions"] = undefined /*out*/;
            inputs["metricName"] = undefined /*out*/;
            inputs["metrics"] = undefined /*out*/;
            inputs["namespace"] = undefined /*out*/;
            inputs["oKActions"] = undefined /*out*/;
            inputs["period"] = undefined /*out*/;
            inputs["statistic"] = undefined /*out*/;
            inputs["threshold"] = undefined /*out*/;
            inputs["thresholdMetricId"] = undefined /*out*/;
            inputs["treatMissingData"] = undefined /*out*/;
            inputs["unit"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Alarm.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Alarm resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Alarm is deprecated: Alarm is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Alarm(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Alarm.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Alarm.__pulumiType;
    }
}
exports.Alarm = Alarm;
/** @internal */
Alarm.__pulumiType = 'aws-native:cloudwatch:Alarm';
