// @pulumi/aws-native/devicefarm/vpceconfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VPCEConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * AWS::DeviceFarm::VPCEConfiguration creates a new Device Farm VPCE Configuration
 */
class VPCEConfiguration extends pulumi.CustomResource {
    /**
     * Create a VPCEConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.serviceDnsName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceDnsName'");
            }
            if ((!args || args.vpceConfigurationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vpceConfigurationName'");
            }
            if ((!args || args.vpceServiceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vpceServiceName'");
            }
            inputs["serviceDnsName"] = args ? args.serviceDnsName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["vpceConfigurationDescription"] = args ? args.vpceConfigurationDescription : undefined;
            inputs["vpceConfigurationName"] = args ? args.vpceConfigurationName : undefined;
            inputs["vpceServiceName"] = args ? args.vpceServiceName : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["serviceDnsName"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["vpceConfigurationDescription"] = undefined /*out*/;
            inputs["vpceConfigurationName"] = undefined /*out*/;
            inputs["vpceServiceName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(VPCEConfiguration.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing VPCEConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VPCEConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VPCEConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VPCEConfiguration.__pulumiType;
    }
}
exports.VPCEConfiguration = VPCEConfiguration;
/** @internal */
VPCEConfiguration.__pulumiType = 'aws-native:devicefarm:VPCEConfiguration';
