// @pulumi/aws-native/servicecatalog/cloudFormationProduct.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudFormationProduct = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::ServiceCatalog::CloudFormationProduct
 *
 * @deprecated CloudFormationProduct is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class CloudFormationProduct extends pulumi.CustomResource {
    /**
     * Create a CloudFormationProduct resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated CloudFormationProduct is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("CloudFormationProduct is deprecated: CloudFormationProduct is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.owner === undefined) && !opts.urn) {
                throw new Error("Missing required property 'owner'");
            }
            if ((!args || args.provisioningArtifactParameters === undefined) && !opts.urn) {
                throw new Error("Missing required property 'provisioningArtifactParameters'");
            }
            inputs["acceptLanguage"] = args ? args.acceptLanguage : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["distributor"] = args ? args.distributor : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["owner"] = args ? args.owner : undefined;
            inputs["provisioningArtifactParameters"] = args ? args.provisioningArtifactParameters : undefined;
            inputs["replaceProvisioningArtifacts"] = args ? args.replaceProvisioningArtifacts : undefined;
            inputs["supportDescription"] = args ? args.supportDescription : undefined;
            inputs["supportEmail"] = args ? args.supportEmail : undefined;
            inputs["supportUrl"] = args ? args.supportUrl : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["productName"] = undefined /*out*/;
            inputs["provisioningArtifactIds"] = undefined /*out*/;
            inputs["provisioningArtifactNames"] = undefined /*out*/;
        }
        else {
            inputs["acceptLanguage"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["distributor"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["owner"] = undefined /*out*/;
            inputs["productName"] = undefined /*out*/;
            inputs["provisioningArtifactIds"] = undefined /*out*/;
            inputs["provisioningArtifactNames"] = undefined /*out*/;
            inputs["provisioningArtifactParameters"] = undefined /*out*/;
            inputs["replaceProvisioningArtifacts"] = undefined /*out*/;
            inputs["supportDescription"] = undefined /*out*/;
            inputs["supportEmail"] = undefined /*out*/;
            inputs["supportUrl"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(CloudFormationProduct.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing CloudFormationProduct resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("CloudFormationProduct is deprecated: CloudFormationProduct is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new CloudFormationProduct(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CloudFormationProduct.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CloudFormationProduct.__pulumiType;
    }
}
exports.CloudFormationProduct = CloudFormationProduct;
/** @internal */
CloudFormationProduct.__pulumiType = 'aws-native:servicecatalog:CloudFormationProduct';
