// @pulumi/aws-native/servicecatalog/launchNotificationConstraint.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LaunchNotificationConstraint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::ServiceCatalog::LaunchNotificationConstraint
 *
 * @deprecated LaunchNotificationConstraint is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class LaunchNotificationConstraint extends pulumi.CustomResource {
    /**
     * Create a LaunchNotificationConstraint resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated LaunchNotificationConstraint is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("LaunchNotificationConstraint is deprecated: LaunchNotificationConstraint is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.notificationArns === undefined) && !opts.urn) {
                throw new Error("Missing required property 'notificationArns'");
            }
            if ((!args || args.portfolioId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'portfolioId'");
            }
            if ((!args || args.productId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'productId'");
            }
            inputs["acceptLanguage"] = args ? args.acceptLanguage : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["notificationArns"] = args ? args.notificationArns : undefined;
            inputs["portfolioId"] = args ? args.portfolioId : undefined;
            inputs["productId"] = args ? args.productId : undefined;
        }
        else {
            inputs["acceptLanguage"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["notificationArns"] = undefined /*out*/;
            inputs["portfolioId"] = undefined /*out*/;
            inputs["productId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(LaunchNotificationConstraint.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing LaunchNotificationConstraint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("LaunchNotificationConstraint is deprecated: LaunchNotificationConstraint is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new LaunchNotificationConstraint(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LaunchNotificationConstraint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LaunchNotificationConstraint.__pulumiType;
    }
}
exports.LaunchNotificationConstraint = LaunchNotificationConstraint;
/** @internal */
LaunchNotificationConstraint.__pulumiType = 'aws-native:servicecatalog:LaunchNotificationConstraint';
