// @pulumi/aws-native/synthetics/canary.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Canary = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Synthetics::Canary
 */
class Canary extends pulumi.CustomResource {
    /**
     * Create a Canary resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.artifactS3Location === undefined) && !opts.urn) {
                throw new Error("Missing required property 'artifactS3Location'");
            }
            if ((!args || args.code === undefined) && !opts.urn) {
                throw new Error("Missing required property 'code'");
            }
            if ((!args || args.executionRoleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'executionRoleArn'");
            }
            if ((!args || args.runtimeVersion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'runtimeVersion'");
            }
            if ((!args || args.schedule === undefined) && !opts.urn) {
                throw new Error("Missing required property 'schedule'");
            }
            if ((!args || args.startCanaryAfterCreation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'startCanaryAfterCreation'");
            }
            inputs["artifactConfig"] = args ? args.artifactConfig : undefined;
            inputs["artifactS3Location"] = args ? args.artifactS3Location : undefined;
            inputs["code"] = args ? args.code : undefined;
            inputs["executionRoleArn"] = args ? args.executionRoleArn : undefined;
            inputs["failureRetentionPeriod"] = args ? args.failureRetentionPeriod : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["runConfig"] = args ? args.runConfig : undefined;
            inputs["runtimeVersion"] = args ? args.runtimeVersion : undefined;
            inputs["schedule"] = args ? args.schedule : undefined;
            inputs["startCanaryAfterCreation"] = args ? args.startCanaryAfterCreation : undefined;
            inputs["successRetentionPeriod"] = args ? args.successRetentionPeriod : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["vPCConfig"] = args ? args.vPCConfig : undefined;
            inputs["visualReference"] = args ? args.visualReference : undefined;
            inputs["state"] = undefined /*out*/;
        }
        else {
            inputs["artifactConfig"] = undefined /*out*/;
            inputs["artifactS3Location"] = undefined /*out*/;
            inputs["code"] = undefined /*out*/;
            inputs["executionRoleArn"] = undefined /*out*/;
            inputs["failureRetentionPeriod"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["runConfig"] = undefined /*out*/;
            inputs["runtimeVersion"] = undefined /*out*/;
            inputs["schedule"] = undefined /*out*/;
            inputs["startCanaryAfterCreation"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
            inputs["successRetentionPeriod"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["vPCConfig"] = undefined /*out*/;
            inputs["visualReference"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Canary.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Canary resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Canary(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Canary.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Canary.__pulumiType;
    }
}
exports.Canary = Canary;
/** @internal */
Canary.__pulumiType = 'aws-native:synthetics:Canary';
