// @pulumi/aws-native/appstream/stack.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Stack = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::AppStream::Stack
 *
 * @deprecated Stack is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Stack extends pulumi.CustomResource {
    /**
     * Create a Stack resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Stack is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Stack is deprecated: Stack is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["accessEndpoints"] = args ? args.accessEndpoints : undefined;
            inputs["applicationSettings"] = args ? args.applicationSettings : undefined;
            inputs["attributesToDelete"] = args ? args.attributesToDelete : undefined;
            inputs["deleteStorageConnectors"] = args ? args.deleteStorageConnectors : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["embedHostDomains"] = args ? args.embedHostDomains : undefined;
            inputs["feedbackURL"] = args ? args.feedbackURL : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["redirectURL"] = args ? args.redirectURL : undefined;
            inputs["storageConnectors"] = args ? args.storageConnectors : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["userSettings"] = args ? args.userSettings : undefined;
        }
        else {
            inputs["accessEndpoints"] = undefined /*out*/;
            inputs["applicationSettings"] = undefined /*out*/;
            inputs["attributesToDelete"] = undefined /*out*/;
            inputs["deleteStorageConnectors"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["embedHostDomains"] = undefined /*out*/;
            inputs["feedbackURL"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["redirectURL"] = undefined /*out*/;
            inputs["storageConnectors"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["userSettings"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Stack.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Stack resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Stack is deprecated: Stack is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Stack(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Stack.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Stack.__pulumiType;
    }
}
exports.Stack = Stack;
/** @internal */
Stack.__pulumiType = 'aws-native:appstream:Stack';
