// @pulumi/aws-native/elasticbeanstalk/configurationTemplate.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigurationTemplate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::ElasticBeanstalk::ConfigurationTemplate
 *
 * @deprecated ConfigurationTemplate is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class ConfigurationTemplate extends pulumi.CustomResource {
    /**
     * Create a ConfigurationTemplate resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated ConfigurationTemplate is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("ConfigurationTemplate is deprecated: ConfigurationTemplate is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.applicationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'applicationName'");
            }
            inputs["applicationName"] = args ? args.applicationName : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["environmentId"] = args ? args.environmentId : undefined;
            inputs["optionSettings"] = args ? args.optionSettings : undefined;
            inputs["platformArn"] = args ? args.platformArn : undefined;
            inputs["solutionStackName"] = args ? args.solutionStackName : undefined;
            inputs["sourceConfiguration"] = args ? args.sourceConfiguration : undefined;
        }
        else {
            inputs["applicationName"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["environmentId"] = undefined /*out*/;
            inputs["optionSettings"] = undefined /*out*/;
            inputs["platformArn"] = undefined /*out*/;
            inputs["solutionStackName"] = undefined /*out*/;
            inputs["sourceConfiguration"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ConfigurationTemplate.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ConfigurationTemplate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ConfigurationTemplate is deprecated: ConfigurationTemplate is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new ConfigurationTemplate(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ConfigurationTemplate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConfigurationTemplate.__pulumiType;
    }
}
exports.ConfigurationTemplate = ConfigurationTemplate;
/** @internal */
ConfigurationTemplate.__pulumiType = 'aws-native:elasticbeanstalk:ConfigurationTemplate';
