// @pulumi/aws-native/kinesis/streamConsumer.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamConsumer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Kinesis::StreamConsumer
 *
 * @deprecated StreamConsumer is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class StreamConsumer extends pulumi.CustomResource {
    /**
     * Create a StreamConsumer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated StreamConsumer is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("StreamConsumer is deprecated: StreamConsumer is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.consumerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'consumerName'");
            }
            if ((!args || args.streamARN === undefined) && !opts.urn) {
                throw new Error("Missing required property 'streamARN'");
            }
            inputs["consumerName"] = args ? args.consumerName : undefined;
            inputs["streamARN"] = args ? args.streamARN : undefined;
            inputs["consumerARN"] = undefined /*out*/;
            inputs["consumerCreationTimestamp"] = undefined /*out*/;
            inputs["consumerStatus"] = undefined /*out*/;
        }
        else {
            inputs["consumerARN"] = undefined /*out*/;
            inputs["consumerCreationTimestamp"] = undefined /*out*/;
            inputs["consumerName"] = undefined /*out*/;
            inputs["consumerStatus"] = undefined /*out*/;
            inputs["streamARN"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(StreamConsumer.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing StreamConsumer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("StreamConsumer is deprecated: StreamConsumer is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new StreamConsumer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StreamConsumer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StreamConsumer.__pulumiType;
    }
}
exports.StreamConsumer = StreamConsumer;
/** @internal */
StreamConsumer.__pulumiType = 'aws-native:kinesis:StreamConsumer';
