// @pulumi/aws-native/codestarconnections/connection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Connection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Schema for AWS::CodeStarConnections::Connection resource which can be used to connect external source providers with AWS CodePipeline
 */
class Connection extends pulumi.CustomResource {
    /**
     * Create a Connection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["connectionName"] = args ? args.connectionName : undefined;
            inputs["hostArn"] = args ? args.hostArn : undefined;
            inputs["providerType"] = args ? args.providerType : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["connectionArn"] = undefined /*out*/;
            inputs["connectionStatus"] = undefined /*out*/;
            inputs["ownerAccountId"] = undefined /*out*/;
        }
        else {
            inputs["connectionArn"] = undefined /*out*/;
            inputs["connectionName"] = undefined /*out*/;
            inputs["connectionStatus"] = undefined /*out*/;
            inputs["hostArn"] = undefined /*out*/;
            inputs["ownerAccountId"] = undefined /*out*/;
            inputs["providerType"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Connection.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Connection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Connection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Connection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Connection.__pulumiType;
    }
}
exports.Connection = Connection;
/** @internal */
Connection.__pulumiType = 'aws-native:codestarconnections:Connection';
