// @pulumi/aws-native/iotfleethub/application.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::IoTFleetHub::Application
 */
class Application extends pulumi.CustomResource {
    /**
     * Create a Application resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            inputs["applicationDescription"] = args ? args.applicationDescription : undefined;
            inputs["applicationName"] = args ? args.applicationName : undefined;
            inputs["roleArn"] = args ? args.roleArn : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["applicationArn"] = undefined /*out*/;
            inputs["applicationCreationDate"] = undefined /*out*/;
            inputs["applicationId"] = undefined /*out*/;
            inputs["applicationLastUpdateDate"] = undefined /*out*/;
            inputs["applicationState"] = undefined /*out*/;
            inputs["applicationUrl"] = undefined /*out*/;
            inputs["errorMessage"] = undefined /*out*/;
            inputs["ssoClientId"] = undefined /*out*/;
        }
        else {
            inputs["applicationArn"] = undefined /*out*/;
            inputs["applicationCreationDate"] = undefined /*out*/;
            inputs["applicationDescription"] = undefined /*out*/;
            inputs["applicationId"] = undefined /*out*/;
            inputs["applicationLastUpdateDate"] = undefined /*out*/;
            inputs["applicationName"] = undefined /*out*/;
            inputs["applicationState"] = undefined /*out*/;
            inputs["applicationUrl"] = undefined /*out*/;
            inputs["errorMessage"] = undefined /*out*/;
            inputs["roleArn"] = undefined /*out*/;
            inputs["ssoClientId"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Application.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Application resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Application(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Application.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Application.__pulumiType;
    }
}
exports.Application = Application;
/** @internal */
Application.__pulumiType = 'aws-native:iotfleethub:Application';
