// @pulumi/aws-native/route53/recordSetGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecordSetGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Route53::RecordSetGroup
 *
 * @deprecated RecordSetGroup is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class RecordSetGroup extends pulumi.CustomResource {
    /**
     * Create a RecordSetGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated RecordSetGroup is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("RecordSetGroup is deprecated: RecordSetGroup is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["comment"] = args ? args.comment : undefined;
            inputs["hostedZoneId"] = args ? args.hostedZoneId : undefined;
            inputs["hostedZoneName"] = args ? args.hostedZoneName : undefined;
            inputs["recordSets"] = args ? args.recordSets : undefined;
        }
        else {
            inputs["comment"] = undefined /*out*/;
            inputs["hostedZoneId"] = undefined /*out*/;
            inputs["hostedZoneName"] = undefined /*out*/;
            inputs["recordSets"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(RecordSetGroup.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing RecordSetGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("RecordSetGroup is deprecated: RecordSetGroup is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new RecordSetGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RecordSetGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RecordSetGroup.__pulumiType;
    }
}
exports.RecordSetGroup = RecordSetGroup;
/** @internal */
RecordSetGroup.__pulumiType = 'aws-native:route53:RecordSetGroup';
