// @pulumi/aws-native/types/enums/robomaker/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimulationApplicationSourceConfigArchitecture = exports.SimulationApplicationSimulationSoftwareSuiteVersion = exports.SimulationApplicationSimulationSoftwareSuiteName = exports.SimulationApplicationRobotSoftwareSuiteVersion = exports.SimulationApplicationRobotSoftwareSuiteName = exports.SimulationApplicationRenderingEngineName = exports.RobotArchitecture = void 0;
exports.RobotArchitecture = {
    X8664: "X86_64",
    Arm64: "ARM64",
    Armhf: "ARMHF",
};
exports.SimulationApplicationRenderingEngineName = {
    Ogre: "OGRE",
};
exports.SimulationApplicationRobotSoftwareSuiteName = {
    Ros: "ROS",
    Ros2: "ROS2",
};
exports.SimulationApplicationRobotSoftwareSuiteVersion = {
    Kinetic: "Kinetic",
    Melodic: "Melodic",
    Dashing: "Dashing",
    Foxy: "Foxy",
};
exports.SimulationApplicationSimulationSoftwareSuiteName = {
    Gazebo: "Gazebo",
    RosbagPlay: "RosbagPlay",
};
exports.SimulationApplicationSimulationSoftwareSuiteVersion = {
    Seven: "7",
    Nine: "9",
    SimulationApplicationSimulationSoftwareSuiteVersion_11: "11",
    Kinetic: "Kinetic",
    Melodic: "Melodic",
    Dashing: "Dashing",
    Foxy: "Foxy",
};
exports.SimulationApplicationSourceConfigArchitecture = {
    X8664: "X86_64",
    Arm64: "ARM64",
    Armhf: "ARMHF",
};
