// @pulumi/aws-native/cloudformation/publisher.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Publisher = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Register as a publisher in the CloudFormation Registry.
 */
class Publisher extends pulumi.CustomResource {
    /**
     * Create a Publisher resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.acceptTermsAndConditions === undefined) && !opts.urn) {
                throw new Error("Missing required property 'acceptTermsAndConditions'");
            }
            inputs["acceptTermsAndConditions"] = args ? args.acceptTermsAndConditions : undefined;
            inputs["connectionArn"] = args ? args.connectionArn : undefined;
            inputs["identityProvider"] = undefined /*out*/;
            inputs["publisherId"] = undefined /*out*/;
            inputs["publisherProfile"] = undefined /*out*/;
            inputs["publisherStatus"] = undefined /*out*/;
        }
        else {
            inputs["acceptTermsAndConditions"] = undefined /*out*/;
            inputs["connectionArn"] = undefined /*out*/;
            inputs["identityProvider"] = undefined /*out*/;
            inputs["publisherId"] = undefined /*out*/;
            inputs["publisherProfile"] = undefined /*out*/;
            inputs["publisherStatus"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Publisher.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Publisher resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Publisher(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Publisher.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Publisher.__pulumiType;
    }
}
exports.Publisher = Publisher;
/** @internal */
Publisher.__pulumiType = 'aws-native:cloudformation:Publisher';
