// @pulumi/aws-native/cloudformation/resourceVersion.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A resource that has been registered in the CloudFormation Registry.
 */
class ResourceVersion extends pulumi.CustomResource {
    /**
     * Create a ResourceVersion resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.schemaHandlerPackage === undefined) && !opts.urn) {
                throw new Error("Missing required property 'schemaHandlerPackage'");
            }
            if ((!args || args.typeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'typeName'");
            }
            inputs["executionRoleArn"] = args ? args.executionRoleArn : undefined;
            inputs["loggingConfig"] = args ? args.loggingConfig : undefined;
            inputs["schemaHandlerPackage"] = args ? args.schemaHandlerPackage : undefined;
            inputs["typeName"] = args ? args.typeName : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["isDefaultVersion"] = undefined /*out*/;
            inputs["provisioningType"] = undefined /*out*/;
            inputs["typeArn"] = undefined /*out*/;
            inputs["versionId"] = undefined /*out*/;
            inputs["visibility"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["executionRoleArn"] = undefined /*out*/;
            inputs["isDefaultVersion"] = undefined /*out*/;
            inputs["loggingConfig"] = undefined /*out*/;
            inputs["provisioningType"] = undefined /*out*/;
            inputs["schemaHandlerPackage"] = undefined /*out*/;
            inputs["typeArn"] = undefined /*out*/;
            inputs["typeName"] = undefined /*out*/;
            inputs["versionId"] = undefined /*out*/;
            inputs["visibility"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ResourceVersion.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ResourceVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ResourceVersion(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ResourceVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResourceVersion.__pulumiType;
    }
}
exports.ResourceVersion = ResourceVersion;
/** @internal */
ResourceVersion.__pulumiType = 'aws-native:cloudformation:ResourceVersion';
