// @pulumi/aws-native/cognito/userPoolIdentityProvider.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserPoolIdentityProvider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Cognito::UserPoolIdentityProvider
 *
 * @deprecated UserPoolIdentityProvider is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class UserPoolIdentityProvider extends pulumi.CustomResource {
    /**
     * Create a UserPoolIdentityProvider resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated UserPoolIdentityProvider is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("UserPoolIdentityProvider is deprecated: UserPoolIdentityProvider is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.providerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'providerName'");
            }
            if ((!args || args.providerType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'providerType'");
            }
            if ((!args || args.userPoolId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'userPoolId'");
            }
            inputs["attributeMapping"] = args ? args.attributeMapping : undefined;
            inputs["idpIdentifiers"] = args ? args.idpIdentifiers : undefined;
            inputs["providerDetails"] = args ? args.providerDetails : undefined;
            inputs["providerName"] = args ? args.providerName : undefined;
            inputs["providerType"] = args ? args.providerType : undefined;
            inputs["userPoolId"] = args ? args.userPoolId : undefined;
        }
        else {
            inputs["attributeMapping"] = undefined /*out*/;
            inputs["idpIdentifiers"] = undefined /*out*/;
            inputs["providerDetails"] = undefined /*out*/;
            inputs["providerName"] = undefined /*out*/;
            inputs["providerType"] = undefined /*out*/;
            inputs["userPoolId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(UserPoolIdentityProvider.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing UserPoolIdentityProvider resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("UserPoolIdentityProvider is deprecated: UserPoolIdentityProvider is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new UserPoolIdentityProvider(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of UserPoolIdentityProvider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === UserPoolIdentityProvider.__pulumiType;
    }
}
exports.UserPoolIdentityProvider = UserPoolIdentityProvider;
/** @internal */
UserPoolIdentityProvider.__pulumiType = 'aws-native:cognito:UserPoolIdentityProvider';
