// @pulumi/aws-native/qldb/stream.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Stream = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::QLDB::Stream.
 */
class Stream extends pulumi.CustomResource {
    /**
     * Create a Stream resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.inclusiveStartTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'inclusiveStartTime'");
            }
            if ((!args || args.kinesisConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kinesisConfiguration'");
            }
            if ((!args || args.ledgerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ledgerName'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            inputs["exclusiveEndTime"] = args ? args.exclusiveEndTime : undefined;
            inputs["inclusiveStartTime"] = args ? args.inclusiveStartTime : undefined;
            inputs["kinesisConfiguration"] = args ? args.kinesisConfiguration : undefined;
            inputs["ledgerName"] = args ? args.ledgerName : undefined;
            inputs["roleArn"] = args ? args.roleArn : undefined;
            inputs["streamName"] = args ? args.streamName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["exclusiveEndTime"] = undefined /*out*/;
            inputs["inclusiveStartTime"] = undefined /*out*/;
            inputs["kinesisConfiguration"] = undefined /*out*/;
            inputs["ledgerName"] = undefined /*out*/;
            inputs["roleArn"] = undefined /*out*/;
            inputs["streamName"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Stream.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Stream resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Stream(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Stream.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Stream.__pulumiType;
    }
}
exports.Stream = Stream;
/** @internal */
Stream.__pulumiType = 'aws-native:qldb:Stream';
