// @pulumi/aws-native/eks/nodegroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Nodegroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::EKS::Nodegroup
 *
 * @deprecated Nodegroup is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Nodegroup extends pulumi.CustomResource {
    /**
     * Create a Nodegroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Nodegroup is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Nodegroup is deprecated: Nodegroup is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.nodeRole === undefined) && !opts.urn) {
                throw new Error("Missing required property 'nodeRole'");
            }
            if ((!args || args.subnets === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subnets'");
            }
            inputs["amiType"] = args ? args.amiType : undefined;
            inputs["capacityType"] = args ? args.capacityType : undefined;
            inputs["clusterName"] = args ? args.clusterName : undefined;
            inputs["diskSize"] = args ? args.diskSize : undefined;
            inputs["forceUpdateEnabled"] = args ? args.forceUpdateEnabled : undefined;
            inputs["instanceTypes"] = args ? args.instanceTypes : undefined;
            inputs["labels"] = args ? args.labels : undefined;
            inputs["launchTemplate"] = args ? args.launchTemplate : undefined;
            inputs["nodeRole"] = args ? args.nodeRole : undefined;
            inputs["nodegroupName"] = args ? args.nodegroupName : undefined;
            inputs["releaseVersion"] = args ? args.releaseVersion : undefined;
            inputs["remoteAccess"] = args ? args.remoteAccess : undefined;
            inputs["scalingConfig"] = args ? args.scalingConfig : undefined;
            inputs["subnets"] = args ? args.subnets : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["taints"] = args ? args.taints : undefined;
            inputs["updateConfig"] = args ? args.updateConfig : undefined;
            inputs["version"] = args ? args.version : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["amiType"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
            inputs["capacityType"] = undefined /*out*/;
            inputs["clusterName"] = undefined /*out*/;
            inputs["diskSize"] = undefined /*out*/;
            inputs["forceUpdateEnabled"] = undefined /*out*/;
            inputs["instanceTypes"] = undefined /*out*/;
            inputs["labels"] = undefined /*out*/;
            inputs["launchTemplate"] = undefined /*out*/;
            inputs["nodeRole"] = undefined /*out*/;
            inputs["nodegroupName"] = undefined /*out*/;
            inputs["releaseVersion"] = undefined /*out*/;
            inputs["remoteAccess"] = undefined /*out*/;
            inputs["scalingConfig"] = undefined /*out*/;
            inputs["subnets"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["taints"] = undefined /*out*/;
            inputs["updateConfig"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Nodegroup.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Nodegroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Nodegroup is deprecated: Nodegroup is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Nodegroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Nodegroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Nodegroup.__pulumiType;
    }
}
exports.Nodegroup = Nodegroup;
/** @internal */
Nodegroup.__pulumiType = 'aws-native:eks:Nodegroup';
