// @pulumi/aws-native/dms/replicationTask.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationTask = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::DMS::ReplicationTask
 *
 * @deprecated ReplicationTask is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class ReplicationTask extends pulumi.CustomResource {
    /**
     * Create a ReplicationTask resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated ReplicationTask is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("ReplicationTask is deprecated: ReplicationTask is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.migrationType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'migrationType'");
            }
            if ((!args || args.replicationInstanceArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'replicationInstanceArn'");
            }
            if ((!args || args.sourceEndpointArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceEndpointArn'");
            }
            if ((!args || args.tableMappings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tableMappings'");
            }
            if ((!args || args.targetEndpointArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetEndpointArn'");
            }
            inputs["cdcStartPosition"] = args ? args.cdcStartPosition : undefined;
            inputs["cdcStartTime"] = args ? args.cdcStartTime : undefined;
            inputs["cdcStopPosition"] = args ? args.cdcStopPosition : undefined;
            inputs["migrationType"] = args ? args.migrationType : undefined;
            inputs["replicationInstanceArn"] = args ? args.replicationInstanceArn : undefined;
            inputs["replicationTaskIdentifier"] = args ? args.replicationTaskIdentifier : undefined;
            inputs["replicationTaskSettings"] = args ? args.replicationTaskSettings : undefined;
            inputs["resourceIdentifier"] = args ? args.resourceIdentifier : undefined;
            inputs["sourceEndpointArn"] = args ? args.sourceEndpointArn : undefined;
            inputs["tableMappings"] = args ? args.tableMappings : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["targetEndpointArn"] = args ? args.targetEndpointArn : undefined;
            inputs["taskData"] = args ? args.taskData : undefined;
        }
        else {
            inputs["cdcStartPosition"] = undefined /*out*/;
            inputs["cdcStartTime"] = undefined /*out*/;
            inputs["cdcStopPosition"] = undefined /*out*/;
            inputs["migrationType"] = undefined /*out*/;
            inputs["replicationInstanceArn"] = undefined /*out*/;
            inputs["replicationTaskIdentifier"] = undefined /*out*/;
            inputs["replicationTaskSettings"] = undefined /*out*/;
            inputs["resourceIdentifier"] = undefined /*out*/;
            inputs["sourceEndpointArn"] = undefined /*out*/;
            inputs["tableMappings"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["targetEndpointArn"] = undefined /*out*/;
            inputs["taskData"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ReplicationTask.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ReplicationTask resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ReplicationTask is deprecated: ReplicationTask is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new ReplicationTask(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReplicationTask.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicationTask.__pulumiType;
    }
}
exports.ReplicationTask = ReplicationTask;
/** @internal */
ReplicationTask.__pulumiType = 'aws-native:dms:ReplicationTask';
