// @pulumi/aws-native/iotcoredeviceadvisor/suiteDefinition.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SuiteDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An example resource schema demonstrating some basic constructs and validation rules.
 */
class SuiteDefinition extends pulumi.CustomResource {
    /**
     * Create a SuiteDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.suiteDefinitionConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'suiteDefinitionConfiguration'");
            }
            inputs["suiteDefinitionConfiguration"] = args ? args.suiteDefinitionConfiguration : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["suiteDefinitionArn"] = undefined /*out*/;
            inputs["suiteDefinitionId"] = undefined /*out*/;
            inputs["suiteDefinitionVersion"] = undefined /*out*/;
        }
        else {
            inputs["suiteDefinitionArn"] = undefined /*out*/;
            inputs["suiteDefinitionConfiguration"] = undefined /*out*/;
            inputs["suiteDefinitionId"] = undefined /*out*/;
            inputs["suiteDefinitionVersion"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(SuiteDefinition.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SuiteDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SuiteDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SuiteDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SuiteDefinition.__pulumiType;
    }
}
exports.SuiteDefinition = SuiteDefinition;
/** @internal */
SuiteDefinition.__pulumiType = 'aws-native:iotcoredeviceadvisor:SuiteDefinition';
