// @pulumi/aws-native/eventschemas/discoverer.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Discoverer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::EventSchemas::Discoverer
 *
 * @deprecated Discoverer is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Discoverer extends pulumi.CustomResource {
    /**
     * Create a Discoverer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Discoverer is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Discoverer is deprecated: Discoverer is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.sourceArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceArn'");
            }
            inputs["crossAccount"] = args ? args.crossAccount : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["sourceArn"] = args ? args.sourceArn : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["discovererArn"] = undefined /*out*/;
            inputs["discovererId"] = undefined /*out*/;
        }
        else {
            inputs["crossAccount"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["discovererArn"] = undefined /*out*/;
            inputs["discovererId"] = undefined /*out*/;
            inputs["sourceArn"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Discoverer.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Discoverer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Discoverer is deprecated: Discoverer is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Discoverer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Discoverer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Discoverer.__pulumiType;
    }
}
exports.Discoverer = Discoverer;
/** @internal */
Discoverer.__pulumiType = 'aws-native:eventschemas:Discoverer';
