// @pulumi/aws-native/wafv2/webACL.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebACL = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Contains the Rules that identify the requests that you want to allow, block, or count. In a WebACL, you also specify a default action (ALLOW or BLOCK), and the action for each Rule that you add to a WebACL, for example, block requests from specified IP addresses or block requests from specified referrers. You also associate the WebACL with a CloudFront distribution to identify the requests that you want AWS WAF to filter. If you add more than one Rule to a WebACL, a request needs to match only one of the specifications to be allowed, blocked, or counted.
 */
class WebACL extends pulumi.CustomResource {
    /**
     * Create a WebACL resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.defaultAction === undefined) && !opts.urn) {
                throw new Error("Missing required property 'defaultAction'");
            }
            if ((!args || args.scope === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scope'");
            }
            if ((!args || args.visibilityConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'visibilityConfig'");
            }
            inputs["customResponseBodies"] = args ? args.customResponseBodies : undefined;
            inputs["defaultAction"] = args ? args.defaultAction : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["rules"] = args ? args.rules : undefined;
            inputs["scope"] = args ? args.scope : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["visibilityConfig"] = args ? args.visibilityConfig : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["capacity"] = undefined /*out*/;
            inputs["labelNamespace"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["capacity"] = undefined /*out*/;
            inputs["customResponseBodies"] = undefined /*out*/;
            inputs["defaultAction"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["labelNamespace"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["rules"] = undefined /*out*/;
            inputs["scope"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["visibilityConfig"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(WebACL.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WebACL resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebACL(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebACL.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebACL.__pulumiType;
    }
}
exports.WebACL = WebACL;
/** @internal */
WebACL.__pulumiType = 'aws-native:wafv2:WebACL';
