// @pulumi/aws-native/types/enums/location/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrackerPricingPlan = exports.TrackerPositionFiltering = exports.RouteCalculatorPricingPlan = exports.PlaceIndexPricingPlan = exports.PlaceIndexIntendedUse = exports.MapPricingPlan = exports.GeofenceCollectionPricingPlan = void 0;
exports.GeofenceCollectionPricingPlan = {
    RequestBasedUsage: "RequestBasedUsage",
    MobileAssetTracking: "MobileAssetTracking",
    MobileAssetManagement: "MobileAssetManagement",
};
exports.MapPricingPlan = {
    RequestBasedUsage: "RequestBasedUsage",
    MobileAssetTracking: "MobileAssetTracking",
    MobileAssetManagement: "MobileAssetManagement",
};
exports.PlaceIndexIntendedUse = {
    SingleUse: "SingleUse",
    Storage: "Storage",
};
exports.PlaceIndexPricingPlan = {
    RequestBasedUsage: "RequestBasedUsage",
    MobileAssetTracking: "MobileAssetTracking",
    MobileAssetManagement: "MobileAssetManagement",
};
exports.RouteCalculatorPricingPlan = {
    RequestBasedUsage: "RequestBasedUsage",
    MobileAssetManagement: "MobileAssetManagement",
};
exports.TrackerPositionFiltering = {
    TimeBased: "TimeBased",
    DistanceBased: "DistanceBased",
};
exports.TrackerPricingPlan = {
    RequestBasedUsage: "RequestBasedUsage",
    MobileAssetTracking: "MobileAssetTracking",
    MobileAssetManagement: "MobileAssetManagement",
};
