// @pulumi/aws-native/iotanalytics/dataset.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Dataset = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::IoTAnalytics::Dataset
 *
 * @deprecated Dataset is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Dataset extends pulumi.CustomResource {
    /**
     * Create a Dataset resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Dataset is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Dataset is deprecated: Dataset is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.actions === undefined) && !opts.urn) {
                throw new Error("Missing required property 'actions'");
            }
            inputs["actions"] = args ? args.actions : undefined;
            inputs["contentDeliveryRules"] = args ? args.contentDeliveryRules : undefined;
            inputs["datasetName"] = args ? args.datasetName : undefined;
            inputs["lateDataRules"] = args ? args.lateDataRules : undefined;
            inputs["retentionPeriod"] = args ? args.retentionPeriod : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["triggers"] = args ? args.triggers : undefined;
            inputs["versioningConfiguration"] = args ? args.versioningConfiguration : undefined;
        }
        else {
            inputs["actions"] = undefined /*out*/;
            inputs["contentDeliveryRules"] = undefined /*out*/;
            inputs["datasetName"] = undefined /*out*/;
            inputs["lateDataRules"] = undefined /*out*/;
            inputs["retentionPeriod"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["triggers"] = undefined /*out*/;
            inputs["versioningConfiguration"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Dataset.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Dataset resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Dataset is deprecated: Dataset is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Dataset(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Dataset.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Dataset.__pulumiType;
    }
}
exports.Dataset = Dataset;
/** @internal */
Dataset.__pulumiType = 'aws-native:iotanalytics:Dataset';
