// @pulumi/aws-native/types/enums/sagemaker/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserProfileSharingSettingsNotebookOutputOption = exports.UserProfileResourceSpecInstanceType = exports.ProjectStatus = exports.MonitoringScheduleStatus = exports.MonitoringScheduleS3OutputS3UploadMode = exports.MonitoringScheduleMonitoringType = exports.MonitoringScheduleMonitoringExecutionSummaryMonitoringExecutionStatus = exports.MonitoringScheduleEndpointInputS3InputMode = exports.MonitoringScheduleEndpointInputS3DataDistributionType = exports.ModelQualityJobDefinitionS3OutputS3UploadMode = exports.ModelQualityJobDefinitionProblemType = exports.ModelQualityJobDefinitionEndpointInputS3InputMode = exports.ModelQualityJobDefinitionEndpointInputS3DataDistributionType = exports.ModelPackageGroupStatus = exports.ModelExplainabilityJobDefinitionS3OutputS3UploadMode = exports.ModelExplainabilityJobDefinitionEndpointInputS3InputMode = exports.ModelExplainabilityJobDefinitionEndpointInputS3DataDistributionType = exports.ModelBiasJobDefinitionS3OutputS3UploadMode = exports.ModelBiasJobDefinitionEndpointInputS3InputMode = exports.ModelBiasJobDefinitionEndpointInputS3DataDistributionType = exports.FeatureGroupFeatureDefinitionFeatureType = exports.DomainSharingSettingsNotebookOutputOption = exports.DomainResourceSpecInstanceType = exports.DomainAuthMode = exports.DomainAppNetworkAccessType = exports.DataQualityJobDefinitionS3OutputS3UploadMode = exports.DataQualityJobDefinitionEndpointInputS3InputMode = exports.DataQualityJobDefinitionEndpointInputS3DataDistributionType = exports.AppType = exports.AppResourceSpecInstanceType = void 0;
exports.AppResourceSpecInstanceType = {
    System: "system",
    MlT3Micro: "ml.t3.micro",
    MlT3Small: "ml.t3.small",
    MlT3Medium: "ml.t3.medium",
    MlT3Large: "ml.t3.large",
    MlT3Xlarge: "ml.t3.xlarge",
    MlT32xlarge: "ml.t3.2xlarge",
    MlM5Large: "ml.m5.large",
    MlM5Xlarge: "ml.m5.xlarge",
    MlM52xlarge: "ml.m5.2xlarge",
    MlM54xlarge: "ml.m5.4xlarge",
    MlM58xlarge: "ml.m5.8xlarge",
    MlM512xlarge: "ml.m5.12xlarge",
    MlM516xlarge: "ml.m5.16xlarge",
    MlM524xlarge: "ml.m5.24xlarge",
    MlC5Large: "ml.c5.large",
    MlC5Xlarge: "ml.c5.xlarge",
    MlC52xlarge: "ml.c5.2xlarge",
    MlC54xlarge: "ml.c5.4xlarge",
    MlC59xlarge: "ml.c5.9xlarge",
    MlC512xlarge: "ml.c5.12xlarge",
    MlC518xlarge: "ml.c5.18xlarge",
    MlC524xlarge: "ml.c5.24xlarge",
    MlP32xlarge: "ml.p3.2xlarge",
    MlP38xlarge: "ml.p3.8xlarge",
    MlP316xlarge: "ml.p3.16xlarge",
    MlG4dnXlarge: "ml.g4dn.xlarge",
    MlG4dn2xlarge: "ml.g4dn.2xlarge",
    MlG4dn4xlarge: "ml.g4dn.4xlarge",
    MlG4dn8xlarge: "ml.g4dn.8xlarge",
    MlG4dn12xlarge: "ml.g4dn.12xlarge",
    MlG4dn16xlarge: "ml.g4dn.16xlarge",
};
exports.AppType = {
    JupyterServer: "JupyterServer",
    KernelGateway: "KernelGateway",
};
exports.DataQualityJobDefinitionEndpointInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.DataQualityJobDefinitionEndpointInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.DataQualityJobDefinitionS3OutputS3UploadMode = {
    Continuous: "Continuous",
    EndOfJob: "EndOfJob",
};
exports.DomainAppNetworkAccessType = {
    PublicInternetOnly: "PublicInternetOnly",
    VpcOnly: "VpcOnly",
};
exports.DomainAuthMode = {
    Sso: "SSO",
    Iam: "IAM",
};
exports.DomainResourceSpecInstanceType = {
    System: "system",
    MlT3Micro: "ml.t3.micro",
    MlT3Small: "ml.t3.small",
    MlT3Medium: "ml.t3.medium",
    MlT3Large: "ml.t3.large",
    MlT3Xlarge: "ml.t3.xlarge",
    MlT32xlarge: "ml.t3.2xlarge",
    MlM5Large: "ml.m5.large",
    MlM5Xlarge: "ml.m5.xlarge",
    MlM52xlarge: "ml.m5.2xlarge",
    MlM54xlarge: "ml.m5.4xlarge",
    MlM58xlarge: "ml.m5.8xlarge",
    MlM512xlarge: "ml.m5.12xlarge",
    MlM516xlarge: "ml.m5.16xlarge",
    MlM524xlarge: "ml.m5.24xlarge",
    MlC5Large: "ml.c5.large",
    MlC5Xlarge: "ml.c5.xlarge",
    MlC52xlarge: "ml.c5.2xlarge",
    MlC54xlarge: "ml.c5.4xlarge",
    MlC59xlarge: "ml.c5.9xlarge",
    MlC512xlarge: "ml.c5.12xlarge",
    MlC518xlarge: "ml.c5.18xlarge",
    MlC524xlarge: "ml.c5.24xlarge",
    MlP32xlarge: "ml.p3.2xlarge",
    MlP38xlarge: "ml.p3.8xlarge",
    MlP316xlarge: "ml.p3.16xlarge",
    MlG4dnXlarge: "ml.g4dn.xlarge",
    MlG4dn2xlarge: "ml.g4dn.2xlarge",
    MlG4dn4xlarge: "ml.g4dn.4xlarge",
    MlG4dn8xlarge: "ml.g4dn.8xlarge",
    MlG4dn12xlarge: "ml.g4dn.12xlarge",
    MlG4dn16xlarge: "ml.g4dn.16xlarge",
};
exports.DomainSharingSettingsNotebookOutputOption = {
    Allowed: "Allowed",
    Disabled: "Disabled",
};
exports.FeatureGroupFeatureDefinitionFeatureType = {
    Integral: "Integral",
    Fractional: "Fractional",
    String: "String",
};
exports.ModelBiasJobDefinitionEndpointInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.ModelBiasJobDefinitionEndpointInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.ModelBiasJobDefinitionS3OutputS3UploadMode = {
    Continuous: "Continuous",
    EndOfJob: "EndOfJob",
};
exports.ModelExplainabilityJobDefinitionEndpointInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.ModelExplainabilityJobDefinitionEndpointInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.ModelExplainabilityJobDefinitionS3OutputS3UploadMode = {
    Continuous: "Continuous",
    EndOfJob: "EndOfJob",
};
exports.ModelPackageGroupStatus = {
    Pending: "Pending",
    InProgress: "InProgress",
    Completed: "Completed",
    Failed: "Failed",
    Deleting: "Deleting",
    DeleteFailed: "DeleteFailed",
};
exports.ModelQualityJobDefinitionEndpointInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.ModelQualityJobDefinitionEndpointInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.ModelQualityJobDefinitionProblemType = {
    BinaryClassification: "BinaryClassification",
    MulticlassClassification: "MulticlassClassification",
    Regression: "Regression",
};
exports.ModelQualityJobDefinitionS3OutputS3UploadMode = {
    Continuous: "Continuous",
    EndOfJob: "EndOfJob",
};
exports.MonitoringScheduleEndpointInputS3DataDistributionType = {
    FullyReplicated: "FullyReplicated",
    ShardedByS3Key: "ShardedByS3Key",
};
exports.MonitoringScheduleEndpointInputS3InputMode = {
    Pipe: "Pipe",
    File: "File",
};
exports.MonitoringScheduleMonitoringExecutionSummaryMonitoringExecutionStatus = {
    Pending: "Pending",
    Completed: "Completed",
    CompletedWithViolations: "CompletedWithViolations",
    InProgress: "InProgress",
    Failed: "Failed",
    Stopping: "Stopping",
    Stopped: "Stopped",
};
exports.MonitoringScheduleMonitoringType = {
    DataQuality: "DataQuality",
    ModelQuality: "ModelQuality",
    ModelBias: "ModelBias",
    ModelExplainability: "ModelExplainability",
};
exports.MonitoringScheduleS3OutputS3UploadMode = {
    Continuous: "Continuous",
    EndOfJob: "EndOfJob",
};
exports.MonitoringScheduleStatus = {
    Pending: "Pending",
    Failed: "Failed",
    Scheduled: "Scheduled",
    Stopped: "Stopped",
};
exports.ProjectStatus = {
    Pending: "Pending",
    CreateInProgress: "CreateInProgress",
    CreateCompleted: "CreateCompleted",
    CreateFailed: "CreateFailed",
    DeleteInProgress: "DeleteInProgress",
    DeleteFailed: "DeleteFailed",
    DeleteCompleted: "DeleteCompleted",
};
exports.UserProfileResourceSpecInstanceType = {
    System: "system",
    MlT3Micro: "ml.t3.micro",
    MlT3Small: "ml.t3.small",
    MlT3Medium: "ml.t3.medium",
    MlT3Large: "ml.t3.large",
    MlT3Xlarge: "ml.t3.xlarge",
    MlT32xlarge: "ml.t3.2xlarge",
    MlM5Large: "ml.m5.large",
    MlM5Xlarge: "ml.m5.xlarge",
    MlM52xlarge: "ml.m5.2xlarge",
    MlM54xlarge: "ml.m5.4xlarge",
    MlM58xlarge: "ml.m5.8xlarge",
    MlM512xlarge: "ml.m5.12xlarge",
    MlM516xlarge: "ml.m5.16xlarge",
    MlM524xlarge: "ml.m5.24xlarge",
    MlC5Large: "ml.c5.large",
    MlC5Xlarge: "ml.c5.xlarge",
    MlC52xlarge: "ml.c5.2xlarge",
    MlC54xlarge: "ml.c5.4xlarge",
    MlC59xlarge: "ml.c5.9xlarge",
    MlC512xlarge: "ml.c5.12xlarge",
    MlC518xlarge: "ml.c5.18xlarge",
    MlC524xlarge: "ml.c5.24xlarge",
    MlP32xlarge: "ml.p3.2xlarge",
    MlP38xlarge: "ml.p3.8xlarge",
    MlP316xlarge: "ml.p3.16xlarge",
    MlG4dnXlarge: "ml.g4dn.xlarge",
    MlG4dn2xlarge: "ml.g4dn.2xlarge",
    MlG4dn4xlarge: "ml.g4dn.4xlarge",
    MlG4dn8xlarge: "ml.g4dn.8xlarge",
    MlG4dn12xlarge: "ml.g4dn.12xlarge",
    MlG4dn16xlarge: "ml.g4dn.16xlarge",
};
exports.UserProfileSharingSettingsNotebookOutputOption = {
    Allowed: "Allowed",
    Disabled: "Disabled",
};
