// @pulumi/aws-native/lightsail/disk.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Disk = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Lightsail::Disk
 */
class Disk extends pulumi.CustomResource {
    /**
     * Create a Disk resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.sizeInGb === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sizeInGb'");
            }
            inputs["addOns"] = args ? args.addOns : undefined;
            inputs["availabilityZone"] = args ? args.availabilityZone : undefined;
            inputs["diskName"] = args ? args.diskName : undefined;
            inputs["sizeInGb"] = args ? args.sizeInGb : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["attachedTo"] = undefined /*out*/;
            inputs["attachmentState"] = undefined /*out*/;
            inputs["diskArn"] = undefined /*out*/;
            inputs["iops"] = undefined /*out*/;
            inputs["isAttached"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["path"] = undefined /*out*/;
            inputs["resourceType"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
            inputs["supportCode"] = undefined /*out*/;
        }
        else {
            inputs["addOns"] = undefined /*out*/;
            inputs["attachedTo"] = undefined /*out*/;
            inputs["attachmentState"] = undefined /*out*/;
            inputs["availabilityZone"] = undefined /*out*/;
            inputs["diskArn"] = undefined /*out*/;
            inputs["diskName"] = undefined /*out*/;
            inputs["iops"] = undefined /*out*/;
            inputs["isAttached"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["path"] = undefined /*out*/;
            inputs["resourceType"] = undefined /*out*/;
            inputs["sizeInGb"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
            inputs["supportCode"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Disk.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Disk resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Disk(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Disk.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Disk.__pulumiType;
    }
}
exports.Disk = Disk;
/** @internal */
Disk.__pulumiType = 'aws-native:lightsail:Disk';
