// @pulumi/aws-native/codebuild/project.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Project = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::CodeBuild::Project
 *
 * @deprecated Project is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Project extends pulumi.CustomResource {
    /**
     * Create a Project resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Project is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Project is deprecated: Project is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.artifacts === undefined) && !opts.urn) {
                throw new Error("Missing required property 'artifacts'");
            }
            if ((!args || args.environment === undefined) && !opts.urn) {
                throw new Error("Missing required property 'environment'");
            }
            if ((!args || args.serviceRole === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceRole'");
            }
            if ((!args || args.source === undefined) && !opts.urn) {
                throw new Error("Missing required property 'source'");
            }
            inputs["artifacts"] = args ? args.artifacts : undefined;
            inputs["badgeEnabled"] = args ? args.badgeEnabled : undefined;
            inputs["buildBatchConfig"] = args ? args.buildBatchConfig : undefined;
            inputs["cache"] = args ? args.cache : undefined;
            inputs["concurrentBuildLimit"] = args ? args.concurrentBuildLimit : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["encryptionKey"] = args ? args.encryptionKey : undefined;
            inputs["environment"] = args ? args.environment : undefined;
            inputs["fileSystemLocations"] = args ? args.fileSystemLocations : undefined;
            inputs["logsConfig"] = args ? args.logsConfig : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["queuedTimeoutInMinutes"] = args ? args.queuedTimeoutInMinutes : undefined;
            inputs["resourceAccessRole"] = args ? args.resourceAccessRole : undefined;
            inputs["secondaryArtifacts"] = args ? args.secondaryArtifacts : undefined;
            inputs["secondarySourceVersions"] = args ? args.secondarySourceVersions : undefined;
            inputs["secondarySources"] = args ? args.secondarySources : undefined;
            inputs["serviceRole"] = args ? args.serviceRole : undefined;
            inputs["source"] = args ? args.source : undefined;
            inputs["sourceVersion"] = args ? args.sourceVersion : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["timeoutInMinutes"] = args ? args.timeoutInMinutes : undefined;
            inputs["triggers"] = args ? args.triggers : undefined;
            inputs["visibility"] = args ? args.visibility : undefined;
            inputs["vpcConfig"] = args ? args.vpcConfig : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["artifacts"] = undefined /*out*/;
            inputs["badgeEnabled"] = undefined /*out*/;
            inputs["buildBatchConfig"] = undefined /*out*/;
            inputs["cache"] = undefined /*out*/;
            inputs["concurrentBuildLimit"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["encryptionKey"] = undefined /*out*/;
            inputs["environment"] = undefined /*out*/;
            inputs["fileSystemLocations"] = undefined /*out*/;
            inputs["logsConfig"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["queuedTimeoutInMinutes"] = undefined /*out*/;
            inputs["resourceAccessRole"] = undefined /*out*/;
            inputs["secondaryArtifacts"] = undefined /*out*/;
            inputs["secondarySourceVersions"] = undefined /*out*/;
            inputs["secondarySources"] = undefined /*out*/;
            inputs["serviceRole"] = undefined /*out*/;
            inputs["source"] = undefined /*out*/;
            inputs["sourceVersion"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["timeoutInMinutes"] = undefined /*out*/;
            inputs["triggers"] = undefined /*out*/;
            inputs["visibility"] = undefined /*out*/;
            inputs["vpcConfig"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Project.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Project resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Project is deprecated: Project is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Project(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Project.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Project.__pulumiType;
    }
}
exports.Project = Project;
/** @internal */
Project.__pulumiType = 'aws-native:codebuild:Project';
