// @pulumi/aws-native/configuration/remediationConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemediationConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Config::RemediationConfiguration
 *
 * @deprecated RemediationConfiguration is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class RemediationConfiguration extends pulumi.CustomResource {
    /**
     * Create a RemediationConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated RemediationConfiguration is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("RemediationConfiguration is deprecated: RemediationConfiguration is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.configRuleName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'configRuleName'");
            }
            if ((!args || args.targetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetId'");
            }
            if ((!args || args.targetType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetType'");
            }
            inputs["automatic"] = args ? args.automatic : undefined;
            inputs["configRuleName"] = args ? args.configRuleName : undefined;
            inputs["executionControls"] = args ? args.executionControls : undefined;
            inputs["maximumAutomaticAttempts"] = args ? args.maximumAutomaticAttempts : undefined;
            inputs["parameters"] = args ? args.parameters : undefined;
            inputs["resourceType"] = args ? args.resourceType : undefined;
            inputs["retryAttemptSeconds"] = args ? args.retryAttemptSeconds : undefined;
            inputs["targetId"] = args ? args.targetId : undefined;
            inputs["targetType"] = args ? args.targetType : undefined;
            inputs["targetVersion"] = args ? args.targetVersion : undefined;
        }
        else {
            inputs["automatic"] = undefined /*out*/;
            inputs["configRuleName"] = undefined /*out*/;
            inputs["executionControls"] = undefined /*out*/;
            inputs["maximumAutomaticAttempts"] = undefined /*out*/;
            inputs["parameters"] = undefined /*out*/;
            inputs["resourceType"] = undefined /*out*/;
            inputs["retryAttemptSeconds"] = undefined /*out*/;
            inputs["targetId"] = undefined /*out*/;
            inputs["targetType"] = undefined /*out*/;
            inputs["targetVersion"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(RemediationConfiguration.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing RemediationConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("RemediationConfiguration is deprecated: RemediationConfiguration is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new RemediationConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RemediationConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RemediationConfiguration.__pulumiType;
    }
}
exports.RemediationConfiguration = RemediationConfiguration;
/** @internal */
RemediationConfiguration.__pulumiType = 'aws-native:configuration:RemediationConfiguration';
