// @pulumi/aws-native/batch/jobDefinition.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Batch::JobDefinition
 *
 * @deprecated JobDefinition is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class JobDefinition extends pulumi.CustomResource {
    /**
     * Create a JobDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated JobDefinition is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("JobDefinition is deprecated: JobDefinition is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            inputs["containerProperties"] = args ? args.containerProperties : undefined;
            inputs["jobDefinitionName"] = args ? args.jobDefinitionName : undefined;
            inputs["nodeProperties"] = args ? args.nodeProperties : undefined;
            inputs["parameters"] = args ? args.parameters : undefined;
            inputs["platformCapabilities"] = args ? args.platformCapabilities : undefined;
            inputs["propagateTags"] = args ? args.propagateTags : undefined;
            inputs["retryStrategy"] = args ? args.retryStrategy : undefined;
            inputs["schedulingPriority"] = args ? args.schedulingPriority : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["timeout"] = args ? args.timeout : undefined;
            inputs["type"] = args ? args.type : undefined;
        }
        else {
            inputs["containerProperties"] = undefined /*out*/;
            inputs["jobDefinitionName"] = undefined /*out*/;
            inputs["nodeProperties"] = undefined /*out*/;
            inputs["parameters"] = undefined /*out*/;
            inputs["platformCapabilities"] = undefined /*out*/;
            inputs["propagateTags"] = undefined /*out*/;
            inputs["retryStrategy"] = undefined /*out*/;
            inputs["schedulingPriority"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["timeout"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(JobDefinition.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing JobDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("JobDefinition is deprecated: JobDefinition is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new JobDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of JobDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === JobDefinition.__pulumiType;
    }
}
exports.JobDefinition = JobDefinition;
/** @internal */
JobDefinition.__pulumiType = 'aws-native:batch:JobDefinition';
