// @pulumi/aws-native/secretsmanager/rotationSchedule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RotationSchedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::SecretsManager::RotationSchedule
 *
 * @deprecated RotationSchedule is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class RotationSchedule extends pulumi.CustomResource {
    /**
     * Create a RotationSchedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated RotationSchedule is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("RotationSchedule is deprecated: RotationSchedule is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.secretId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'secretId'");
            }
            inputs["hostedRotationLambda"] = args ? args.hostedRotationLambda : undefined;
            inputs["rotationLambdaARN"] = args ? args.rotationLambdaARN : undefined;
            inputs["rotationRules"] = args ? args.rotationRules : undefined;
            inputs["secretId"] = args ? args.secretId : undefined;
        }
        else {
            inputs["hostedRotationLambda"] = undefined /*out*/;
            inputs["rotationLambdaARN"] = undefined /*out*/;
            inputs["rotationRules"] = undefined /*out*/;
            inputs["secretId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(RotationSchedule.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing RotationSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("RotationSchedule is deprecated: RotationSchedule is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new RotationSchedule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RotationSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RotationSchedule.__pulumiType;
    }
}
exports.RotationSchedule = RotationSchedule;
/** @internal */
RotationSchedule.__pulumiType = 'aws-native:secretsmanager:RotationSchedule';
