// @pulumi/aws-native/appflow/flow.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Flow = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::AppFlow::Flow.
 */
class Flow extends pulumi.CustomResource {
    /**
     * Create a Flow resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.destinationFlowConfigList === undefined) && !opts.urn) {
                throw new Error("Missing required property 'destinationFlowConfigList'");
            }
            if ((!args || args.sourceFlowConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceFlowConfig'");
            }
            if ((!args || args.tasks === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tasks'");
            }
            if ((!args || args.triggerConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'triggerConfig'");
            }
            inputs["description"] = args ? args.description : undefined;
            inputs["destinationFlowConfigList"] = args ? args.destinationFlowConfigList : undefined;
            inputs["flowName"] = args ? args.flowName : undefined;
            inputs["kMSArn"] = args ? args.kMSArn : undefined;
            inputs["sourceFlowConfig"] = args ? args.sourceFlowConfig : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["tasks"] = args ? args.tasks : undefined;
            inputs["triggerConfig"] = args ? args.triggerConfig : undefined;
            inputs["flowArn"] = undefined /*out*/;
        }
        else {
            inputs["description"] = undefined /*out*/;
            inputs["destinationFlowConfigList"] = undefined /*out*/;
            inputs["flowArn"] = undefined /*out*/;
            inputs["flowName"] = undefined /*out*/;
            inputs["kMSArn"] = undefined /*out*/;
            inputs["sourceFlowConfig"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["tasks"] = undefined /*out*/;
            inputs["triggerConfig"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Flow.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Flow resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Flow(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Flow.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Flow.__pulumiType;
    }
}
exports.Flow = Flow;
/** @internal */
Flow.__pulumiType = 'aws-native:appflow:Flow';
