// @pulumi/aws-native/redshift/endpointAccess.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointAccess = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for a Redshift-managed VPC endpoint.
 */
class EndpointAccess extends pulumi.CustomResource {
    /**
     * Create a EndpointAccess resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.endpointName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'endpointName'");
            }
            if ((!args || args.vpcSecurityGroupIds === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vpcSecurityGroupIds'");
            }
            inputs["clusterIdentifier"] = args ? args.clusterIdentifier : undefined;
            inputs["endpointName"] = args ? args.endpointName : undefined;
            inputs["resourceOwner"] = args ? args.resourceOwner : undefined;
            inputs["subnetGroupName"] = args ? args.subnetGroupName : undefined;
            inputs["vpcSecurityGroupIds"] = args ? args.vpcSecurityGroupIds : undefined;
            inputs["address"] = undefined /*out*/;
            inputs["endpointCreateTime"] = undefined /*out*/;
            inputs["endpointStatus"] = undefined /*out*/;
            inputs["port"] = undefined /*out*/;
            inputs["vpcEndpoint"] = undefined /*out*/;
            inputs["vpcSecurityGroups"] = undefined /*out*/;
        }
        else {
            inputs["address"] = undefined /*out*/;
            inputs["clusterIdentifier"] = undefined /*out*/;
            inputs["endpointCreateTime"] = undefined /*out*/;
            inputs["endpointName"] = undefined /*out*/;
            inputs["endpointStatus"] = undefined /*out*/;
            inputs["port"] = undefined /*out*/;
            inputs["resourceOwner"] = undefined /*out*/;
            inputs["subnetGroupName"] = undefined /*out*/;
            inputs["vpcEndpoint"] = undefined /*out*/;
            inputs["vpcSecurityGroupIds"] = undefined /*out*/;
            inputs["vpcSecurityGroups"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(EndpointAccess.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing EndpointAccess resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EndpointAccess(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EndpointAccess.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EndpointAccess.__pulumiType;
    }
}
exports.EndpointAccess = EndpointAccess;
/** @internal */
EndpointAccess.__pulumiType = 'aws-native:redshift:EndpointAccess';
