// @pulumi/aws-native/redshift/endpointAuthorization.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointAuthorization = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Describes an endpoint authorization for authorizing Redshift-managed VPC endpoint access to a cluster across AWS accounts.
 */
class EndpointAuthorization extends pulumi.CustomResource {
    /**
     * Create a EndpointAuthorization resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.account === undefined) && !opts.urn) {
                throw new Error("Missing required property 'account'");
            }
            if ((!args || args.clusterIdentifier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterIdentifier'");
            }
            inputs["account"] = args ? args.account : undefined;
            inputs["clusterIdentifier"] = args ? args.clusterIdentifier : undefined;
            inputs["force"] = args ? args.force : undefined;
            inputs["vpcIds"] = args ? args.vpcIds : undefined;
            inputs["allowedAllVPCs"] = undefined /*out*/;
            inputs["allowedVPCs"] = undefined /*out*/;
            inputs["authorizeTime"] = undefined /*out*/;
            inputs["clusterStatus"] = undefined /*out*/;
            inputs["endpointCount"] = undefined /*out*/;
            inputs["grantee"] = undefined /*out*/;
            inputs["grantor"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
        }
        else {
            inputs["account"] = undefined /*out*/;
            inputs["allowedAllVPCs"] = undefined /*out*/;
            inputs["allowedVPCs"] = undefined /*out*/;
            inputs["authorizeTime"] = undefined /*out*/;
            inputs["clusterIdentifier"] = undefined /*out*/;
            inputs["clusterStatus"] = undefined /*out*/;
            inputs["endpointCount"] = undefined /*out*/;
            inputs["force"] = undefined /*out*/;
            inputs["grantee"] = undefined /*out*/;
            inputs["grantor"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["vpcIds"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(EndpointAuthorization.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing EndpointAuthorization resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EndpointAuthorization(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EndpointAuthorization.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EndpointAuthorization.__pulumiType;
    }
}
exports.EndpointAuthorization = EndpointAuthorization;
/** @internal */
EndpointAuthorization.__pulumiType = 'aws-native:redshift:EndpointAuthorization';
