// @pulumi/aws-native/redshift/scheduledAction.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduledAction = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The `AWS::Redshift::ScheduledAction` resource creates an Amazon Redshift Scheduled Action.
 */
class ScheduledAction extends pulumi.CustomResource {
    /**
     * Create a ScheduledAction resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["enable"] = args ? args.enable : undefined;
            inputs["endTime"] = args ? args.endTime : undefined;
            inputs["iamRole"] = args ? args.iamRole : undefined;
            inputs["schedule"] = args ? args.schedule : undefined;
            inputs["scheduledActionDescription"] = args ? args.scheduledActionDescription : undefined;
            inputs["scheduledActionName"] = args ? args.scheduledActionName : undefined;
            inputs["startTime"] = args ? args.startTime : undefined;
            inputs["targetAction"] = args ? args.targetAction : undefined;
            inputs["nextInvocations"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
        }
        else {
            inputs["enable"] = undefined /*out*/;
            inputs["endTime"] = undefined /*out*/;
            inputs["iamRole"] = undefined /*out*/;
            inputs["nextInvocations"] = undefined /*out*/;
            inputs["schedule"] = undefined /*out*/;
            inputs["scheduledActionDescription"] = undefined /*out*/;
            inputs["scheduledActionName"] = undefined /*out*/;
            inputs["startTime"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
            inputs["targetAction"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ScheduledAction.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ScheduledAction resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ScheduledAction(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ScheduledAction.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ScheduledAction.__pulumiType;
    }
}
exports.ScheduledAction = ScheduledAction;
/** @internal */
ScheduledAction.__pulumiType = 'aws-native:redshift:ScheduledAction';
