// @pulumi/aws-native/iotwireless/partnerAccount.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartnerAccount = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Create and manage partner account
 *
 * @deprecated PartnerAccount is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class PartnerAccount extends pulumi.CustomResource {
    /**
     * Create a PartnerAccount resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated PartnerAccount is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("PartnerAccount is deprecated: PartnerAccount is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["accountLinked"] = args ? args.accountLinked : undefined;
            inputs["fingerprint"] = args ? args.fingerprint : undefined;
            inputs["partnerAccountId"] = args ? args.partnerAccountId : undefined;
            inputs["partnerType"] = args ? args.partnerType : undefined;
            inputs["sidewalk"] = args ? args.sidewalk : undefined;
            inputs["sidewalkUpdate"] = args ? args.sidewalkUpdate : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["sidewalkResponse"] = undefined /*out*/;
        }
        else {
            inputs["accountLinked"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
            inputs["fingerprint"] = undefined /*out*/;
            inputs["partnerAccountId"] = undefined /*out*/;
            inputs["partnerType"] = undefined /*out*/;
            inputs["sidewalk"] = undefined /*out*/;
            inputs["sidewalkResponse"] = undefined /*out*/;
            inputs["sidewalkUpdate"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(PartnerAccount.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing PartnerAccount resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("PartnerAccount is deprecated: PartnerAccount is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new PartnerAccount(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PartnerAccount.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PartnerAccount.__pulumiType;
    }
}
exports.PartnerAccount = PartnerAccount;
/** @internal */
PartnerAccount.__pulumiType = 'aws-native:iotwireless:PartnerAccount';
