// @pulumi/aws-native/iotwireless/wirelessGateway.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WirelessGateway = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Create and manage wireless gateways, including LoRa gateways.
 */
class WirelessGateway extends pulumi.CustomResource {
    /**
     * Create a WirelessGateway resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.loRaWAN === undefined) && !opts.urn) {
                throw new Error("Missing required property 'loRaWAN'");
            }
            inputs["description"] = args ? args.description : undefined;
            inputs["lastUplinkReceivedAt"] = args ? args.lastUplinkReceivedAt : undefined;
            inputs["loRaWAN"] = args ? args.loRaWAN : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["thingArn"] = args ? args.thingArn : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["thingName"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["lastUplinkReceivedAt"] = undefined /*out*/;
            inputs["loRaWAN"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["thingArn"] = undefined /*out*/;
            inputs["thingName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(WirelessGateway.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WirelessGateway resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WirelessGateway(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WirelessGateway.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WirelessGateway.__pulumiType;
    }
}
exports.WirelessGateway = WirelessGateway;
/** @internal */
WirelessGateway.__pulumiType = 'aws-native:iotwireless:WirelessGateway';
