// @pulumi/aws-native/workspaces/connectionAlias.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectionAlias = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::WorkSpaces::ConnectionAlias
 */
class ConnectionAlias extends pulumi.CustomResource {
    /**
     * Create a ConnectionAlias resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.connectionString === undefined) && !opts.urn) {
                throw new Error("Missing required property 'connectionString'");
            }
            inputs["connectionString"] = args ? args.connectionString : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["aliasId"] = undefined /*out*/;
            inputs["associations"] = undefined /*out*/;
            inputs["connectionAliasState"] = undefined /*out*/;
        }
        else {
            inputs["aliasId"] = undefined /*out*/;
            inputs["associations"] = undefined /*out*/;
            inputs["connectionAliasState"] = undefined /*out*/;
            inputs["connectionString"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ConnectionAlias.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ConnectionAlias resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ConnectionAlias(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ConnectionAlias.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConnectionAlias.__pulumiType;
    }
}
exports.ConnectionAlias = ConnectionAlias;
/** @internal */
ConnectionAlias.__pulumiType = 'aws-native:workspaces:ConnectionAlias';
