// @pulumi/aws-native/budgets/budgetsAction.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BudgetsAction = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An example resource schema demonstrating some basic constructs and validation rules.
 */
class BudgetsAction extends pulumi.CustomResource {
    /**
     * Create a BudgetsAction resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.actionThreshold === undefined) && !opts.urn) {
                throw new Error("Missing required property 'actionThreshold'");
            }
            if ((!args || args.actionType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'actionType'");
            }
            if ((!args || args.budgetName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'budgetName'");
            }
            if ((!args || args.definition === undefined) && !opts.urn) {
                throw new Error("Missing required property 'definition'");
            }
            if ((!args || args.executionRoleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'executionRoleArn'");
            }
            if ((!args || args.notificationType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'notificationType'");
            }
            if ((!args || args.subscribers === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subscribers'");
            }
            inputs["actionThreshold"] = args ? args.actionThreshold : undefined;
            inputs["actionType"] = args ? args.actionType : undefined;
            inputs["approvalModel"] = args ? args.approvalModel : undefined;
            inputs["budgetName"] = args ? args.budgetName : undefined;
            inputs["definition"] = args ? args.definition : undefined;
            inputs["executionRoleArn"] = args ? args.executionRoleArn : undefined;
            inputs["notificationType"] = args ? args.notificationType : undefined;
            inputs["subscribers"] = args ? args.subscribers : undefined;
            inputs["actionId"] = undefined /*out*/;
        }
        else {
            inputs["actionId"] = undefined /*out*/;
            inputs["actionThreshold"] = undefined /*out*/;
            inputs["actionType"] = undefined /*out*/;
            inputs["approvalModel"] = undefined /*out*/;
            inputs["budgetName"] = undefined /*out*/;
            inputs["definition"] = undefined /*out*/;
            inputs["executionRoleArn"] = undefined /*out*/;
            inputs["notificationType"] = undefined /*out*/;
            inputs["subscribers"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(BudgetsAction.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing BudgetsAction resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BudgetsAction(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BudgetsAction.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BudgetsAction.__pulumiType;
    }
}
exports.BudgetsAction = BudgetsAction;
/** @internal */
BudgetsAction.__pulumiType = 'aws-native:budgets:BudgetsAction';
