// @pulumi/aws-native/appsync/resolver.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Resolver = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::AppSync::Resolver
 *
 * @deprecated Resolver is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Resolver extends pulumi.CustomResource {
    /**
     * Create a Resolver resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Resolver is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Resolver is deprecated: Resolver is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.apiId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'apiId'");
            }
            if ((!args || args.fieldName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fieldName'");
            }
            if ((!args || args.typeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'typeName'");
            }
            inputs["apiId"] = args ? args.apiId : undefined;
            inputs["cachingConfig"] = args ? args.cachingConfig : undefined;
            inputs["dataSourceName"] = args ? args.dataSourceName : undefined;
            inputs["fieldName"] = args ? args.fieldName : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["pipelineConfig"] = args ? args.pipelineConfig : undefined;
            inputs["requestMappingTemplate"] = args ? args.requestMappingTemplate : undefined;
            inputs["requestMappingTemplateS3Location"] = args ? args.requestMappingTemplateS3Location : undefined;
            inputs["responseMappingTemplate"] = args ? args.responseMappingTemplate : undefined;
            inputs["responseMappingTemplateS3Location"] = args ? args.responseMappingTemplateS3Location : undefined;
            inputs["syncConfig"] = args ? args.syncConfig : undefined;
            inputs["typeName"] = args ? args.typeName : undefined;
            inputs["resolverArn"] = undefined /*out*/;
        }
        else {
            inputs["apiId"] = undefined /*out*/;
            inputs["cachingConfig"] = undefined /*out*/;
            inputs["dataSourceName"] = undefined /*out*/;
            inputs["fieldName"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["pipelineConfig"] = undefined /*out*/;
            inputs["requestMappingTemplate"] = undefined /*out*/;
            inputs["requestMappingTemplateS3Location"] = undefined /*out*/;
            inputs["resolverArn"] = undefined /*out*/;
            inputs["responseMappingTemplate"] = undefined /*out*/;
            inputs["responseMappingTemplateS3Location"] = undefined /*out*/;
            inputs["syncConfig"] = undefined /*out*/;
            inputs["typeName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Resolver.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Resolver resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Resolver is deprecated: Resolver is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Resolver(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Resolver.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Resolver.__pulumiType;
    }
}
exports.Resolver = Resolver;
/** @internal */
Resolver.__pulumiType = 'aws-native:appsync:Resolver';
