// @pulumi/aws-native/macie/findingsFilter.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FindingsFilter = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Macie FindingsFilter resource schema.
 */
class FindingsFilter extends pulumi.CustomResource {
    /**
     * Create a FindingsFilter resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.findingCriteria === undefined) && !opts.urn) {
                throw new Error("Missing required property 'findingCriteria'");
            }
            inputs["action"] = args ? args.action : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["findingCriteria"] = args ? args.findingCriteria : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["position"] = args ? args.position : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["findingsFilterListItems"] = undefined /*out*/;
        }
        else {
            inputs["action"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["findingCriteria"] = undefined /*out*/;
            inputs["findingsFilterListItems"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["position"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(FindingsFilter.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing FindingsFilter resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FindingsFilter(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FindingsFilter.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FindingsFilter.__pulumiType;
    }
}
exports.FindingsFilter = FindingsFilter;
/** @internal */
FindingsFilter.__pulumiType = 'aws-native:macie:FindingsFilter';
