// @pulumi/aws-native/iam/virtualMFADevice.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMFADevice = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::IAM::VirtualMFADevice
 */
class VirtualMFADevice extends pulumi.CustomResource {
    /**
     * Create a VirtualMFADevice resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.users === undefined) && !opts.urn) {
                throw new Error("Missing required property 'users'");
            }
            inputs["path"] = args ? args.path : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["users"] = args ? args.users : undefined;
            inputs["virtualMfaDeviceName"] = args ? args.virtualMfaDeviceName : undefined;
            inputs["serialNumber"] = undefined /*out*/;
        }
        else {
            inputs["path"] = undefined /*out*/;
            inputs["serialNumber"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["users"] = undefined /*out*/;
            inputs["virtualMfaDeviceName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(VirtualMFADevice.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing VirtualMFADevice resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VirtualMFADevice(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualMFADevice.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualMFADevice.__pulumiType;
    }
}
exports.VirtualMFADevice = VirtualMFADevice;
/** @internal */
VirtualMFADevice.__pulumiType = 'aws-native:iam:VirtualMFADevice';
