// @pulumi/aws-native/sagemaker/image.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Image = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::SageMaker::Image
 */
class Image extends pulumi.CustomResource {
    /**
     * Create a Image resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.imageRoleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'imageRoleArn'");
            }
            inputs["imageDescription"] = args ? args.imageDescription : undefined;
            inputs["imageDisplayName"] = args ? args.imageDisplayName : undefined;
            inputs["imageName"] = args ? args.imageName : undefined;
            inputs["imageRoleArn"] = args ? args.imageRoleArn : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["imageArn"] = undefined /*out*/;
        }
        else {
            inputs["imageArn"] = undefined /*out*/;
            inputs["imageDescription"] = undefined /*out*/;
            inputs["imageDisplayName"] = undefined /*out*/;
            inputs["imageName"] = undefined /*out*/;
            inputs["imageRoleArn"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Image.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Image resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Image(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Image.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Image.__pulumiType;
    }
}
exports.Image = Image;
/** @internal */
Image.__pulumiType = 'aws-native:sagemaker:Image';
