// @pulumi/aws-native/sagemaker/modelExplainabilityJobDefinition.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelExplainabilityJobDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::SageMaker::ModelExplainabilityJobDefinition
 */
class ModelExplainabilityJobDefinition extends pulumi.CustomResource {
    /**
     * Create a ModelExplainabilityJobDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.jobResources === undefined) && !opts.urn) {
                throw new Error("Missing required property 'jobResources'");
            }
            if ((!args || args.modelExplainabilityAppSpecification === undefined) && !opts.urn) {
                throw new Error("Missing required property 'modelExplainabilityAppSpecification'");
            }
            if ((!args || args.modelExplainabilityJobInput === undefined) && !opts.urn) {
                throw new Error("Missing required property 'modelExplainabilityJobInput'");
            }
            if ((!args || args.modelExplainabilityJobOutputConfig === undefined) && !opts.urn) {
                throw new Error("Missing required property 'modelExplainabilityJobOutputConfig'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            inputs["jobDefinitionName"] = args ? args.jobDefinitionName : undefined;
            inputs["jobResources"] = args ? args.jobResources : undefined;
            inputs["modelExplainabilityAppSpecification"] = args ? args.modelExplainabilityAppSpecification : undefined;
            inputs["modelExplainabilityBaselineConfig"] = args ? args.modelExplainabilityBaselineConfig : undefined;
            inputs["modelExplainabilityJobInput"] = args ? args.modelExplainabilityJobInput : undefined;
            inputs["modelExplainabilityJobOutputConfig"] = args ? args.modelExplainabilityJobOutputConfig : undefined;
            inputs["networkConfig"] = args ? args.networkConfig : undefined;
            inputs["roleArn"] = args ? args.roleArn : undefined;
            inputs["stoppingCondition"] = args ? args.stoppingCondition : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["creationTime"] = undefined /*out*/;
            inputs["jobDefinitionArn"] = undefined /*out*/;
        }
        else {
            inputs["creationTime"] = undefined /*out*/;
            inputs["jobDefinitionArn"] = undefined /*out*/;
            inputs["jobDefinitionName"] = undefined /*out*/;
            inputs["jobResources"] = undefined /*out*/;
            inputs["modelExplainabilityAppSpecification"] = undefined /*out*/;
            inputs["modelExplainabilityBaselineConfig"] = undefined /*out*/;
            inputs["modelExplainabilityJobInput"] = undefined /*out*/;
            inputs["modelExplainabilityJobOutputConfig"] = undefined /*out*/;
            inputs["networkConfig"] = undefined /*out*/;
            inputs["roleArn"] = undefined /*out*/;
            inputs["stoppingCondition"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ModelExplainabilityJobDefinition.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ModelExplainabilityJobDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ModelExplainabilityJobDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ModelExplainabilityJobDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ModelExplainabilityJobDefinition.__pulumiType;
    }
}
exports.ModelExplainabilityJobDefinition = ModelExplainabilityJobDefinition;
/** @internal */
ModelExplainabilityJobDefinition.__pulumiType = 'aws-native:sagemaker:ModelExplainabilityJobDefinition';
