// @pulumi/aws-native/ec2/capacityReservationFleet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapacityReservationFleet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::EC2::CapacityReservationFleet
 */
class CapacityReservationFleet extends pulumi.CustomResource {
    /**
     * Create a CapacityReservationFleet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["allocationStrategy"] = args ? args.allocationStrategy : undefined;
            inputs["endDate"] = args ? args.endDate : undefined;
            inputs["instanceMatchCriteria"] = args ? args.instanceMatchCriteria : undefined;
            inputs["instanceTypeSpecifications"] = args ? args.instanceTypeSpecifications : undefined;
            inputs["noRemoveEndDate"] = args ? args.noRemoveEndDate : undefined;
            inputs["removeEndDate"] = args ? args.removeEndDate : undefined;
            inputs["tagSpecifications"] = args ? args.tagSpecifications : undefined;
            inputs["tenancy"] = args ? args.tenancy : undefined;
            inputs["totalTargetCapacity"] = args ? args.totalTargetCapacity : undefined;
            inputs["capacityReservationFleetId"] = undefined /*out*/;
        }
        else {
            inputs["allocationStrategy"] = undefined /*out*/;
            inputs["capacityReservationFleetId"] = undefined /*out*/;
            inputs["endDate"] = undefined /*out*/;
            inputs["instanceMatchCriteria"] = undefined /*out*/;
            inputs["instanceTypeSpecifications"] = undefined /*out*/;
            inputs["noRemoveEndDate"] = undefined /*out*/;
            inputs["removeEndDate"] = undefined /*out*/;
            inputs["tagSpecifications"] = undefined /*out*/;
            inputs["tenancy"] = undefined /*out*/;
            inputs["totalTargetCapacity"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(CapacityReservationFleet.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing CapacityReservationFleet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CapacityReservationFleet(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CapacityReservationFleet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CapacityReservationFleet.__pulumiType;
    }
}
exports.CapacityReservationFleet = CapacityReservationFleet;
/** @internal */
CapacityReservationFleet.__pulumiType = 'aws-native:ec2:CapacityReservationFleet';
