// @pulumi/aws-native/ec2/enclaveCertificateIamRoleAssociation.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnclaveCertificateIamRoleAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Associates an AWS Identity and Access Management (IAM) role with an AWS Certificate Manager (ACM) certificate. This association is based on Amazon Resource Names and it enables the certificate to be used by the ACM for Nitro Enclaves application inside an enclave.
 */
class EnclaveCertificateIamRoleAssociation extends pulumi.CustomResource {
    /**
     * Create a EnclaveCertificateIamRoleAssociation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.certificateArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'certificateArn'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            inputs["certificateArn"] = args ? args.certificateArn : undefined;
            inputs["roleArn"] = args ? args.roleArn : undefined;
            inputs["certificateS3BucketName"] = undefined /*out*/;
            inputs["certificateS3ObjectKey"] = undefined /*out*/;
            inputs["encryptionKmsKeyId"] = undefined /*out*/;
        }
        else {
            inputs["certificateArn"] = undefined /*out*/;
            inputs["certificateS3BucketName"] = undefined /*out*/;
            inputs["certificateS3ObjectKey"] = undefined /*out*/;
            inputs["encryptionKmsKeyId"] = undefined /*out*/;
            inputs["roleArn"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(EnclaveCertificateIamRoleAssociation.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing EnclaveCertificateIamRoleAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EnclaveCertificateIamRoleAssociation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EnclaveCertificateIamRoleAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EnclaveCertificateIamRoleAssociation.__pulumiType;
    }
}
exports.EnclaveCertificateIamRoleAssociation = EnclaveCertificateIamRoleAssociation;
/** @internal */
EnclaveCertificateIamRoleAssociation.__pulumiType = 'aws-native:ec2:EnclaveCertificateIamRoleAssociation';
