// @pulumi/aws-native/ec2/localGatewayRouteTableVPCAssociation.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocalGatewayRouteTableVPCAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Describes an association between a local gateway route table and a VPC.
 */
class LocalGatewayRouteTableVPCAssociation extends pulumi.CustomResource {
    /**
     * Create a LocalGatewayRouteTableVPCAssociation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.localGatewayRouteTableId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'localGatewayRouteTableId'");
            }
            if ((!args || args.vpcId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vpcId'");
            }
            inputs["localGatewayRouteTableId"] = args ? args.localGatewayRouteTableId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["vpcId"] = args ? args.vpcId : undefined;
            inputs["localGatewayId"] = undefined /*out*/;
            inputs["localGatewayRouteTableVpcAssociationId"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
        }
        else {
            inputs["localGatewayId"] = undefined /*out*/;
            inputs["localGatewayRouteTableId"] = undefined /*out*/;
            inputs["localGatewayRouteTableVpcAssociationId"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["vpcId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(LocalGatewayRouteTableVPCAssociation.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing LocalGatewayRouteTableVPCAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LocalGatewayRouteTableVPCAssociation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LocalGatewayRouteTableVPCAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LocalGatewayRouteTableVPCAssociation.__pulumiType;
    }
}
exports.LocalGatewayRouteTableVPCAssociation = LocalGatewayRouteTableVPCAssociation;
/** @internal */
LocalGatewayRouteTableVPCAssociation.__pulumiType = 'aws-native:ec2:LocalGatewayRouteTableVPCAssociation';
