// @pulumi/aws-native/ec2/localGatewayRoute.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocalGatewayRoute = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Describes a route for a local gateway route table.
 */
class LocalGatewayRoute extends pulumi.CustomResource {
    /**
     * Create a LocalGatewayRoute resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.destinationCidrBlock === undefined) && !opts.urn) {
                throw new Error("Missing required property 'destinationCidrBlock'");
            }
            if ((!args || args.localGatewayRouteTableId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'localGatewayRouteTableId'");
            }
            if ((!args || args.localGatewayVirtualInterfaceGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'localGatewayVirtualInterfaceGroupId'");
            }
            inputs["destinationCidrBlock"] = args ? args.destinationCidrBlock : undefined;
            inputs["localGatewayRouteTableId"] = args ? args.localGatewayRouteTableId : undefined;
            inputs["localGatewayVirtualInterfaceGroupId"] = args ? args.localGatewayVirtualInterfaceGroupId : undefined;
            inputs["state"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["destinationCidrBlock"] = undefined /*out*/;
            inputs["localGatewayRouteTableId"] = undefined /*out*/;
            inputs["localGatewayVirtualInterfaceGroupId"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(LocalGatewayRoute.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing LocalGatewayRoute resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LocalGatewayRoute(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LocalGatewayRoute.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LocalGatewayRoute.__pulumiType;
    }
}
exports.LocalGatewayRoute = LocalGatewayRoute;
/** @internal */
LocalGatewayRoute.__pulumiType = 'aws-native:ec2:LocalGatewayRoute';
