// @pulumi/aws-native/ec2/networkInsightsAnalysis.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkInsightsAnalysis = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::EC2::NetworkInsightsAnalysis
 */
class NetworkInsightsAnalysis extends pulumi.CustomResource {
    /**
     * Create a NetworkInsightsAnalysis resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.networkInsightsPathId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkInsightsPathId'");
            }
            inputs["filterInArns"] = args ? args.filterInArns : undefined;
            inputs["networkInsightsPathId"] = args ? args.networkInsightsPathId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["alternatePathHints"] = undefined /*out*/;
            inputs["explanations"] = undefined /*out*/;
            inputs["forwardPathComponents"] = undefined /*out*/;
            inputs["networkInsightsAnalysisArn"] = undefined /*out*/;
            inputs["networkInsightsAnalysisId"] = undefined /*out*/;
            inputs["networkPathFound"] = undefined /*out*/;
            inputs["returnPathComponents"] = undefined /*out*/;
            inputs["startDate"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["statusMessage"] = undefined /*out*/;
        }
        else {
            inputs["alternatePathHints"] = undefined /*out*/;
            inputs["explanations"] = undefined /*out*/;
            inputs["filterInArns"] = undefined /*out*/;
            inputs["forwardPathComponents"] = undefined /*out*/;
            inputs["networkInsightsAnalysisArn"] = undefined /*out*/;
            inputs["networkInsightsAnalysisId"] = undefined /*out*/;
            inputs["networkInsightsPathId"] = undefined /*out*/;
            inputs["networkPathFound"] = undefined /*out*/;
            inputs["returnPathComponents"] = undefined /*out*/;
            inputs["startDate"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["statusMessage"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(NetworkInsightsAnalysis.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing NetworkInsightsAnalysis resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkInsightsAnalysis(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkInsightsAnalysis.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkInsightsAnalysis.__pulumiType;
    }
}
exports.NetworkInsightsAnalysis = NetworkInsightsAnalysis;
/** @internal */
NetworkInsightsAnalysis.__pulumiType = 'aws-native:ec2:NetworkInsightsAnalysis';
