// @pulumi/aws-native/ec2/networkInterface.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkInterface = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The AWS::EC2::NetworkInterface resource creates network interface
 */
class NetworkInterface extends pulumi.CustomResource {
    /**
     * Create a NetworkInterface resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.subnetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subnetId'");
            }
            inputs["description"] = args ? args.description : undefined;
            inputs["groupSet"] = args ? args.groupSet : undefined;
            inputs["interfaceType"] = args ? args.interfaceType : undefined;
            inputs["ipv6AddressCount"] = args ? args.ipv6AddressCount : undefined;
            inputs["ipv6Addresses"] = args ? args.ipv6Addresses : undefined;
            inputs["privateIpAddress"] = args ? args.privateIpAddress : undefined;
            inputs["privateIpAddresses"] = args ? args.privateIpAddresses : undefined;
            inputs["secondaryPrivateIpAddressCount"] = args ? args.secondaryPrivateIpAddressCount : undefined;
            inputs["sourceDestCheck"] = args ? args.sourceDestCheck : undefined;
            inputs["subnetId"] = args ? args.subnetId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["primaryPrivateIpAddress"] = undefined /*out*/;
            inputs["secondaryPrivateIpAddresses"] = undefined /*out*/;
        }
        else {
            inputs["description"] = undefined /*out*/;
            inputs["groupSet"] = undefined /*out*/;
            inputs["interfaceType"] = undefined /*out*/;
            inputs["ipv6AddressCount"] = undefined /*out*/;
            inputs["ipv6Addresses"] = undefined /*out*/;
            inputs["primaryPrivateIpAddress"] = undefined /*out*/;
            inputs["privateIpAddress"] = undefined /*out*/;
            inputs["privateIpAddresses"] = undefined /*out*/;
            inputs["secondaryPrivateIpAddressCount"] = undefined /*out*/;
            inputs["secondaryPrivateIpAddresses"] = undefined /*out*/;
            inputs["sourceDestCheck"] = undefined /*out*/;
            inputs["subnetId"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(NetworkInterface.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing NetworkInterface resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkInterface(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkInterface.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkInterface.__pulumiType;
    }
}
exports.NetworkInterface = NetworkInterface;
/** @internal */
NetworkInterface.__pulumiType = 'aws-native:ec2:NetworkInterface';
