// @pulumi/aws-native/ec2/securityGroupEgress.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityGroupEgress = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::EC2::SecurityGroupEgress
 *
 * @deprecated SecurityGroupEgress is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class SecurityGroupEgress extends pulumi.CustomResource {
    /**
     * Create a SecurityGroupEgress resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated SecurityGroupEgress is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("SecurityGroupEgress is deprecated: SecurityGroupEgress is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.groupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'groupId'");
            }
            if ((!args || args.ipProtocol === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ipProtocol'");
            }
            inputs["cidrIp"] = args ? args.cidrIp : undefined;
            inputs["cidrIpv6"] = args ? args.cidrIpv6 : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["destinationPrefixListId"] = args ? args.destinationPrefixListId : undefined;
            inputs["destinationSecurityGroupId"] = args ? args.destinationSecurityGroupId : undefined;
            inputs["fromPort"] = args ? args.fromPort : undefined;
            inputs["groupId"] = args ? args.groupId : undefined;
            inputs["ipProtocol"] = args ? args.ipProtocol : undefined;
            inputs["toPort"] = args ? args.toPort : undefined;
        }
        else {
            inputs["cidrIp"] = undefined /*out*/;
            inputs["cidrIpv6"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["destinationPrefixListId"] = undefined /*out*/;
            inputs["destinationSecurityGroupId"] = undefined /*out*/;
            inputs["fromPort"] = undefined /*out*/;
            inputs["groupId"] = undefined /*out*/;
            inputs["ipProtocol"] = undefined /*out*/;
            inputs["toPort"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(SecurityGroupEgress.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SecurityGroupEgress resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SecurityGroupEgress is deprecated: SecurityGroupEgress is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new SecurityGroupEgress(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SecurityGroupEgress.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecurityGroupEgress.__pulumiType;
    }
}
exports.SecurityGroupEgress = SecurityGroupEgress;
/** @internal */
SecurityGroupEgress.__pulumiType = 'aws-native:ec2:SecurityGroupEgress';
