// @pulumi/aws-native/ec2/transitGatewayMulticastGroupMember.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransitGatewayMulticastGroupMember = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The AWS::EC2::TransitGatewayMulticastGroupMember registers and deregisters members and sources (network interfaces) with the transit gateway multicast group
 */
class TransitGatewayMulticastGroupMember extends pulumi.CustomResource {
    /**
     * Create a TransitGatewayMulticastGroupMember resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.groupIpAddress === undefined) && !opts.urn) {
                throw new Error("Missing required property 'groupIpAddress'");
            }
            if ((!args || args.networkInterfaceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkInterfaceId'");
            }
            if ((!args || args.transitGatewayMulticastDomainId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'transitGatewayMulticastDomainId'");
            }
            inputs["groupIpAddress"] = args ? args.groupIpAddress : undefined;
            inputs["networkInterfaceId"] = args ? args.networkInterfaceId : undefined;
            inputs["transitGatewayMulticastDomainId"] = args ? args.transitGatewayMulticastDomainId : undefined;
            inputs["groupMember"] = undefined /*out*/;
            inputs["groupSource"] = undefined /*out*/;
            inputs["memberType"] = undefined /*out*/;
            inputs["resourceId"] = undefined /*out*/;
            inputs["resourceType"] = undefined /*out*/;
            inputs["sourceType"] = undefined /*out*/;
            inputs["subnetId"] = undefined /*out*/;
            inputs["transitGatewayAttachmentId"] = undefined /*out*/;
        }
        else {
            inputs["groupIpAddress"] = undefined /*out*/;
            inputs["groupMember"] = undefined /*out*/;
            inputs["groupSource"] = undefined /*out*/;
            inputs["memberType"] = undefined /*out*/;
            inputs["networkInterfaceId"] = undefined /*out*/;
            inputs["resourceId"] = undefined /*out*/;
            inputs["resourceType"] = undefined /*out*/;
            inputs["sourceType"] = undefined /*out*/;
            inputs["subnetId"] = undefined /*out*/;
            inputs["transitGatewayAttachmentId"] = undefined /*out*/;
            inputs["transitGatewayMulticastDomainId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(TransitGatewayMulticastGroupMember.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing TransitGatewayMulticastGroupMember resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TransitGatewayMulticastGroupMember(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TransitGatewayMulticastGroupMember.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TransitGatewayMulticastGroupMember.__pulumiType;
    }
}
exports.TransitGatewayMulticastGroupMember = TransitGatewayMulticastGroupMember;
/** @internal */
TransitGatewayMulticastGroupMember.__pulumiType = 'aws-native:ec2:TransitGatewayMulticastGroupMember';
