// @pulumi/aws-native/ec2/transitGateway.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransitGateway = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::EC2::TransitGateway
 */
class TransitGateway extends pulumi.CustomResource {
    /**
     * Create a TransitGateway resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["amazonSideAsn"] = args ? args.amazonSideAsn : undefined;
            inputs["associationDefaultRouteTableId"] = args ? args.associationDefaultRouteTableId : undefined;
            inputs["autoAcceptSharedAttachments"] = args ? args.autoAcceptSharedAttachments : undefined;
            inputs["defaultRouteTableAssociation"] = args ? args.defaultRouteTableAssociation : undefined;
            inputs["defaultRouteTablePropagation"] = args ? args.defaultRouteTablePropagation : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["dnsSupport"] = args ? args.dnsSupport : undefined;
            inputs["multicastSupport"] = args ? args.multicastSupport : undefined;
            inputs["propagationDefaultRouteTableId"] = args ? args.propagationDefaultRouteTableId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["transitGatewayCidrBlocks"] = args ? args.transitGatewayCidrBlocks : undefined;
            inputs["vpnEcmpSupport"] = args ? args.vpnEcmpSupport : undefined;
        }
        else {
            inputs["amazonSideAsn"] = undefined /*out*/;
            inputs["associationDefaultRouteTableId"] = undefined /*out*/;
            inputs["autoAcceptSharedAttachments"] = undefined /*out*/;
            inputs["defaultRouteTableAssociation"] = undefined /*out*/;
            inputs["defaultRouteTablePropagation"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["dnsSupport"] = undefined /*out*/;
            inputs["multicastSupport"] = undefined /*out*/;
            inputs["propagationDefaultRouteTableId"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["transitGatewayCidrBlocks"] = undefined /*out*/;
            inputs["vpnEcmpSupport"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(TransitGateway.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing TransitGateway resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TransitGateway(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TransitGateway.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TransitGateway.__pulumiType;
    }
}
exports.TransitGateway = TransitGateway;
/** @internal */
TransitGateway.__pulumiType = 'aws-native:ec2:TransitGateway';
