// @pulumi/aws-native/ec2/vpc.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VPC = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::EC2::VPC
 */
class VPC extends pulumi.CustomResource {
    /**
     * Create a VPC resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.cidrBlock === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cidrBlock'");
            }
            inputs["cidrBlock"] = args ? args.cidrBlock : undefined;
            inputs["enableDnsHostnames"] = args ? args.enableDnsHostnames : undefined;
            inputs["enableDnsSupport"] = args ? args.enableDnsSupport : undefined;
            inputs["instanceTenancy"] = args ? args.instanceTenancy : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["cidrBlockAssociations"] = undefined /*out*/;
            inputs["defaultNetworkAcl"] = undefined /*out*/;
            inputs["defaultSecurityGroup"] = undefined /*out*/;
            inputs["ipv6CidrBlocks"] = undefined /*out*/;
            inputs["vpcId"] = undefined /*out*/;
        }
        else {
            inputs["cidrBlock"] = undefined /*out*/;
            inputs["cidrBlockAssociations"] = undefined /*out*/;
            inputs["defaultNetworkAcl"] = undefined /*out*/;
            inputs["defaultSecurityGroup"] = undefined /*out*/;
            inputs["enableDnsHostnames"] = undefined /*out*/;
            inputs["enableDnsSupport"] = undefined /*out*/;
            inputs["instanceTenancy"] = undefined /*out*/;
            inputs["ipv6CidrBlocks"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["vpcId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(VPC.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing VPC resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VPC(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VPC.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VPC.__pulumiType;
    }
}
exports.VPC = VPC;
/** @internal */
VPC.__pulumiType = 'aws-native:ec2:VPC';
