// @pulumi/aws-native/wisdom/assistantAssociation.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssistantAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::Wisdom::AssistantAssociation Resource Type
 */
class AssistantAssociation extends pulumi.CustomResource {
    /**
     * Create a AssistantAssociation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.assistantId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'assistantId'");
            }
            if ((!args || args.association === undefined) && !opts.urn) {
                throw new Error("Missing required property 'association'");
            }
            if ((!args || args.associationType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'associationType'");
            }
            inputs["assistantId"] = args ? args.assistantId : undefined;
            inputs["association"] = args ? args.association : undefined;
            inputs["associationType"] = args ? args.associationType : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["assistantArn"] = undefined /*out*/;
            inputs["assistantAssociationArn"] = undefined /*out*/;
            inputs["assistantAssociationId"] = undefined /*out*/;
        }
        else {
            inputs["assistantArn"] = undefined /*out*/;
            inputs["assistantAssociationArn"] = undefined /*out*/;
            inputs["assistantAssociationId"] = undefined /*out*/;
            inputs["assistantId"] = undefined /*out*/;
            inputs["association"] = undefined /*out*/;
            inputs["associationType"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(AssistantAssociation.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing AssistantAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AssistantAssociation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AssistantAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AssistantAssociation.__pulumiType;
    }
}
exports.AssistantAssociation = AssistantAssociation;
/** @internal */
AssistantAssociation.__pulumiType = 'aws-native:wisdom:AssistantAssociation';
