// @pulumi/aws-native/autoscaling/launchConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LaunchConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The AWS::AutoScaling::LaunchConfiguration resource specifies the launch configuration that can be used by an Auto Scaling group to configure Amazon EC2 instances.
 */
class LaunchConfiguration extends pulumi.CustomResource {
    /**
     * Create a LaunchConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.imageId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'imageId'");
            }
            if ((!args || args.instanceType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'instanceType'");
            }
            inputs["associatePublicIpAddress"] = args ? args.associatePublicIpAddress : undefined;
            inputs["blockDeviceMappings"] = args ? args.blockDeviceMappings : undefined;
            inputs["classicLinkVPCId"] = args ? args.classicLinkVPCId : undefined;
            inputs["classicLinkVPCSecurityGroups"] = args ? args.classicLinkVPCSecurityGroups : undefined;
            inputs["ebsOptimized"] = args ? args.ebsOptimized : undefined;
            inputs["iamInstanceProfile"] = args ? args.iamInstanceProfile : undefined;
            inputs["imageId"] = args ? args.imageId : undefined;
            inputs["instanceId"] = args ? args.instanceId : undefined;
            inputs["instanceMonitoring"] = args ? args.instanceMonitoring : undefined;
            inputs["instanceType"] = args ? args.instanceType : undefined;
            inputs["kernelId"] = args ? args.kernelId : undefined;
            inputs["keyName"] = args ? args.keyName : undefined;
            inputs["launchConfigurationName"] = args ? args.launchConfigurationName : undefined;
            inputs["metadataOptions"] = args ? args.metadataOptions : undefined;
            inputs["placementTenancy"] = args ? args.placementTenancy : undefined;
            inputs["ramDiskId"] = args ? args.ramDiskId : undefined;
            inputs["securityGroups"] = args ? args.securityGroups : undefined;
            inputs["spotPrice"] = args ? args.spotPrice : undefined;
            inputs["userData"] = args ? args.userData : undefined;
        }
        else {
            inputs["associatePublicIpAddress"] = undefined /*out*/;
            inputs["blockDeviceMappings"] = undefined /*out*/;
            inputs["classicLinkVPCId"] = undefined /*out*/;
            inputs["classicLinkVPCSecurityGroups"] = undefined /*out*/;
            inputs["ebsOptimized"] = undefined /*out*/;
            inputs["iamInstanceProfile"] = undefined /*out*/;
            inputs["imageId"] = undefined /*out*/;
            inputs["instanceId"] = undefined /*out*/;
            inputs["instanceMonitoring"] = undefined /*out*/;
            inputs["instanceType"] = undefined /*out*/;
            inputs["kernelId"] = undefined /*out*/;
            inputs["keyName"] = undefined /*out*/;
            inputs["launchConfigurationName"] = undefined /*out*/;
            inputs["metadataOptions"] = undefined /*out*/;
            inputs["placementTenancy"] = undefined /*out*/;
            inputs["ramDiskId"] = undefined /*out*/;
            inputs["securityGroups"] = undefined /*out*/;
            inputs["spotPrice"] = undefined /*out*/;
            inputs["userData"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(LaunchConfiguration.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing LaunchConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LaunchConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LaunchConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LaunchConfiguration.__pulumiType;
    }
}
exports.LaunchConfiguration = LaunchConfiguration;
/** @internal */
LaunchConfiguration.__pulumiType = 'aws-native:autoscaling:LaunchConfiguration';
