// @pulumi/aws-native/lookoutequipment/inferenceScheduler.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.InferenceScheduler = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for LookoutEquipment InferenceScheduler.
 *
 * @deprecated InferenceScheduler is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class InferenceScheduler extends pulumi.CustomResource {
    /**
     * Create a InferenceScheduler resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated InferenceScheduler is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("InferenceScheduler is deprecated: InferenceScheduler is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dataInputConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataInputConfiguration'");
            }
            if ((!args || args.dataOutputConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataOutputConfiguration'");
            }
            if ((!args || args.dataUploadFrequency === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataUploadFrequency'");
            }
            if ((!args || args.modelName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'modelName'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            inputs["dataDelayOffsetInMinutes"] = args ? args.dataDelayOffsetInMinutes : undefined;
            inputs["dataInputConfiguration"] = args ? args.dataInputConfiguration : undefined;
            inputs["dataOutputConfiguration"] = args ? args.dataOutputConfiguration : undefined;
            inputs["dataUploadFrequency"] = args ? args.dataUploadFrequency : undefined;
            inputs["inferenceSchedulerName"] = args ? args.inferenceSchedulerName : undefined;
            inputs["modelName"] = args ? args.modelName : undefined;
            inputs["roleArn"] = args ? args.roleArn : undefined;
            inputs["serverSideKmsKeyId"] = args ? args.serverSideKmsKeyId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["inferenceSchedulerArn"] = undefined /*out*/;
        }
        else {
            inputs["dataDelayOffsetInMinutes"] = undefined /*out*/;
            inputs["dataInputConfiguration"] = undefined /*out*/;
            inputs["dataOutputConfiguration"] = undefined /*out*/;
            inputs["dataUploadFrequency"] = undefined /*out*/;
            inputs["inferenceSchedulerArn"] = undefined /*out*/;
            inputs["inferenceSchedulerName"] = undefined /*out*/;
            inputs["modelName"] = undefined /*out*/;
            inputs["roleArn"] = undefined /*out*/;
            inputs["serverSideKmsKeyId"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(InferenceScheduler.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing InferenceScheduler resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("InferenceScheduler is deprecated: InferenceScheduler is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new InferenceScheduler(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of InferenceScheduler.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === InferenceScheduler.__pulumiType;
    }
}
exports.InferenceScheduler = InferenceScheduler;
/** @internal */
InferenceScheduler.__pulumiType = 'aws-native:lookoutequipment:InferenceScheduler';
