// @pulumi/aws-native/rds/dbproxyTargetGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DBProxyTargetGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::RDS::DBProxyTargetGroup
 */
class DBProxyTargetGroup extends pulumi.CustomResource {
    /**
     * Create a DBProxyTargetGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dBProxyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dBProxyName'");
            }
            if ((!args || args.targetGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetGroupName'");
            }
            inputs["connectionPoolConfigurationInfo"] = args ? args.connectionPoolConfigurationInfo : undefined;
            inputs["dBClusterIdentifiers"] = args ? args.dBClusterIdentifiers : undefined;
            inputs["dBInstanceIdentifiers"] = args ? args.dBInstanceIdentifiers : undefined;
            inputs["dBProxyName"] = args ? args.dBProxyName : undefined;
            inputs["targetGroupName"] = args ? args.targetGroupName : undefined;
            inputs["targetGroupArn"] = undefined /*out*/;
        }
        else {
            inputs["connectionPoolConfigurationInfo"] = undefined /*out*/;
            inputs["dBClusterIdentifiers"] = undefined /*out*/;
            inputs["dBInstanceIdentifiers"] = undefined /*out*/;
            inputs["dBProxyName"] = undefined /*out*/;
            inputs["targetGroupArn"] = undefined /*out*/;
            inputs["targetGroupName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(DBProxyTargetGroup.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DBProxyTargetGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DBProxyTargetGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DBProxyTargetGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DBProxyTargetGroup.__pulumiType;
    }
}
exports.DBProxyTargetGroup = DBProxyTargetGroup;
/** @internal */
DBProxyTargetGroup.__pulumiType = 'aws-native:rds:DBProxyTargetGroup';
