// @pulumi/aws-native/rds/dbproxyEndpoint.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DBProxyEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::RDS::DBProxyEndpoint.
 */
class DBProxyEndpoint extends pulumi.CustomResource {
    /**
     * Create a DBProxyEndpoint resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dBProxyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dBProxyName'");
            }
            if ((!args || args.vpcSubnetIds === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vpcSubnetIds'");
            }
            inputs["dBProxyEndpointName"] = args ? args.dBProxyEndpointName : undefined;
            inputs["dBProxyName"] = args ? args.dBProxyName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["targetRole"] = args ? args.targetRole : undefined;
            inputs["vpcSecurityGroupIds"] = args ? args.vpcSecurityGroupIds : undefined;
            inputs["vpcSubnetIds"] = args ? args.vpcSubnetIds : undefined;
            inputs["dBProxyEndpointArn"] = undefined /*out*/;
            inputs["endpoint"] = undefined /*out*/;
            inputs["isDefault"] = undefined /*out*/;
            inputs["vpcId"] = undefined /*out*/;
        }
        else {
            inputs["dBProxyEndpointArn"] = undefined /*out*/;
            inputs["dBProxyEndpointName"] = undefined /*out*/;
            inputs["dBProxyName"] = undefined /*out*/;
            inputs["endpoint"] = undefined /*out*/;
            inputs["isDefault"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["targetRole"] = undefined /*out*/;
            inputs["vpcId"] = undefined /*out*/;
            inputs["vpcSecurityGroupIds"] = undefined /*out*/;
            inputs["vpcSubnetIds"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(DBProxyEndpoint.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DBProxyEndpoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DBProxyEndpoint(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DBProxyEndpoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DBProxyEndpoint.__pulumiType;
    }
}
exports.DBProxyEndpoint = DBProxyEndpoint;
/** @internal */
DBProxyEndpoint.__pulumiType = 'aws-native:rds:DBProxyEndpoint';
