// @pulumi/aws-native/rds/dbproxy.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DBProxy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::RDS::DBProxy
 */
class DBProxy extends pulumi.CustomResource {
    /**
     * Create a DBProxy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.auth === undefined) && !opts.urn) {
                throw new Error("Missing required property 'auth'");
            }
            if ((!args || args.engineFamily === undefined) && !opts.urn) {
                throw new Error("Missing required property 'engineFamily'");
            }
            if ((!args || args.roleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            if ((!args || args.vpcSubnetIds === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vpcSubnetIds'");
            }
            inputs["auth"] = args ? args.auth : undefined;
            inputs["dBProxyName"] = args ? args.dBProxyName : undefined;
            inputs["debugLogging"] = args ? args.debugLogging : undefined;
            inputs["engineFamily"] = args ? args.engineFamily : undefined;
            inputs["idleClientTimeout"] = args ? args.idleClientTimeout : undefined;
            inputs["requireTLS"] = args ? args.requireTLS : undefined;
            inputs["roleArn"] = args ? args.roleArn : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["vpcSecurityGroupIds"] = args ? args.vpcSecurityGroupIds : undefined;
            inputs["vpcSubnetIds"] = args ? args.vpcSubnetIds : undefined;
            inputs["dBProxyArn"] = undefined /*out*/;
            inputs["endpoint"] = undefined /*out*/;
            inputs["vpcId"] = undefined /*out*/;
        }
        else {
            inputs["auth"] = undefined /*out*/;
            inputs["dBProxyArn"] = undefined /*out*/;
            inputs["dBProxyName"] = undefined /*out*/;
            inputs["debugLogging"] = undefined /*out*/;
            inputs["endpoint"] = undefined /*out*/;
            inputs["engineFamily"] = undefined /*out*/;
            inputs["idleClientTimeout"] = undefined /*out*/;
            inputs["requireTLS"] = undefined /*out*/;
            inputs["roleArn"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["vpcId"] = undefined /*out*/;
            inputs["vpcSecurityGroupIds"] = undefined /*out*/;
            inputs["vpcSubnetIds"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(DBProxy.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DBProxy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DBProxy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DBProxy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DBProxy.__pulumiType;
    }
}
exports.DBProxy = DBProxy;
/** @internal */
DBProxy.__pulumiType = 'aws-native:rds:DBProxy';
