// @pulumi/aws-native/rds/optionGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OptionGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::RDS::OptionGroup
 *
 * @deprecated OptionGroup is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class OptionGroup extends pulumi.CustomResource {
    /**
     * Create a OptionGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated OptionGroup is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("OptionGroup is deprecated: OptionGroup is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.engineName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'engineName'");
            }
            if ((!args || args.majorEngineVersion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'majorEngineVersion'");
            }
            if ((!args || args.optionConfigurations === undefined) && !opts.urn) {
                throw new Error("Missing required property 'optionConfigurations'");
            }
            if ((!args || args.optionGroupDescription === undefined) && !opts.urn) {
                throw new Error("Missing required property 'optionGroupDescription'");
            }
            inputs["engineName"] = args ? args.engineName : undefined;
            inputs["majorEngineVersion"] = args ? args.majorEngineVersion : undefined;
            inputs["optionConfigurations"] = args ? args.optionConfigurations : undefined;
            inputs["optionGroupDescription"] = args ? args.optionGroupDescription : undefined;
            inputs["tags"] = args ? args.tags : undefined;
        }
        else {
            inputs["engineName"] = undefined /*out*/;
            inputs["majorEngineVersion"] = undefined /*out*/;
            inputs["optionConfigurations"] = undefined /*out*/;
            inputs["optionGroupDescription"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(OptionGroup.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing OptionGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("OptionGroup is deprecated: OptionGroup is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new OptionGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of OptionGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OptionGroup.__pulumiType;
    }
}
exports.OptionGroup = OptionGroup;
/** @internal */
OptionGroup.__pulumiType = 'aws-native:rds:OptionGroup';
