// @pulumi/aws-native/datasync/locationObjectStorage.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationObjectStorage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource schema for AWS::DataSync::LocationObjectStorage.
 */
class LocationObjectStorage extends pulumi.CustomResource {
    /**
     * Create a LocationObjectStorage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.agentArns === undefined) && !opts.urn) {
                throw new Error("Missing required property 'agentArns'");
            }
            if ((!args || args.bucketName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'bucketName'");
            }
            if ((!args || args.serverHostname === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverHostname'");
            }
            inputs["accessKey"] = args ? args.accessKey : undefined;
            inputs["agentArns"] = args ? args.agentArns : undefined;
            inputs["bucketName"] = args ? args.bucketName : undefined;
            inputs["secretKey"] = args ? args.secretKey : undefined;
            inputs["serverHostname"] = args ? args.serverHostname : undefined;
            inputs["serverPort"] = args ? args.serverPort : undefined;
            inputs["serverProtocol"] = args ? args.serverProtocol : undefined;
            inputs["subdirectory"] = args ? args.subdirectory : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["locationArn"] = undefined /*out*/;
            inputs["locationUri"] = undefined /*out*/;
        }
        else {
            inputs["accessKey"] = undefined /*out*/;
            inputs["agentArns"] = undefined /*out*/;
            inputs["bucketName"] = undefined /*out*/;
            inputs["locationArn"] = undefined /*out*/;
            inputs["locationUri"] = undefined /*out*/;
            inputs["secretKey"] = undefined /*out*/;
            inputs["serverHostname"] = undefined /*out*/;
            inputs["serverPort"] = undefined /*out*/;
            inputs["serverProtocol"] = undefined /*out*/;
            inputs["subdirectory"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(LocationObjectStorage.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing LocationObjectStorage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LocationObjectStorage(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LocationObjectStorage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LocationObjectStorage.__pulumiType;
    }
}
exports.LocationObjectStorage = LocationObjectStorage;
/** @internal */
LocationObjectStorage.__pulumiType = 'aws-native:datasync:LocationObjectStorage';
