// @pulumi/aws-native/opsworks/userProfile.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::OpsWorks::UserProfile
 *
 * @deprecated UserProfile is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class UserProfile extends pulumi.CustomResource {
    /**
     * Create a UserProfile resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated UserProfile is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("UserProfile is deprecated: UserProfile is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.iamUserArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iamUserArn'");
            }
            inputs["allowSelfManagement"] = args ? args.allowSelfManagement : undefined;
            inputs["iamUserArn"] = args ? args.iamUserArn : undefined;
            inputs["sshPublicKey"] = args ? args.sshPublicKey : undefined;
            inputs["sshUsername"] = args ? args.sshUsername : undefined;
        }
        else {
            inputs["allowSelfManagement"] = undefined /*out*/;
            inputs["iamUserArn"] = undefined /*out*/;
            inputs["sshPublicKey"] = undefined /*out*/;
            inputs["sshUsername"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(UserProfile.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing UserProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("UserProfile is deprecated: UserProfile is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new UserProfile(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of UserProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === UserProfile.__pulumiType;
    }
}
exports.UserProfile = UserProfile;
/** @internal */
UserProfile.__pulumiType = 'aws-native:opsworks:UserProfile';
