// @pulumi/aws-native/lambda/eventSourceMapping.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSourceMapping = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Lambda::EventSourceMapping
 */
class EventSourceMapping extends pulumi.CustomResource {
    /**
     * Create a EventSourceMapping resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.functionName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'functionName'");
            }
            inputs["batchSize"] = args ? args.batchSize : undefined;
            inputs["bisectBatchOnFunctionError"] = args ? args.bisectBatchOnFunctionError : undefined;
            inputs["destinationConfig"] = args ? args.destinationConfig : undefined;
            inputs["enabled"] = args ? args.enabled : undefined;
            inputs["eventSourceArn"] = args ? args.eventSourceArn : undefined;
            inputs["filterCriteria"] = args ? args.filterCriteria : undefined;
            inputs["functionName"] = args ? args.functionName : undefined;
            inputs["functionResponseTypes"] = args ? args.functionResponseTypes : undefined;
            inputs["maximumBatchingWindowInSeconds"] = args ? args.maximumBatchingWindowInSeconds : undefined;
            inputs["maximumRecordAgeInSeconds"] = args ? args.maximumRecordAgeInSeconds : undefined;
            inputs["maximumRetryAttempts"] = args ? args.maximumRetryAttempts : undefined;
            inputs["parallelizationFactor"] = args ? args.parallelizationFactor : undefined;
            inputs["queues"] = args ? args.queues : undefined;
            inputs["selfManagedEventSource"] = args ? args.selfManagedEventSource : undefined;
            inputs["sourceAccessConfigurations"] = args ? args.sourceAccessConfigurations : undefined;
            inputs["startingPosition"] = args ? args.startingPosition : undefined;
            inputs["startingPositionTimestamp"] = args ? args.startingPositionTimestamp : undefined;
            inputs["topics"] = args ? args.topics : undefined;
            inputs["tumblingWindowInSeconds"] = args ? args.tumblingWindowInSeconds : undefined;
        }
        else {
            inputs["batchSize"] = undefined /*out*/;
            inputs["bisectBatchOnFunctionError"] = undefined /*out*/;
            inputs["destinationConfig"] = undefined /*out*/;
            inputs["enabled"] = undefined /*out*/;
            inputs["eventSourceArn"] = undefined /*out*/;
            inputs["filterCriteria"] = undefined /*out*/;
            inputs["functionName"] = undefined /*out*/;
            inputs["functionResponseTypes"] = undefined /*out*/;
            inputs["maximumBatchingWindowInSeconds"] = undefined /*out*/;
            inputs["maximumRecordAgeInSeconds"] = undefined /*out*/;
            inputs["maximumRetryAttempts"] = undefined /*out*/;
            inputs["parallelizationFactor"] = undefined /*out*/;
            inputs["queues"] = undefined /*out*/;
            inputs["selfManagedEventSource"] = undefined /*out*/;
            inputs["sourceAccessConfigurations"] = undefined /*out*/;
            inputs["startingPosition"] = undefined /*out*/;
            inputs["startingPositionTimestamp"] = undefined /*out*/;
            inputs["topics"] = undefined /*out*/;
            inputs["tumblingWindowInSeconds"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(EventSourceMapping.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing EventSourceMapping resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EventSourceMapping(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EventSourceMapping.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EventSourceMapping.__pulumiType;
    }
}
exports.EventSourceMapping = EventSourceMapping;
/** @internal */
EventSourceMapping.__pulumiType = 'aws-native:lambda:EventSourceMapping';
