// @pulumi/aws-native/lambda/url.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Url = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::Lambda::Url
 *
 * @deprecated Url is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.
 */
class Url extends pulumi.CustomResource {
    /**
     * Create a Url resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Url is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible. */
    constructor(name, args, opts) {
        pulumi.log.warn("Url is deprecated: Url is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.authorizationType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'authorizationType'");
            }
            if ((!args || args.targetFunctionArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetFunctionArn'");
            }
            inputs["authorizationType"] = args ? args.authorizationType : undefined;
            inputs["cors"] = args ? args.cors : undefined;
            inputs["qualifier"] = args ? args.qualifier : undefined;
            inputs["targetFunctionArn"] = args ? args.targetFunctionArn : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["functionUrl"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["authorizationType"] = undefined /*out*/;
            inputs["cors"] = undefined /*out*/;
            inputs["functionUrl"] = undefined /*out*/;
            inputs["qualifier"] = undefined /*out*/;
            inputs["targetFunctionArn"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Url.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Url resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Url is deprecated: Url is not yet supported by AWS Native, so its creation will currently fail. Please use the classic AWS provider, if possible.");
        return new Url(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Url.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Url.__pulumiType;
    }
}
exports.Url = Url;
/** @internal */
Url.__pulumiType = 'aws-native:lambda:Url';
