// @pulumi/aws-native/kinesisfirehose/deliveryStream.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeliveryStream = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::KinesisFirehose::DeliveryStream
 */
class DeliveryStream extends pulumi.CustomResource {
    /**
     * Create a DeliveryStream resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["amazonopensearchserviceDestinationConfiguration"] = args ? args.amazonopensearchserviceDestinationConfiguration : undefined;
            inputs["deliveryStreamEncryptionConfigurationInput"] = args ? args.deliveryStreamEncryptionConfigurationInput : undefined;
            inputs["deliveryStreamName"] = args ? args.deliveryStreamName : undefined;
            inputs["deliveryStreamType"] = args ? args.deliveryStreamType : undefined;
            inputs["elasticsearchDestinationConfiguration"] = args ? args.elasticsearchDestinationConfiguration : undefined;
            inputs["extendedS3DestinationConfiguration"] = args ? args.extendedS3DestinationConfiguration : undefined;
            inputs["httpEndpointDestinationConfiguration"] = args ? args.httpEndpointDestinationConfiguration : undefined;
            inputs["kinesisStreamSourceConfiguration"] = args ? args.kinesisStreamSourceConfiguration : undefined;
            inputs["redshiftDestinationConfiguration"] = args ? args.redshiftDestinationConfiguration : undefined;
            inputs["s3DestinationConfiguration"] = args ? args.s3DestinationConfiguration : undefined;
            inputs["splunkDestinationConfiguration"] = args ? args.splunkDestinationConfiguration : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["arn"] = undefined /*out*/;
        }
        else {
            inputs["amazonopensearchserviceDestinationConfiguration"] = undefined /*out*/;
            inputs["arn"] = undefined /*out*/;
            inputs["deliveryStreamEncryptionConfigurationInput"] = undefined /*out*/;
            inputs["deliveryStreamName"] = undefined /*out*/;
            inputs["deliveryStreamType"] = undefined /*out*/;
            inputs["elasticsearchDestinationConfiguration"] = undefined /*out*/;
            inputs["extendedS3DestinationConfiguration"] = undefined /*out*/;
            inputs["httpEndpointDestinationConfiguration"] = undefined /*out*/;
            inputs["kinesisStreamSourceConfiguration"] = undefined /*out*/;
            inputs["redshiftDestinationConfiguration"] = undefined /*out*/;
            inputs["s3DestinationConfiguration"] = undefined /*out*/;
            inputs["splunkDestinationConfiguration"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(DeliveryStream.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DeliveryStream resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DeliveryStream(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DeliveryStream.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DeliveryStream.__pulumiType;
    }
}
exports.DeliveryStream = DeliveryStream;
/** @internal */
DeliveryStream.__pulumiType = 'aws-native:kinesisfirehose:DeliveryStream';
