// @pulumi/aws-native/timestream/scheduledQuery.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduledQuery = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The AWS::Timestream::ScheduledQuery resource creates a Timestream Scheduled Query.
 */
class ScheduledQuery extends pulumi.CustomResource {
    /**
     * Create a ScheduledQuery resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.errorReportConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'errorReportConfiguration'");
            }
            if ((!args || args.notificationConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'notificationConfiguration'");
            }
            if ((!args || args.queryString === undefined) && !opts.urn) {
                throw new Error("Missing required property 'queryString'");
            }
            if ((!args || args.scheduleConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scheduleConfiguration'");
            }
            if ((!args || args.scheduledQueryExecutionRoleArn === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scheduledQueryExecutionRoleArn'");
            }
            inputs["clientToken"] = args ? args.clientToken : undefined;
            inputs["errorReportConfiguration"] = args ? args.errorReportConfiguration : undefined;
            inputs["kmsKeyId"] = args ? args.kmsKeyId : undefined;
            inputs["notificationConfiguration"] = args ? args.notificationConfiguration : undefined;
            inputs["queryString"] = args ? args.queryString : undefined;
            inputs["scheduleConfiguration"] = args ? args.scheduleConfiguration : undefined;
            inputs["scheduledQueryExecutionRoleArn"] = args ? args.scheduledQueryExecutionRoleArn : undefined;
            inputs["scheduledQueryName"] = args ? args.scheduledQueryName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["targetConfiguration"] = args ? args.targetConfiguration : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["sQErrorReportConfiguration"] = undefined /*out*/;
            inputs["sQKmsKeyId"] = undefined /*out*/;
            inputs["sQName"] = undefined /*out*/;
            inputs["sQNotificationConfiguration"] = undefined /*out*/;
            inputs["sQQueryString"] = undefined /*out*/;
            inputs["sQScheduleConfiguration"] = undefined /*out*/;
            inputs["sQScheduledQueryExecutionRoleArn"] = undefined /*out*/;
            inputs["sQTargetConfiguration"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["clientToken"] = undefined /*out*/;
            inputs["errorReportConfiguration"] = undefined /*out*/;
            inputs["kmsKeyId"] = undefined /*out*/;
            inputs["notificationConfiguration"] = undefined /*out*/;
            inputs["queryString"] = undefined /*out*/;
            inputs["sQErrorReportConfiguration"] = undefined /*out*/;
            inputs["sQKmsKeyId"] = undefined /*out*/;
            inputs["sQName"] = undefined /*out*/;
            inputs["sQNotificationConfiguration"] = undefined /*out*/;
            inputs["sQQueryString"] = undefined /*out*/;
            inputs["sQScheduleConfiguration"] = undefined /*out*/;
            inputs["sQScheduledQueryExecutionRoleArn"] = undefined /*out*/;
            inputs["sQTargetConfiguration"] = undefined /*out*/;
            inputs["scheduleConfiguration"] = undefined /*out*/;
            inputs["scheduledQueryExecutionRoleArn"] = undefined /*out*/;
            inputs["scheduledQueryName"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["targetConfiguration"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ScheduledQuery.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ScheduledQuery resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ScheduledQuery(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ScheduledQuery.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ScheduledQuery.__pulumiType;
    }
}
exports.ScheduledQuery = ScheduledQuery;
/** @internal */
ScheduledQuery.__pulumiType = 'aws-native:timestream:ScheduledQuery';
