// @pulumi/aws-native/types/enums/rds/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlobalClusterEngine = exports.DBProxyTargetGroupTargetGroupName = exports.DBProxyEngineFamily = exports.DBProxyEndpointTargetRole = exports.DBProxyAuthFormatIAMAuth = exports.DBProxyAuthFormatAuthScheme = void 0;
exports.DBProxyAuthFormatAuthScheme = {
    Secrets: "SECRETS",
};
exports.DBProxyAuthFormatIAMAuth = {
    Disabled: "DISABLED",
    Required: "REQUIRED",
};
exports.DBProxyEndpointTargetRole = {
    ReadWrite: "READ_WRITE",
    ReadOnly: "READ_ONLY",
};
exports.DBProxyEngineFamily = {
    Mysql: "MYSQL",
    Postgresql: "POSTGRESQL",
};
exports.DBProxyTargetGroupTargetGroupName = {
    Default: "default",
};
exports.GlobalClusterEngine = {
    Aurora: "aurora",
    AuroraMysql: "aurora-mysql",
    AuroraPostgresql: "aurora-postgresql",
};
