// @pulumi/aws-native/ssm/resourceDataSync.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceDataSync = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource Type definition for AWS::SSM::ResourceDataSync
 */
class ResourceDataSync extends pulumi.CustomResource {
    /**
     * Create a ResourceDataSync resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["bucketName"] = args ? args.bucketName : undefined;
            inputs["bucketPrefix"] = args ? args.bucketPrefix : undefined;
            inputs["bucketRegion"] = args ? args.bucketRegion : undefined;
            inputs["kMSKeyArn"] = args ? args.kMSKeyArn : undefined;
            inputs["s3Destination"] = args ? args.s3Destination : undefined;
            inputs["syncFormat"] = args ? args.syncFormat : undefined;
            inputs["syncSource"] = args ? args.syncSource : undefined;
            inputs["syncType"] = args ? args.syncType : undefined;
            inputs["syncName"] = undefined /*out*/;
        }
        else {
            inputs["bucketName"] = undefined /*out*/;
            inputs["bucketPrefix"] = undefined /*out*/;
            inputs["bucketRegion"] = undefined /*out*/;
            inputs["kMSKeyArn"] = undefined /*out*/;
            inputs["s3Destination"] = undefined /*out*/;
            inputs["syncFormat"] = undefined /*out*/;
            inputs["syncName"] = undefined /*out*/;
            inputs["syncSource"] = undefined /*out*/;
            inputs["syncType"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ResourceDataSync.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ResourceDataSync resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ResourceDataSync(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ResourceDataSync.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResourceDataSync.__pulumiType;
    }
}
exports.ResourceDataSync = ResourceDataSync;
/** @internal */
ResourceDataSync.__pulumiType = 'aws-native:ssm:ResourceDataSync';
