// @pulumi/aws-native/location/map.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Map = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Definition of AWS::Location::Map Resource Type
 */
class Map extends pulumi.CustomResource {
    /**
     * Create a Map resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.configuration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'configuration'");
            }
            if ((!args || args.pricingPlan === undefined) && !opts.urn) {
                throw new Error("Missing required property 'pricingPlan'");
            }
            inputs["configuration"] = args ? args.configuration : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["mapName"] = args ? args.mapName : undefined;
            inputs["pricingPlan"] = args ? args.pricingPlan : undefined;
            inputs["arn"] = undefined /*out*/;
            inputs["createTime"] = undefined /*out*/;
            inputs["dataSource"] = undefined /*out*/;
            inputs["mapArn"] = undefined /*out*/;
            inputs["updateTime"] = undefined /*out*/;
        }
        else {
            inputs["arn"] = undefined /*out*/;
            inputs["configuration"] = undefined /*out*/;
            inputs["createTime"] = undefined /*out*/;
            inputs["dataSource"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["mapArn"] = undefined /*out*/;
            inputs["mapName"] = undefined /*out*/;
            inputs["pricingPlan"] = undefined /*out*/;
            inputs["updateTime"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Map.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Map resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Map(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Map.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Map.__pulumiType;
    }
}
exports.Map = Map;
/** @internal */
Map.__pulumiType = 'aws-native:location:Map';
