// @pulumi/aws-native/signer/profilePermission.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfilePermission = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An example resource schema demonstrating some basic constructs and validation rules.
 */
class ProfilePermission extends pulumi.CustomResource {
    /**
     * Create a ProfilePermission resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.action === undefined) && !opts.urn) {
                throw new Error("Missing required property 'action'");
            }
            if ((!args || args.principal === undefined) && !opts.urn) {
                throw new Error("Missing required property 'principal'");
            }
            if ((!args || args.profileName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'profileName'");
            }
            if ((!args || args.statementId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'statementId'");
            }
            inputs["action"] = args ? args.action : undefined;
            inputs["principal"] = args ? args.principal : undefined;
            inputs["profileName"] = args ? args.profileName : undefined;
            inputs["profileVersion"] = args ? args.profileVersion : undefined;
            inputs["statementId"] = args ? args.statementId : undefined;
        }
        else {
            inputs["action"] = undefined /*out*/;
            inputs["principal"] = undefined /*out*/;
            inputs["profileName"] = undefined /*out*/;
            inputs["profileVersion"] = undefined /*out*/;
            inputs["statementId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ProfilePermission.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ProfilePermission resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ProfilePermission(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ProfilePermission.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ProfilePermission.__pulumiType;
    }
}
exports.ProfilePermission = ProfilePermission;
/** @internal */
ProfilePermission.__pulumiType = 'aws-native:signer:ProfilePermission';
